{
 "cells": [
  {
   "attachments": {
    "image-2.png": {
     "image/png": "iVBORw0KGgoAAAANSUhEUgAAAGUAAABfCAYAAADxnQkEAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAEjASURBVHhezX0FgFzV2fYzPuu+WctKNu5GQkiCBAkEhwJtKZQibSnaYqWlhRZaSv0rLQQLWtwpFiQeQtxdN7tZdxmf+Z/n3JnN4oHvg/7v7t25c+XIq897zrl3bUOHDo8hTjabHSkZ+dxxwOFwIBaLmS0ajfZ+2qL2+NVfH7FWtiEa/2bjxjpj+q79ONmsfTYrscsv3OEBO6+LJu7nSbvdbo6JEv1IkG4R2Xq58PVRJBJEMNDKSg/W/2nUy2EJxOXJhsPmRiQYRTgQRsgf5mcE0RDZFOY1X7NAEoyxk1N21hWLOdgBOyJkaNjmRMDuQpCHQmRyiO0NUXnCZp8KxPN2KpIjhde7Q3C5bXB4XHA5nXA7XSrGbB8n1flNCETkcLjhJo/F688jYym6yO3NoUY544e/XkrwIKHgCdLxqJSdumL3eJFTXIScAUOQVViAtKwCJGdmwJWchKSkZKuttIZIJIKg34dIexe6uzrR3lqP9tpatOzfhZb6A4h0+yym0zqi1FAZkKzFlrA0brb432+KopEQLaaF7fh0i6FQRsQ83mxqmTt+6Oskdt8WISfEAO3zr/wPd6JuL7LKK1A+bizKh49DVlk5XCnJvMplro3EmZhgnRgrSjA31OM3VhGLhlUyNwksjNaqKuzdtAZVK1ehft8u2OlCyBZTL23JSEXXJ6zom7Ia48r8Ldz7eIVs1+gx02IOpyd+4OslsSGmXuuXvNT3zNKBGDT1SAw/YhpScvLpjsgouiTx3KHrE8zX/dwPRenM7PpmMwJx8FpRLBQ27osRQ8WTLGFJZmK6IxpDR3M9Ni9ZjF0fLELb3p20RwqQysqSLD3R9dbHN0LhsB/hYBtbJ05YfTXtHTfhWKsPXzNJFhH+iZELDiShZOJEjDxxFkqHj0eEzIyQ0TEyzhH3t9o3jY1YJq77JAw748qhkDooedrZyzDlLOY7WSZlxmNh1Gxciw3vvoVdqxbCFWKbeHHUxtijC75Bcjij6O5sYIPjB9Tnr1soicL1GXY6MGDiVEw6/TxkVFZQOArSZBY/I0ZF+YdCkO/XDbIEISdRAk055XIOkVSnYpSdO9qiUgoJi/XJenRF696tWPHyS6hasRSOcMDcpDPS3K+HWDu9QQIBehkje7qaEfC1m+6L/s+FknADIuN6GENkGzkDRuKo712E/JEjecJunIwYJWSVYLyCdmI/ESu+TlLHFWzttKDm7Zsw74n70LRzu7EaWfb/iWBo3cbNcrMEkfi0yOGgQrLfAX8rAQCVgvQ1CcUqUn9tyek44lvfwdATTobT4aK2kumJvvICo7UmRvz3yDhKSsEeDNKlvY4VTz2FaI801/L1X5YkBOmUFCwit0ghSMifR1HGymCgmW6bavp/LxS2Rv7f40FexUCccMnVSC7sT6ugq3LxHGuTLcitWESh6J7/Iql2wwQpSTSE7tq9eGP2vWjdvpFWw77EBWMsXwrEX+VS5jvJtD9+3KZYKEGYM/xuXXJIFDXJZQschUUDbosf+4qkWtkaQ2yKWmH3YPzxJ+O4H14NZ0YBvZU00bDfCMTsqxMybXPkv0uJ1ltEEJ6aiZHTj0KgO4CGXdt5RH2UcCyrdioZ7eOCJBRZG6XB44mDH/k4JLIx1kjO/0uhqLFqFPMD5XJseNjlxVEX/BDjTzsPYU+SajJblKZsrIg+1Bwzzf0yTf4GiM2xG5fF+MauDRg3AcnpGajeuhFOOyOj020geAKIKB4ZiM9+2ahpNnoDy3WxBFPOl++f0OVXFooqls+UdG2EmGE1MikDJ1/xc1QcdTQhLv2pQ42jFvEjxv0Ytey/bxefQ0R+4YAf0XDEQPIot+yKSuSXFWL3uvVUuRCDMoGB8l/po2G+buQxbuqn/IGR2FcQiEh8/UKhmIq4Ce8LKdic0nQekKlyC1EjIkRYGhU4+Wc3o2jCYbzHclQuBg4DdXU//8R3/78muSUhJfVXQdrOPqcXlKGkchi2L1uKaCBg0FqQbHBSSGNKOnHeTBeOPSwEj8OHmvoo/HaBfFmMCqQEZU1GWIdGvUJRAaaQOInhRgiyBp1gudKcaFwY/CWx8bTkqDcNJ193PfJHTeAhXi9T5lkjEMmQP05eb4ZU+lbyNZCaZdVgGhgn68in1ay+mP6J+BmJhmF3CSWy6W4X9+WaHUjOz0dJxQBsX7WSAbmHN9px5MhW3HVtMSaMSMLwAS4cPc4FtyOMdZsjhPoajxCpz/o8dKHwSiUzagGZycw6QWqohBCjuZrNkkKcZKo0ce0x6J146dUoGDWZdTMd5H0G4spVcVPT1Bwz0Kjj2pcvjoXNZhr9FUi1Wy3oS2IAy9YZuVRqjKmL2p9wR5ZC8Trua+sVSJxcbq9xyYofdhtbT+YLOdq432/MZBz7o58g5PQgx9mFq84rQFpGEhxpg2BLGQEX4f95xyahKCvIGBqlUoqf3DTa/SWISuGCNknzo4z/DEpcoj6x0knnno/K6dOMUCWQLyJZmiqLmUCj67/4nk8jWZ9ljx8leQrmy8gPVqHCvwCeaCcc1Ahl89okBG1SGKOMh0BWK63YWHb4kZh27rdx1rFOlPZPoQ/LZ99z2fdMXpFKXkaQlRuDk0jBeRCgfSmySxAmw2Qhxr0cClFzokRc5YdPxbhTz0WQ3tVq9hdTlAyJUJu90QCSot08coh1foyke6pVVmwCs7E+tiISRmlwIab7fodj/f/AqNAzjBE+nTDCSpD2vjzP2EObCzNmjcS3z8yzLMiVzaOac2oFwp3opvEfYFyJCt5qRPwrkF0pfigkV2KZ9xcTzZ9/k/sV47iLf0z5iD3UOh48mBB+Ouk+GwOfAEBJ7H2Uh18zHTqUWvuSGaykMILBkOWajDJpbCCKoshqTPLfj4IkL5JS+mFo4G1kRzcbrVWMs1pvkfYT9yaOWnt9N4vMcV7roSIND76NTC/rddDDyA0H9lMoO4jcOvDkGz2o60hlP8WMQ1HTTxItJb53CGQ1kxZFV3X6ZechP6UF6dFaBvEgtUJ69/mFqYnKhDU30mkrJrN20bV0GgsVcwyDxHBeZ7SY+9FwEJFQEGEKIBqi5gUDiGnjvoOCMdeYa6PIiOzChMBsZDrJKE8e4M5BqjOCwWSiSwFcF3Iz9dCyPNEmpEeqWIQUg0iSJ20xP88FjPKY3pp2MT7wx04B9AuvQUV0FRVAsTKAiG8n21XLW9rx/IJuPPJmmPdaE3Cyzi9NisNfJk+x/HUYZ5yQjMsn78HQyCsoiSxi58JocPXnFYl5mU/XEIvZmtuwIWBPR25kO9oc5WRKGi3OmkMxd8ZjU4xuyUZLFsPVR14RZyg3IiLl2HAyAyfic8KHCbSQ8tgmnkqmJXXBFvazjwE4wwdQ5ZxGqJqmwk35Gr4fFHgfmVSMBtcIOOiKMinUUf5nkRtdTYFnoNueY9ytITbAjXZM8t2DfFszG+NmSWpwCHZfD976IIT/eTKMQISxhffoLonts0jFut1udu2g4IR2Tdvi37+QrLgcw9ASoo7T/chMaYfXm4MCdxBjIs+gIrCcjTA6+5kkoRpLocqGbB402EchJ7wLAQeTNmmk6uB1JmMmKXuWwjpdHkR1jZGME06P10BVp4fZrxAjtaswvA4VoblwhlrYjjb6ZVpgmPsRP9Ii1VSAjWYCzbgwVuKKBGjl29DoGMnrnWx5FOWh9zA28Cgmk/HT/b/mPevZHo1xy4JsGBiYj9LQKjZIjpKsi4QAXwsWbYzgrn/70B2k0NlGozhGLOKx9SkS0x1uO9xJhM5JRHcaC4xfJ1JiKrd8yEJRTySU444dgNRkStNVAWdSJSLuUmJzG333a+xoFzth9Iebfiyy5qJ1PytlGfaoBREbHEPZOcaYMLVNViH3xHp6IYM0jsyPqjMeNztEH04LSUBrkamH5eaHV9NiO3h7F6I9+7jtQNRHP++vNgLIjW6Ci67J0m7GBtQRoKSjxT6QR1gPDzfZx6DaNQ2hWBEyojsx1P867wmYxRkZkXqMCL5IAfXwepYTbUUk2IB9tpF433c22gNZiDgsNyg5uNhuD5UnOTmJn27zaVAu+ycLUSyPmqkK1t1HcKIvFErCQtT91OJSDJ8yhYrJhMpOL+twwuHKYIxJRUFsJ4qDawjirNzATjSnzfhkoROjPeKH7ImNJ0rxxzIQjOUYTVSCZtNGAQhaq2Fivs1tBtX4awEKCcAksYo9pi6YibJmx2haUpaxJN0cc4R4NdtBNwV7EN5IN4O9yuUxVp5FJtc5DmMfmJfwGFN31Lon492U3+D95DvQYhuBfMYOZ4z3UZgDwy8hm8Ec0SBs3TVsfB1qXIPxYerlyD/mQmSVDjABWlBfXRXjxXDNJtldVg6osT+qF/zdfgqUACtAHso64rxJ0CFYinWDmDH5tPNQkzYZXZEURP2NZHo7mUzBOJPhIRIZEnodXvpdIR3dpU06oKBpZ6BGMMyM2ZpjlyY5maB1OUtpCTz20XZ9KmnVjZK5BMpLlM/UAA3OERRMJdrtpai1TTUMb3UMJKAowT5q/07P8bJJc5OZmyezm8lUlZBQVMUPP2NdlXsiPky5iMyidhPEuGM+VPoWkp2MU2yoFLWOcehD70/RbBuOkMuLMaeegQj71Jf8WmVDiB4MBtHd3Q1/wI8wPYIlCDblY8JI0BcEet7EBsi83f1KcMwPfowuVwGZ7kNBZB3dEJkpuTqCxOg+BvAGNGMI2h2FFk63h5Ad2YER/ueRxjOt3lIKxEPtVci2mBCiGDW076BwNeWrcTMpAFnFK+LcIhmEZtpCCZAxjqidsYktk1WwHEcshKLwBnPLSu9lWJf8Pex3HGkARZVrBva7pvA6S1s9dHFhCqjH3o/XWxZo6tAPy3ZS8D7el8O21zinwGNvxyjf87yjjY22080Nw4een6GOMVH3C4nlF/fHlqWLmap0mNLEfPE8EmafNOQsVqrTh0BfYCnywPoBJhx7OuBNoX9NxmbvGdiGMZR4O9WhBraubrLKg2R2ZnDwPXYqwFvDSIk2YnrPnzE69ATKo2/DHdawSh9Gxz+JXxFk8Jdrc9K1uJlYSlzsEt2UFfRlHU50Uxk2Wm6CPdSCPVMKfzWj2WHPo7fPg8+ZQ7CajQ5HkWF8o3OQOW/IuE8HOoX6eN9BPrFMlivGBaTw0STUukYZIQ7ueZtusgdNseHY4ToWi1OuQY17vAEZAiNa7BF2OTDq6Jnx8qg6bHfCZX9Z+gKhiOgunE5UTJ9iGuwgfAlQ7zd5LkFNqJiVa7GB/JVighf5kQ+RGdsGZ8SOotBWI5i17p9guecmBG0pfURiMcQsZtA+O58Sa0JleC6mhB7DYf6HURhaTcQboBDIRm6pwWYK/SW4Iz28j1qtyTPVTZJr6qYAQgzekRjrMQxxodlZZrReAmeTeA+TTnsK92WxGtvidXKvyk1o+WIkbdAo2QHbRGQHNyIvsBlVnqmYl34X5qf/Ck2ucXS5NvaHlmdqJ5eokCOmHYuYN4ll0joUp0iJ81+GvnjonmysnDAVw46daRorTZNqBuwZ6HLkoJ9/HbzODraKHaS78obqzBLTWtdE5Ia30pWVYG3Sd9DjzOC9vMZI5aBooowvbrbfS00cH3gak5zr0M/rQz/nfhT1zKMXzzOMldBchKBF4ZVESJNYR5IBBCpKfBXUTmYySKdEfz+awd9SgCAtW8og7RODzCQWtdu4XR5Q4li3fR3+/cubsPj5R1lHBKWDR5t2ySWWhVag2T0A2zynsb/9MDiwCOXht4jQUtDN7waAmHbQpXo8qN+1A+21+9leacDHnfCh0SeEEue5KcgkM0RYY848G9nllZS90A+7w4u0lspnK+D3GJm/mNCdMUX3BVuQEmmjZk3m9Ul0E0XosjGxpCYZlGPciNVM/VV5GlHNDW7BhMgTSBLaivgIj8PwksmOUD0TvyOoDtRAoqgs+vk65xiWxFxF99MHyQrIZ37K7QXRQncVYR4U5jEpiK7TChVZdNT4dyEkoh9C8UhnC+6/8hI07lyLnpZqrJ03lxA2GQPHTECqfS/aCByaXMMpkCLW2GpGIbKYW/nsuXSPA9gtxUe5eW7cd9Ft7Vq+lKL6agIRqb29ZEZPBUmVF9BfCnZGmLkOGMn4wU5J09gjBIU62bEg4ec2zyzscp7I2NKJWM9+ngjSue1HJbPlFmcluuwlFKAGAztZwUdjikoUc2LMW7yxZiQx2UM3YWc3k7YuAgnGq9RYPdulGBXi1W7Gjf7kqSSgEg6WFWUZPfZC+Bx5LFOL6ixSOxl1EaAAtIpSi/v0qWEaNy9a8dIzaG/ci9Fjx+L88y+Dl7nQq/ffhZbd29EdKyWCG45uW75paw8Ksd57Nuan/AJ7nTNUuqlHVVjWYkPJqLE0e2X71s9XIUUp5hp69MGST0xogZtJbiiI7AEVcGVl8cJ4NRQEr+ad/EufKv+8Num7RDdHsFHGlqixdvSPzDforNuehazobkzv/iv6B5dRALrGYqe0S1pOllLziuBTA4I1dPHNPNPOuOHjGeZEFIKDvQ/Z3Oix5RF5UftlYSwhMQQvlxR0pBur1JSCytRSVVmm5kJUm5RNIYhqJxCFYHsT5j7+AFOPMMaQmW63B0dMP5o5hB8LX36K8DiVuWoHrbMa/aJriMa209UGaX1pJqENsw1qvQZjtSnmujIzkUuvwt2vTIyhsT7Y+ZNFlQ0dwQqI7/tQQv5G09l55QLLky5Bi2MID7p5Qci4MJetlQyNoNK/gFrZSRjqjd9nbYbiEmpwDcBux0y6HRfBhBQkStdRQdc1xQRxkeTns2WboZJPEAtUiO6y5/DTZZhkwBkLl/CFkpTA2T0UMpXQSQ+w8MUn0dHShEGDhhhhPf74oygqKkIyM/GVi9+HgzA/M1SNwvAijOt+Bid2XY8jfH9l7NLCbKvsvmS+sZHlI8ZZ3z7JzkMixT0j4QT17nJHmtBvwFAKSxWwh9r6kGkEtVT5QrNzKFa6r0aDfThdTDHBawmR2Cbjy+uY2C3x3oAGxwjaWJjslo6LVapblsXKKMxt3jOZuE3FLs8M7HUdh83u81DtPJwM1tgUtZ1AoceZx7u4T2VSu02LZBF0g/oMEgBozbI5Y+IY95y0apeHO3TJbK9GCkKtDZj39KPISElGUlKSGXvr9nejZn8VSgpL0F1XB3+PnznJUGzxnIeFKcxLXEMxKPA6hbTYDBW5FCItLhhSrYLeORWDyDt+OXjqS9FHuUxS4FVpJoGjdhVUlhsGHiRr3zpGVhERJSn/4H17PFOwIOVmrPL+COuTT2PI9bCkEGq7B+Pxv/0dvz7zBPzzyu/DV71XFZlyestjfa2O/tjhOBV7XEdjn3M6E7fD6EKI2kS8XGLsIby1plktslYfRmEP+WEL+Mgoul1ap3IQjUZ/nC/G3fHggucfR7ClkW5rFNrbiR5ZutfrRVNjI1LTWIfikM/PujQd7CDSKsB251lUvkpemm5AhDHoOKkXibryBxAtWg22ti9JnxRKvCRV6E7JgjebQU6+Uz3hJvCi2b2kCHOKwKuY5vsjJvn/hmGBV5AR3cdGD0a9awwRyxBaxlREunvwj5/+ECv+/Qh8B6qwb8l7+Pfvfs0qKEhpOJsgjK9PAYcaxib57GYG9G7kwG/LpaUy69cVsirGE1mKVtBoaVPtppV48Kaf4NbTp+NXpx6Bf1x2Lra99Qo8mn/p6wLipP4Fm2rx/nOPIDc3F8X9yUBe1tbahn65eRRGiH1V1CRUd9INss9mQJSxrcYzCe8k/wkHnFOpCBbTVUNio+EYhqaSZ940ulnlUQlJfQaZ86zDjCDLvXL/U4RikZbRpGVnGYEkyNqTBXVgcvc9OLrndxTGczTp94jnX8co39MY5FvEYJxNHJ9BJjvw4uzZqF63GKl5mZh10knIysvBzuWL0LxniwEsXsLYfqH1cDEnsFPrfTYvGmJl6I7mMfAnmWAb7msZvEluVQ3/8Jk5+OslZ2Hr3BfJ1SYk0Vrq1q3AozddgXuvuww93Qcod7nKgyR3Ov+pxxBpacaY0RPYBuY3KSnYtGkTDps4Gf2KShGhG3OlJpuHlvqSAESXq5DoUzkZFUSjEB8pnTzSV5aZVVxITklMnyT1W2vmJAgDPvijmG7AFfn+CaEoQJpyeSq1X5HxvwbpsCBVIcFkRavRP/Qer/UTGh6PN1P+hvnJf8TKlCupTcPZ8TCFkoNAQyOWvfw0kljxhNHj4PMFMHDgUMKobnxIdBML9iDMxuT4l8MbPsAch40jXO0AM3MKI0oBRUwcsFpkLJa91khv9br1eOYvf4CTHZkyZSrOOuc7OPuc75ltxMgh2LZsLh6+/moigy62Rd1WGTH4G+sw//knUNq/FDm5Oabcsv4l2Lt7F1JTUjFh/ERU1+xH6UgKzJMSv0sarXE+urhQEDmhzRjhfwGj/M+gOLwUnli74Zs1Pid3SkHn5VoCipME4KQwFd8oEeN2Dbgi0uUtH6FPCMXywtrsSElN5y739TVOZpQ2moRuZwFq3IdhRdKPiJwGE/pmE45mMoMvYlKVz3zBg2XvvoVoTxPKKkdg29bt2LN7N9KTU5BBZmyY9z5svm4zT1TvHo90XxW7LIjLFtq96HZlEi3RrKk5Dh5KNMFqXQxr5r9JIXZj4KBBGDBgIIqL+tMd5aG8ohzHHXsypk86ArtXfIBFTz9oRgzMvWTamw/NhqOnDaNGjabbstQsJTkZJSUlePHF5/Hy88+gpcuPo84+n/mVXKumADS4aGc53RgWegIzu2/E1NDDmBJ5DscH/onDe2YjFVW8jkwms+WVk9IyjAukThurkBVYKFfW8DEpfIw+IZS+5E4mYpG/ZyXShMTF7YSq76T8CQu8v+f+AGo0YSaP6xrTEO5LeLvWrTbILTsnC3X1ddQWG7WwBgPKytFefwDbli8z7qjFUU505ITXSaG4U8wiOKfGpmglZkiEPTNTrNzk7fVTNmS4GX/auWM79tdUo6a2GmHmEBphSElNw/iJh2HYwEF458k5iPnaDcPad+/FsleexNDhI5FEl6XRalFqehYOm3wEMjMzsGfPbpSMGI/BU45CcqQDA4LvEzwo8Y2gLDgPE3seQia/Ox1pcCaVIDltEAZ56jC651VaexfrsUgPyxr1IU8sIVhnZD19LejTKMFnQ7o2Uaj23W4rr0iQjsmNaSau3SmLKGTAtZn5E7NahFIxZfCPi1rYTleRkpJG7SAMpoZkMgltamxAcXF/FhTGsvdeg83NoM16WrxlbIweElUB+lVLEq0hqXfxDyGbscefglN+8ksKwoFlSxeirq4WVfv2obury1yUmppCVzQBgc5WVG3fShcRxssP/BGZtIrKyoFWWfQbbrcT+XmFyMrMMkty7amZuOCW38HrsaEgsoLAxk6r97I/HRjqm8twH6DLpbZzCwSCCEW8VKh+KLNtRGa4xrRYwdvlcpt2fhXqFYpBAaZE+U6aK/f7rpiUMHTMukEXEgWR+0mxNqRgL4+HYaN/1BosLQZQwOohpHRS2/093XCw8+mZ6fATtjoJPYtLS7B58Xz0EJbSU6OHcSSqldMqW9YpogDkb5Wgqm7zzCQPC46GHE4cc8kVOO6KaxEJ2PHBB4vgZ4w6ULsfzUwI5W7kznJoBfu3bkPVyhVY//7rFNQ48xSx/IrGqoTAWgkSFtOd7q1txPdv/wtyhwxkn/awTjf2eY6gwtmRRDTnstdYjLZTMKEWxrNOuqpWFhUiYOnk+XoDodV+BwVsePoVqFcovWRsS53nj/hDS7B09CAl6jJTvRRAbnAzcgIb4WbjnOywXc9vMGsWepPpykJGMR/Yt3cvcnNyqIXJGFw5DPbuTqx75w24mKLLlfjpEqzarc1Ai7AfDRs3Ytv7c4ms6A7pItyCraZ9Dpx40bU4/MKL0dbZiQ3r15v2txBZtbe1mWGT3OwcdFTvwLN/+jXGDx+ObFpEQvB6zsTv92PevHexpbYGF97xNwyfcRIDd48ZRai2HUZLoQDZz4CTthtN5X6QbpRIkcroQCucoWbAz7wr3GagvBpu2q/cId6LL0u9QonLgr+aMScqJz80jWmhCl4Y3xK+XU8sKXD6HDnY6z2WyXYG0lANJ90WbzVllVUOQU9PD4WQZJDNzl27UFExABnJaSYw5+TlYcWbL1Cw1DwKRVPL0kQzjEIEsPKVJ/Dbs47GnefPwgPXfR9/vnAWfnf2cVj77mumHtUfpjWfcdUvMHzKCagikOju7DFxqLmlBaFQCClJKVj56nNw+X2oJPJT/1SHqrAxKVy4cB4OtLbiqn8+hlEzZtEdscyoB+2xUnoH5inMi6SYOZHdZtjIZqdHYAlacR8LtTHUNFBOVWiwFTM2llkl8zdsEmoy4SvQRyxFZs36LabrRQF0RXb6Yk3+CKn0HRsTolAgVgskxAZXKdptpSaH0OChOj7y6BPYSTs7/j6WLFqICgb4wqIiZGVl04dnYtCgwajauA4HNm+KK5VxDnAw4Xzy1z/Dk7ddh0BrA8YOH4zJEydiUEUZump24pFbrsbSZ5+kklg5iJ0W8e1b7gD9oxG82imU00nrUZaenZWJww47jP2ycgLVEyWq2rlzC/bu3IGcgmJ01O4z/Y3ReiK0WLt5QEhT2nZkhRsx3Pcf+GO5CMYI1820AFVPAKCnFk3RZKxKuoRiyjCKqxoCPVrUx52vQI7C/hW3ub1uaHMRAkooKR4iHyIgd0YGBk+cZhCSSWniAkuQ9qx6peXSQPnq+AlqSVZ+AbauXIZWMnLUqLEYPmyUGYPKIMpxe63h9QP7diNA0xg97WgzLiULfe3uO/HhS49gxJCROHzyZBQUUJB0Q0WF5cgv7I+aqj3YvmwxJp92FpO8dDKRPj81A7GuTmxkbBk4sJJNIGygpaelZ6C4pL/JlhNWLt0SdN20aTOmHXkcg78XS954GXVV+zD6CLbDxFJr3MxDN1URfpWuWaPWbmz0nmOmmP22PLOQcKf7KKzzXIlG53CTU8nl67GPXUvmo6lqpxHSlyVjKeb1GWypNicbpIAaJGLqbmNSpI4ogdRmhGMJwrIay4VZw+9x4q6+GsxOWHvqlTcg4kyhO4rQhxOvM2A2NjEQ8yc7OxuVg4dh+TsvMfuuNy6xZvNazH/yUYwaNgIjR47mPVp1Kat0moHD7MxsFBYyq/Z3YPf6dUZhlMuoNZNPOxsBuqyw5k3iAhDJqhOU6GcgEMCkSYTNwwYzs5+IGVNnYMu817F9+ULjdVSivES/8AbGvCj8tIJV3iuxh0JYwc93U37HtOBOJsxXoNk10LhRpQOs1NTb0U5XZ3j05clu3kxEaOXvDrChIfjItB66j6gviNYD1TRJmrTpshoqClP6IaRH9iMtXI0UMwVLGMrgZx43UEBheYn2VI6djOO/fznWbduGuoZ6nUJHRzt8NO9Uanll5SAksY6Vr75JtxfFkleeQVa6F4OHjLIKYAcz6Jb6E0ZHIgGsWfUBqpl925OzGbMG0bIYg8IMvoxBVVs3myF6kRTIbMbDmijXS2JaEmONNvP2JpcdmTksj3VsnDeX7WC8YF+8qIMbzfDZs7De81002UsR4n7IkUr3RQu1E5jEBH2VZEoNLSWI2oNor67SUfP9y5K5S8FYpLQ/qtRfX/inp7OLAuru/W7taEHEWswM/B6nhH+LmaGbcZTvDowIvEicXmu0wxIiNYafIQaYmT+8BhNPOA0rln6ILuYR0tTGxkbTBUHSkSNGYsELj8BB7V+3+D0MZpZuJXZkHnOY9KwMVFfvxbz5b2HP7p1IKi3Dz2Y/gpSiMgS1+DsSRGf1brxx913oR/BgLIOMN1ucBNM72ttNrEmQkKHPZ6bWzDo0h9OBTl4j0itD3FEfArYC7HWdaKZ/LV+kMg+Wa6zR/CYsk33u6ERXa2LO5cuTeT7lk1ZmYWwnta913y5qgJ2BkRepcp6wowMpLh+yUopRyEx3MFbiCP+fcUTgdqQJjfRptIMdt7uScPav7kL5tGOw7MMPyIww3UwQ9RRMelo6Bg0YAldXO+Y//TACzU2MHcVGrNLWDMa1xvp6LF+0AAeq61E4/kjcNOcVFIw6jO2g2+1sQceOLZj3xEMoKchGUXGRgeViuMmkCYJam1qxePECdPf4TGxJkPpuCSlGZemgW61HZnE/kxxr+jpCW9nvnMgYIijsNFbZS8b6PmqBIkGJlr17yNiw6cOhkvht0jMq8WfblwTASmt3b+39LtLNXURZEeL7MPF5LNaNaMjH3MGHnFANMqNVvEqrXnS1imfwj/qxe8VSpBIltTF/2Lxls06io7PDfM/Ly8eEcZPwxsP3IJ3Q2clsWAWoSjFt06b12N9Yi+EzTyd0fQDe/GJjkZveeRZ/u/AczP3nH7DolSdRkF1g3JyYnaDWtmYmlkswevQY5Ofnx48eJMUp5So1NftpjTW0tGJWTYZGktFpL6K7tYQoRKnpjIQrlyYrydaCq3hnDSkeNe4hD4wfjzPtU0jKoQHKj346abGE4fFrekmFajOjatwObNtOrfmoRviceUyb8pnY1hOr19GfdzFfScdu19GMD1WMMcwViMYidsaZWABP//5mvHzHzfD2tOHEE2aiob7BTCCp/KamZnT7elA2oBzjR45HTnYm3Q9FyQa6PV5eW4vNhMylE4/GBb/5A5yMJVKpqs2r8fQfbsPhowahKDNdTz+Z8SzrEfKENUSxbsMGDGSGnk4U1sebGRIsjxAG72aMWkrBZbGc/9z7Fyx85AEzXSCob4RAqwgT6BiNpGdwMZMvDm7EON9jGOR/m6e1qENuX3yyo3rzOhPPPksmhvEsT4LQynuzQlTJKRkfCX+epZBUZu3mLYj2KK5IUpbAwrFUopAjEAoeYBtrGfxcDIQXYHXyD1HjPNKsVI86QnBEnNjwNhHN+29j+hFTUFRQaIbLJ06c0BvHVG5zczMCBBiDhwzhuYnGN8v1CNJu3LQOrpxcfP+OP8PhtWYhxaolzz+J7PR0pKSmopvARPd8kgcxtLW3MC/SfR89m5aWRp2IYP2GNXj33beY7IUwbdp0TJs8Ca/Ovhste7byDqWaAgx68iaAzEgNhva8g1GBZ5AXWUHVy0WDc6SB8gaX0nqi/m7sozs1ZLHMUF+FSIxOK7lNWLX6GuImz/C5QhHFGHyrCVOF2sVA+XlZpqDhfufR9LwD0Gwbad7taGMzu+zZTLByjUBizH6Xv/EqhpSXkqEuMjAF/fuX0l0VkFEa7qBeyX9zEwCQoFR2hFm1iXX8UrX/AE6/6iZ4sopYu7Jr6j+3mq3b6XpcBq6L1BkNFH6UHAZh7d5VZeC27Eeb/H41c5K5b72Jd98j2iJUn3HMiRg8cATy+xUgxR3GVuZBGmNT6Q6iPlesCU5bB2rdQ7DZeyZzkwuwzXuyWRemjF9jXpp+btjORLhbYIHMpc9LuKa+SiFAFWLKoWVbeken3tkpFCzvIqX/XKFIhprc2bVmtRGJSJ8BVww9tnwsS7kZr6bdh/kpP0dDbCrRVwt9Lyti/fK/QjBtTbXwJDNOsCp3ihbU2U2e4aYJJyzCrLuVNOKU0Hpl5CGPG6OOOIYttTTWdE2KwX6GzPshY0hOTjaNbWxo0NmP0NixY82k1dx33sQKJrIfLl+Ct2m9c995AwcI+YcPG4ZjjjmObSqgu2QC7fUy+YuhvbXNuKCIPcK6XMxTStBoH4Z22wAisWzLvan9bIfmewxooAVsXbqUiWacs2S+5lD69k1kFI7bZ9EXWooy7O0fLGO2rMcexA8yNyaGOtnAfLTby9DqGIJaz0jUOwbQJ5LBGjPjZ5ANyszJREebBQ+D/qBaZJ7d0HCLh5/BoB/79u0xY0UHdcmiYFCPxvnw2uy/Ec3ouRIe5KaP0opyNLc0M8mNIIWuyEPBbN25jYxgOVrZEqd8gogZxxyLbELlA3UH0NraaoQ4dsw4nHTSKUxQNWrsNvWr3OaGZnQRpWUWFfIblcMs7HMZbaDbp1KpVPl/TfbxDsYA9dmuRekdLdixYiE6mIN1tTahieCktp4AorYK7USJppA4WbyMf/kYfe5aYkvgrJzoKrukFGkDBoqnPNHHf2uf2i/3FgnoSSn5RI2VESSsW4Et77+D5tpq9C+rMB3XyK3H4zGjCGmMCQpuenZj5crVZGAuXBSUVaxlRSkOO9avXY0pZGCUyaZgo4M+TC9GWPXma0hNSzFjaXLTe/fuRSqDfaZGgnm/RdYqFSlBV0cXpk87GqVl5chhnJJyWGHVhmwCjE4mtQsXvoMuvx3fuv4XcLN9xl+SjADU63ixapsWcuhuJ/vwFMHMh888hP55aTh8/Ggcd+yROO3Uk/HjH12K00+bhYfmPIgMAgstKv8i+vwF3myA/KUWh3W0tGLk0XIjhLgko7XGKtm4uJBorTwhKwLWvPUSHvvNjagozEEV/besJj+/H+NGt0E8HjJK8SSFEFjISA/YrCXzy8sreD9FzEKU0GWReS2NDWjYX4VAUwvyC4tgZ0KZw1xmxbv/QS3zqMrKSuTl55qAv23rNiMEja9JMTTjmZ6SZtq4c+cuwxjNCpoUWfWQ6Rlkfic9wcKFC9nWA5h56VUYdtRxZgwryvPqq1lJQ8nrKeC6XRux5tUXsejfc/Dh3Jfha6hBuc2HFx97EN897xycOPN4HD5lMkaNGo6i/sUM6AHMvv8B5lzZjC8fXdj4afSFq+7ZG6Mt3cwnyoeMQHaOy5rcoaZoAM4EWv7qxZzmjQ60AEZt3H/tpRhaVogBlYMpgCRs2rjJwMDsnGwzY9fBrFeBXahDzEtK8jJ/2UhYmm0EJdJiAzF7f1UVuhtqEaQreOHh+zHmyGlwZxfRJeVgxduvIEoLLCCy68fN7XZhw4b12Ll7O2qrq01ukpqaZtqoAdflyz+gNTrgZX4iFOTr6SIk3oGVa1agvasTU844H7N+fDUceskn7wnT9dZsXInl/3kO7zz6L6xmPuSp34cTxg7GFRcw4M9fiG2rFuLfD82Bi8BDIxG8zZB4VN/QiHfeeRdz575HC85jnyyl/jz6wiuk9bJRrVBZ+erj+EX5MKQ5qtFFONjlKEOHoxgdWgdsz6HTcspxYeeW1fA11yFvwmiTdFVS+7sphM0bNqKdFjdu3DgjqAB9sY/5iurQQm/lFz305/wiPTZjY0JPQnwaThk3bizqG+vx6r1/xwW/vwfjjjkJG2eegQ2vvYSsjDy6yFIMHDSYFsuEc8M6ZFBIeiNRl896s4Ugp978vXLrVsRWriEDwwjTLWlUIX/AMJz7g8swaNI0A7kXPP8ENr33H2S5oxg2ZDC+O30qJlz+baLHYiu/UEzhT1H/fFpvNq3dWvmimddnnn8eC+YvwPIVK7Frzz6DDJ0OayrgUMhpmE7ZJtCAvidGVeVCzIOlLJR8wZ7NOzB31ykoHXU2v2uqt81cnxJpMUs425hUqpQuPWLGE3pPsCQq5zZ69GjjNtatX0OXcgCDBg008Dg1Pc2gr21MUjUOZbJu3mt0ge0I0wp8tCgFY40UlxSWYsXqlWZFZNSZjPNuvpPBu4UWsBi+YJcRypDBg9jmMF3ZFuzYtQsulqeY4CIg+NZPf4HSgZX44NXXsHvnJvgJX0uKi2npHrz5+By8/9Qjxjv4GZjnvfIC+sWXIVn8YavECJF22bes7FyivvreU2HWe8ONN6C7K0I3moTsrALjBRRH7TZrpOKLyDZl6qyY32+95VOkp3Nldom69aG8wMQQUsbAwfjObXci4NbbtTVsLtioBrHhzHyVDVSvWYy7Lz4Xk8ZNQAXBgfKTLsYSCbC5uQmbNm1EbV2NcR+CoVRsI8RxvL68bIDZF6WYDD2Gf//73xhMISqXWLViGZZt3Ijfv7WM2X0mcwlC8M5WvDX7H1jw3MPIS0lGSVExkgjDa5uaUTFuPKp2bSfEbcVExsTda1egnHGppKIUxfl5KOmXj7zcXNZbitLS/kgiIw8/8hjjWtev/NAslhNJJhr/U36hOLVpw1ZsYmL99ty5GFxZhnvv+aeuQjvBxIDBw+iG5TaViyn9pFs3xbBfBjB8Pn1CKPKjIglFt+u7JtlUnkgv+j/8Bz/CsBNPMahLGigwYL02VgvniL78nbjjnJMQa6nDjBknoqy0DJ0UilaaqE1iend3m4Gn+wgCtOYqP58a5bWW5Zh6WJ7mW1bTKlasWIHjTziOOcUovP3Gy2gOunDTC28j4vQwX9Az8SG46TZq16/Fu4/djw0fzoctHEQW4fCAgYMwZeIYo2j33PsQHnzgbpxxymlGkUwAN70UUcXifTzlzHPJ8E3YvG610fDm5hZcc8212F9Tgz3cuggoIkyOhRTl1s85Yxbu+df/mHtrDtRjCNuZl1dIRGg3bjedQCIpWdME9ByHIBS7kptEY0QSRsJKdNjs9znvjIXwwTNz4Cfml/o4zY/QEmExr6O3BTwpOOXyn6KNeH3JksXszH6acRaDdj94aRlMXwllUxmY+xn/XNq/AkkeBmMKVfMS2pzswIHq/Vi7ehX6MdgX8t6WpiZU7d2DignjhQLYthhad2zCs3f8An//4fcwd/ZfMWFgCf74+9vp09/FupUfYO6rz+E3t96CGUcdSRTUZQV4Y9msjv3xB0Oob6rHgfoGKpfV+aKiAmbaAiDijc24z7cXLEEwuwxjT78I5/zyj7hqzou44YEnUVRSzlLiDGPffUSR5eWlOO6Eo/CLX16Hp556mNDfTZd+6JNeFAqTps+5to88TNUamIsyGL/z4N1wBH1Go2WeZpLHXMw/tKYJM0/DqVf9Eo1Mot548xXMm/8O2lo7jTDSuQWYSK5asQpZ6Vm8V0G4B3t2b8WWrRuwZOk8bN2+AfPmvUO/7MKEseOodRFm4x/Az+Knnnq2ckQzLfDCn3+NG849HoteehQL3n4Wf/3LHbjk4gswcsRwuj+6WDLVRCi1j5hdC+50rKG+HjNPmoURI0cTuk7BT664mr2wAEZOdg41PGS8QILheljqsBNnobhcCwmbmPgC8599AQGfnjIjNGD54kUZc6A1tO777rkbl15yIaZNm4TXXnke5RUldG3NveV9HlkO81ApznThgMaNa7DmlRf5LYIQ/RvbZDpvTQ3bKDw3jqGbu/gvs5FcUIlVq1bh6acfwjNPP46XXnwOr736ElrM+qwohbAIGzetwZChFfjhZRfirj/cTkCw0rzBSDBh586d2LljJ3YzJ5kw6xwUjZjY27VITwhTp0xBdmb6wVlHbnKpn9Z9TXeLujq7sWb9JpQfeTJOvOZmNLUL5lNwvKmAVskv5joBHY1RDSwuQMfK91HQWYWxyQHcd9n3CJWXsnVRVO+v7R13M2N5jEMSkpVg2kx5551zNtoYTz+dqNhkm5lo4/blhEIyLGfDBXWXv/wc6lZ8CCehrWEA/2h5kPHVYgq3ETNm4ZdPvYZL//YoJp1zIXqY9bfTenJzMnHi8cfisksvwHPPPoEVyxabYHnhBedj5MiRSE9Ow7U/vRpLly7E0iXv4+c3/YwK6cWpl/6EFklXyc6rTuM2492Qxe/evRcPP/Iovv/9H+C66643x/uS4pTIDKyS4UfR6tIIClpbWuJrtYCiwgJTlvXNQahdgsW03kfu+xduu+lGfOucM9HV0kC3ETBPfa1YvRrfY321dIFr16zFk08+hV/cfAsef/zfxipNKQZCi/lS6oNs14ClhvLVD713LUzP9QmhSNGNsscp8d0UJktg0RKK9NFBf/zqA39B+95t/B4xQU+v17CFu6wZwT3U7qVLsOytV7B360YU5BThtNNOx6JFC7GVieLDjz6ASy66CGNHj4STCZ2GXIT+UlOtRxBG0QUVFuSZ49FoCE7mLd6MfrzWbUBGYoQ4rhKmjeeddz6uIkPmr91GJn1SMzVkLxLTo2TCC3f+HGnNe/DMEw/CoRSeZfm1RsEwU/FNI7ga34rQve4xaOv5Z58lyKCPCNtpRU4U5FXggyVrGeDH4KhjTsDlP7kW997/IFauXGnyNuUpcqVOWoF5R0sfriumSxiyKpF4+8XpZS9ZHddwSJj+1s+YoAHDdF8mnrnjVuQNGoS2+kZ0dTRAr6BJc9mQT5dSUVmBCWXF6Mqx4W9EKLf8/OcYM3KEEbQZHWazNWUb4V4Nkc2uXbuxmtoWDITMJJSGTMTsFpq+xqocmuplO8yib6M9ic0irZo8/rs/xnSiw6WP/yt+1CJzj2E2iPby8OJzT2Pa1KlEUXq41hqV/t2df8L9Dz9hxuYEgXXP6tVrcda556GDcFcrZUReojKtzjHPJzpsyCXaSmPOpWdIJQBfoAftdIkh8kkjGS6nCyFagRLbvpbyafQxobARbIjRF3Y4EtGMWpgFJpkG1jXsx+GHjcOw4cNM8peanoFLL70MSZ0N+OONl6NUGfXAAWYBgjpvWZTIho2EmH/+259N51WD5kzee38et/dNvNAMYSuz/SA5rqeA3XJPfZhIDwk3IbPe86WpAbnPj5KlaSJ7WhYCnlTW+tGL9HonL92N+qiAf9yMo+NnYlhHxv/4qp9h485dGHfsSWjauc0ooIBJj68HPd1+ZGXk8/5UoxxqVjDUwzqIFB1axGgzeZUS4O6uduY5XQjoBTkUkgYsBZ9lDWbB3+fLpK9QCP18HWhnNh6mr0xJcZssvK6ugQG51VR88skn4P7ZSpIspNFI/C4tUCNnnnB8XGHFcu1YWEaaI1dXUlxCc9fcd3xNFu978cWX8Op7C1A8ZDQGHX828soGoai8kginEm889GdTjkF3vL6JlmKGy/k9QZa4DgpfH0KAmt+wzvYRCi/Sf6XQNLOFGHXMKkFXNbW1YtueXTjn57/DpJPPwuyrLjIDiQnBap1YakqMnqDZvBQhJzsLkydPwLvz5vdaz9DBg3DU9KMwcNAAIroRGM3Y6NGL4qhg/3njdXPfoZDTeAARmaeR0kfn3IshQ4eggDmEOvDaa//BxT+43JjkCcfpgX7rBo0puYVkaI5ZGWkUJH0+A9Y+5hYLFi4yDwk1MPO94sqfYPSoMYwTqWZgUF00QtEemTxi0lEYfcLJOLBzO9FWFKXjJ/Cwg77aRVlKA3VHjK6jA166B8KTXiHoUwE70QWLJERTi9lPkBRELskMGLL+tWvW4d7Zs3Hvvf8yx44hM48n8DCPbRA5xlh3NGwpldp99pmnmzmYYcOHYtiwYcydNBxEeD59Oo4+8kj86le/ghZh6Bkc8UaxRMJXE15+5TUG/2fM4nLLcyXa9+nUa0jSQo/XjRnHHoPiEmuZjk5OnzbdWIKyfs1hJyhs/GOUPt9jlgFZfjKGhx6aQ0Fciwfpl+cv+hBbKRxVrynXfv36WSZstD2G/NxcJPsaEN6wECdW5ODAglew6tUXTd4SpZCTzUiu2MlYRlcgTbcE+rEu6YI+9LGvhqR8aWnpRjCijq5OPPvs88yf5pvvcpW/veWXePexhxBurbUYI+Yy0I8YORz33XcPrrj8hzj+2KPQv7gf8xQ7lZCxJCcDWdnpdHNJcDP4SyAmFrERtQdqjTe4/vrr2f9k5OZaE2efJxBRr1DUdXVXSEYoK3FCwWva9EkYNXogsrLSEKIZB+g3NeSuovVwjBZ6G0azJfr0JiWjoKg/MnPz6XYazXG2lZlyf7S0tdHFEL/z3tt/82u88sKz+Mcf78RF3/8eRo0djN1rVirhNwHRgoqKQnY0NDLQJycjTO5acxuEj10dRHrWqLLVFbo3VhQig50Rtc8CBVa0sdGVWG+rE+mYMyUN/5g9x/RZY2yDB1bgsnNm4W0mxqGeDgSVJ8k903W7KUzB1wQGFb+0l5GWycJYN/3mrt378BQF/bPrb8KEw6Zg5OjxuPiSH7HPnSjoV0yL1FxKL8s/kz5xRUITLW3mJwPT4YdPximnnEwtt+abBfFESWSSEqXc3JzegTtRlKisqbYKUX8nBmq20nTChry8POPmrBpI7LDyGbVC/zWiMCeXWX89me6Ar6vLLDdKkGCjLCfC/ODDV57EnBt/iAev+TbaqncaofdSxI9Nrz2Ph395LcYOLTcdNLVRmLL+BAm5OegBDhBRLV++lt+tK6+56grsXjIPNdvW9wbwxNbY1Ij35s/DPffdj8uvugZHzjgeb7zzLl545VWMoxBGjTsMF196OebMeQIHDjQRPfZDaelAlJcNNiPFh0qmleK/penkDz80/GweA+CP0+XAqSfPMqOmDpuTzAkZ36kO6sleDUUItiZgXjZh7Pnf+y6+deYZmHrkNB6h9vK6l15+GfMZFL/NzJa1UXIxLPrgA/zn9Tewo2o3tm7bh5q9+1E8bgLL0nx5hC7CYqKY2tnWjp3r1uHpm/bilBNn4vIbr8CEceNw+NRp5LdVt6771c03o7ioGOPH6V0rjA3sF/WI5yylsF53ogP6h6OMk9+9DHf9/e947ok5ZhzOneTBD37wHdx++x8wf8Fi7Nu3ly54K2PkDmwnSrRGCmK0HDctz2tGgxsb2wgEwnTPJUjiMYPwjIXz2rhyfxkyvTYmzUYq4Gm4wBUfipB2qNDKAZU8bWnLtm3bDCpLEA/BbVY0knjtT6/9qdpiOi4k1NzYiBuZzL34wossg0JlLDLEG598+lm89Nb7yCkegLzR0zDmjAqMPGKGcT8f78olF1+Eu4YMYVs0tK82s444s/vSmWeeZdwbRWEe/1abzGsK+Zltniyj25FV81NtKa4YiFf21eDSH19OAezH9q3bzBAM7QjXXH296YdctMftQV5OIVyMoRKGyx5/8YHhERNMKqyUzQw6ynIVxL4iUQ4WAwRXo/TTsgAxP8zvgoHmInZCx8SC+fMXcl93qOOUKq/XQzmmDUa4bBg/ZDkS8vnfOx8vvPAC0tNzzXMmQjOmueIqrXHM0cfj6oeewXd+8Qccee4PkV5uvVrQaHMfkvvU5JV5IYCxyoNuRQxPaKRpptrB802E8kuWL8ejjz+Bhx99lMiwGrfefjvOPPfbuODiSxFgsvjwbb+Cjdb/5uvzsXnzboaHJOTkF6O0ohLlFFh55RD0J1QvKCpBVk4OUpMzaCV0RXRtQmhWO6wRa8uLsm4zDC2N+WgfDpXMW4xMF9ibvpmmAvmWLVvMsUTnRatXrzafImmK5gp0jcUSYB1dzAGijgRdcsklZk6iX7881qP5l/gJEW+q372T/nsLzwVYjmKVSlLHqNFWlYYS9csyrPbYCeG7exWno70DixYvxoMPzsF1N9yEqUfNwOSpR+O0M87DdXf8Hku37UbAm44HHn4KK5ZvoNIROdH1RNqZgzFZ1YNFcntaxanVMZrplDtzmIUgfRpiSAoZ4KbhfQtNfnKLu8mvQM7ezpIXepI3QXZawIJFC81itgRpIfSa1etYmcUkxmaTxcqcLXHG8MBDDzH+BPDQ/bMNb7/1rXPw2ONP0x/vNkMNZs7CXC2LiqJ+5xa8+tsbkUxkVjlmMvqPGouKwcPgUFwjBNYgoTqpmctVq9dg48Yt2Lh5PTZs3GzcjZ8u6uTTzsTe3VVo11w8NTiLSKd06AhMmjrLaHogEkJbTQ32rV2OksL+bJbVBtN3KqX5SiEbsjAMr4m7cPNXdJA3pv96QJUkJflUIlj5SuQgX8YddpyaZDofCrVi/Xq9XB/wUQMv/P6FePapp3s1pa6uHmPGjEdd7X62xtLY6UcdjTPOOBPX/fQalYIriEqeevJ5LF08D4OH0BWx0evXb8DMmScjMysHs2adgL//9S5z7U+u+hkTq9dR2K+MnbTwfUgvVnankIk7cdGF52H7tq1maZBm/RSkXa4kpOUXot8gJnCVFSgbPBI71qxBlC6o/4BypOVkE85G0XigDo3796J57xZEfH7ozayC4hKIXoojd2tctTUJ1BsCLBf0XyQKpddfGQ3u474EVTdt3mw0VJoq0jKhAN1aU2urEYiMU25ESE1Di2K0dEfMu+e++6xjZMSYMSPNkHx7e7u5zzJtXq0q5X/JkcSLMPXKDqe/C2nMou/+172EoEtQ19BG+J2F7Jwi5BeWoF92DmzdQdRv3IElzz6LBrarbv16LH3uBbx9z/2YP+d+bH77ZTRuXA0/40qgp1OiQIiW3tRQw9iWQ7gbQ2dnK+skGNB7V9iWb1Ig4qnio94v4HTrnf3xMEHl6ZWCtDQxrC0yLqOrB8s/XB4/ArPiRNdpoC5BaWmpcDNmHCQ9upaMF55/wcy/S7iKDTfccB0Zm4Sa6hq8/sYbeGjOI1izhq5Q5s9fuyMMm0OLos1XeFwpqChjkC2qZOZfQqidg5TkVGq8Dd2MH531+9FVsweh+hqE2psQ87cj1NkEX0cDO+dHQ/0+dHU3IhjpJDS2Mf60oLmlFr/5zS+QmZmEwqJ8NDQwsTW1fTOk2KuJLCXFgutCgkoxtEmRRWZizOzFSRmrRfL3CvwOLFryAc1eKhRjxrqbwrKhtdV6BE03S5B5eTkmgFPfaBvWKKiXKOV//nEvVq1ai0ceeYL7d/PaFLzz3nu48MIfU0i3mNHhTOY1LqeGLKQt9NNxbbUgON2LrFd8I8JRPh1hB6B/VMBONDfVEVTQRTVWo6XpAJGdD2X989Ha0oj0VA9uveUmDB00EKNGDDVz5G6PAwWFhaipPYClS5aYZ+wNfVX//zkkpSYLjQCsV8hbMUhJcEIQukYzkwkL1fdwqPujQulLEooC7cKFS3oZtWWzxrGIegglew+S9B9/9u3bh5defAnr6EYU/ASBn37qJcw4ZhauuvpG3P/AY2hv60H/kgGoGDDYvGJqEONCelomBWq9w9LAcmqPfsJhHy2yE+0dTQiGurjfgXoyv5hM7wm0o+bAblxx5WW4/sarceedt+P0009Fbl4uRo4cDl+3ljNZ7lVkUJAEyybf/c9/YuYJM3HvPf9CKBgwyvS/JTG/L3LVd63UFHoT88WfMPuWGAn5LIpEguRBVx+hkCkf96mqSEFar/MLUjv3Vu2XHaCtQ88HtmDZ8pVmCvSqa36GceMn46KLf4T1G7bSfckN2lBUWIaikgri/cHE+pWEnKXMopPgortzONxsgPKhCAIhv3mfSktrMyFuF/1+La0HmHTYOJx6yok4ULMfrc31+N3vfovMjFS6njx8+ztnY+kHi3HrrbfiyiuvxPEnHE83ac3B6IUE4VifRxB4PCTGcFuzag0euO8B8wS0z99NRmgqQYy0gMtnkYCAtgRZlzIuuBkDCE7M/0KJT/JQ4Y1FqG+iRLn6/Hg5CZJihgJtlvXEj8Ursa5WVyQQPTIQ8AfwxhtvMWlcQB/cYF7noYmtwcNGYcZxJ2HVyg3M6FORn98fZaWDMLByOHKV+dJczf8NIWTWBI/KM0McZJS0Rw0WQzq7WhGkIM467SRcctH5yMpI5nEfbvvNLRTwajz55ONkqJ8JarZpj2DwmtVrMWXKFLqgZczS88zCu/Z2rSnTrF+qmREtKSllnrXVxLSxY0fjexd8G0cedQT+eNeduPGGGyjgO80QkmB6ot+fRSZfic+lJ+KC3JE2nZMUehXgK5AEEQy08jMuxAQk1v+9ysj0MvtmTrF1G7Zv34lbb7sNYUJlrZSX+cnHl5UN4He/KUAmqpffaOxLbw2S5khLPs9MjbboIl4ejvjZmG78+te3mPkIteGBB+43Qr/9jttxz71EcOxr9f5q3YnDJ08w0wfp6WkmBv3+zj8QjHSyTWU466yzcdddf8AcAgg9EiHG3Xrrr03MGjFiOPMbIrS6OlOeFjE4nB4zwpDkPQhuPo00epKIB1YM6ONc4v1MBOmvSsFAB72G1jtbdDBPiYZRW1sFn6/HaLFeBK0gW1RcYmk5j+nxBcUZox0GQypIUUvYKC3PSZipYfpnUF+hKF5ceMF38cZ/XsP2nTvM27Nvu+1WPHD/A7hn9r144vEnseSDZdi9a4+ZDgjRovTYxI3XX4+HH3sE199wPQN9ncnq//SnPxuXwRpouR7zzIuDgsvKyDSPSAjtSNN1v6oXqFG/zH+i/hjFu2Guk0vSvUHWIUVQcmksnid7+/K/oHCoh57AAk4Jso0Zf2TMTv+ux5RD1NTOeAc8Ljcb4SREdRmmSwZmPRN9tqxGg346qDmLRMMOpZG9/8mBQm9va8RPr70S//rn3fDT5ej9YH/961/w8xtvonBu47nrTAzSm+40c/nLX95gAMV9s+9HaxtzDEmW1YXDUXjcXhRTgeSJurq7zKoYQWiNUljZ+UEXZSmF9d3SLSHNuELxQ2DDZPoknZOQzLH/YwqH/RRIW1z6B8k2bPjImMubbRJA06x4EiNrkHmaRWVxoeg4dcS6MV5OX0H03f8s6isUX087RhMtHX3MdDz9zDM48aRZjAObMffNt3DBBRfggQfnYGDFcBO429ub0djMnISAQy6zsLDYKISQjaC0x0sFMozVOJx+WIWpUYOFZseQ2qd2SiiyFDkjxQMx/bP615e+qH+HSkJaQb8euftkebahw4bHlJUnp+dTHk4zxWvcE8myCGbrPPZ/1ZiPkh6q0cOgg1E5qJJgYj72V1UbcOCn1ehRAkFoYxFiNJsg16FpWEFzI4A4zxJM7Ps90eaEFRil4b6O6y2n/y2KRgiTAy1sx6e3wQhFO3oWT4KRQzLaxGMSkHmU+BAs4KuSGtbSWm+W5qgKvRxHCWlCORLMT5CluWqLxeiPa7axAFq7FognEjRRbx/iAkq4p2+a9AoUWUgCaX0a2YYOtYRiiBbjdmcaZPJNkvhsMc9iWF+XoXVn1o5lvYlzCe2PMBbqTr2EVEdcdGWycK3/lVD6uq7/NkXCAbpbxsK4onwWHVRBEbVWZhUOHRzb+mZIwTQ+INdHIJ9Gsh7Nd+spY6EprRPQE8F97zWjA0zG4jL+/4LEU/H2iwQCAP8P2ZNyt+8FWt4AAAAASUVORK5CYII="
    }
   },
   "cell_type": "markdown",
   "id": "67ca79fb",
   "metadata": {},
   "source": [
    "## AUTOR\n",
    "\n",
    "## **<p style='text-align: center;'>LUIS EDGARD JUNIOR FLORES BEJARANO</p>**\n",
    "![image-2.png](attachment:image-2.png)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7333c45f",
   "metadata": {},
   "source": [
    "## **ABSTRACT**\n",
    "\n",
    "Currently, poly(3-hydroxybutyrate) (PHB) is an alternative to replace conventional plastics. Bacteria of the genus Bacillus were the first microorganisms in which the accumulation of PHB was discovered. These microorganisms lack lipopeptides in their cell membrane, making their extraction easier, they are genetically very stable and have higher growth rates compared to other bacteria. Due to these and other qualities, researchers have been evaluating its use to produce PHAs.\n",
    "\n",
    "The objective of this work was to determine the batch and feed bacth simulation in jupyter notebook, to obtain the kinetic parameters for the batch and feef bacth simulation and their corresponding graphs in a Jupyer Notebook, using sources of carbon, nitrogen and other nutrients. . This work presents Bacillus subtilis iYO844 as a candidate for PHB production.\n",
    "\n",
    "**Keywords**: Bacillus subtilis iYO844, batch culture, feed bacth culture, PHB\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3147634d",
   "metadata": {},
   "source": [
    "## **RESUMEN**\t\n",
    "\n",
    "En la actualidad el poli(3-hidroxibutirato) (PHB) es una alternativa para el reemplazo de los plásticos convencionales. Las bacterias del género Bacillus fueron los primeros microorganismos en los que se descubrió la acumulación de PHB. Estos microorganismos carecen de lipopéptidos en su membrana celular, haciendo más fácil su extracción, son muy estables genéticamente y poseen tasas de crecimiento más altas en comparación con otras bacterias. Debido a estas y otras cualidades, los investigadores vienen evaluando su utilización para producir PHAs.\n",
    "\n",
    "El objetivo del presente trabajo fue determinar la simulación en batch y feed bacth en jupyter notebook, obtener los parámetros cinéticos para la simulación en batch y feef bacth y sus gráficas correspondientes en un cuaderno de Jupyer Notebook, utilizando fuentes de carbono, nitrógeno y otros nutrientes. Este trabajo presenta a Bacillus subtilis iYO844 como un candidato en la producción de PHB.\n",
    "\n",
    "**Palabras clave**: Bacillus subtilis iYO844, cultivo batch, cultivo feed bacth, PHB\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a8cf458",
   "metadata": {},
   "source": [
    "## **INTRODUCCIÓN**\n",
    "\n",
    "En la actualidad, la mayoría de los plásticos se producen a partir de combustibles fósiles, los cuales no son renovables. Además, los plásticos de este origen no son biodegradables, por lo que se acumulan en el medio ambiente, permaneciendo inalterables en la superficie terrestre por años. Los plásticos biodegradables o bioplásticos aparecen como una alternativa a la utilización de los plásticos convencionales. (Almeida, 2010)\n",
    "Entre los materiales que se han desarrollado hasta la actualidad, los polihidroxialcanoatos (PHA) se presentan como uno de los polímeros verdes del futuro, que se espera reemplacen gradualmente a los plásticos tradicionales (Kourmentza et al., 2017, como se citó en Honorio, 2019). Los polihidroxialcanoatos fueron descubiertos en la década de 1920 por Lemoigne, quien aisló y caracterizó las inclusiones de poli(3-hidroxibutirato) en Bacillus megaterium (Lemoigne, 1926, como se citó en Almeida, 2010).\n",
    "Actualmente, existen varias compañías que se dedican a la producción de PHA a escala industrial (Kourmentza et al., 2017, como se citó en Honorio, 2019). Sin embargo, debido a su elevado costo, la aplicación de los PHAs en mercados de demanda masiva, como la industria del empaquetado y embalaje, es muy limitada (Hartley Yee & Ray Foster, 2014, como se citó en Honorio, 2019).\n",
    "Los costos de producción elevados de los PHAs están asociados principalmente al uso de materias primas de alta pureza como fuente de carbono y otros nutrientes, contribuyendo hasta en un 50% de los costos del proceso (Koller & Braunegg, 2018, como se citó en Honorio, 2019). En este aspecto, el empleo de materias primas residuales, baratas y abundantes, es una alternativa potencial que puede ayudar a reducir ampliamente el costo de estos biopolímeros, y así, aumentar su competitividad frente a los plásticos tradicionales (G. Kaur, 2015, como se citó en Honorio, 2019).\n",
    "Las bacterias del género Bacillus fueron los primeros microorganismos en los que se descubrió la acumulación de polihidroxibutirato (PHB), el PHA más común y el más estudiado. Además de ser muy estables genéticamente, y tener tasas de crecimiento más altas que otras bacterias, las especies de Bacillus son ventajosas al momento de producir PHAs porque carecen de lipopéptidos en su membrana celular, lo que facilita los procesos de extracción y purificación de estos biopolímeros (Mohapatra, Maity, et al., 2017, como se citó en Honorio, 2019). Adicionalmente, estos microorganismos pueden producir copolímeros de PHAs a partir de sustratos simples, baratos y diversos en estructura (Valappil, Boccaccini, Bucke, & Roy, 2007, como se citó en Honorio, 2019). Estas y otras cualidades han sido reconocidas por muchos investigadores, que ya vienen evaluando el uso de diversas cepas de Bacillus como productoras de PHAs a partir de numerosos residuos agroindustriales (Krueger & Radetski, 2012; Ramadas, Singh, Soccol, & Pandey, 2009; Santimano, Prabhu, & Garg, 2009; Halami, 2008, como se citó en Honorio, 2019). Gracias a su amplia capacidad para metabolizar diversos azúcares y producir múltiples enzimas extracelulares, se deberían explorar como candidatos para la producción de PHA (Honorio, 2019).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9859e5ca",
   "metadata": {},
   "source": [
    "## **Objetivo General**\n",
    "\n",
    "Determinar la simulación en batch y feef bacth en jupyter notebook\n",
    "\n",
    "## **Objetivos Específicos**\n",
    "\n",
    "•\tObtención de los parámetros cinéticos para la simulación en batch y feef bacth\n",
    "\n",
    "•\tProducir PHB mediante el cultivo batch y Feed Batch por Bacillus subtilis iYO844\n",
    "\n",
    "•\tInterpretación de los resultados en simulación batch y feef bacth \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4511bb28",
   "metadata": {},
   "source": [
    "## **I.\tCapítulo 1: Los biopolímeros**\t"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2734516",
   "metadata": {},
   "source": [
    "**El polihidroxibutirato (PHB)**\n",
    "\n",
    "Es un biopoliéster termoplástico producido por una gran variedad de bacterias, este biopolímero es acumulado como reserva de carbono y energía en condiciones limitantes de algún nutriente esencial para el crecimiento celular y en exceso de una fuente de carbono. En otras palabras, el PHB es para las bacterias lo que el almidón es para las plantas y la grasa para los animales. El homopolímero contiene como monómero el ácido 3-hidroxibutírico, el biopolímero es producido en forma de gránulos insolubles en el citoplasma celular, los cuales pueden ser observados claramente por el uso de microscopia electrónica de transmisión (López, 2011).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "727b4c22",
   "metadata": {},
   "source": [
    "## **II.\tCapítulo 2: Producción de PHB a partir de Bacillus subtilis**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc9dea83",
   "metadata": {},
   "source": [
    "**2.1\tMateriales y Métodos**\t"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e527491",
   "metadata": {},
   "source": [
    "**2.1.1\tCepas Bacterianas**\n",
    "\n",
    "Existen más de 250 microorganismos productores naturales de PHB, pero las principales limitaciones de las bacterias respecto a la producción natural de PHB a escala industrial son: el complicado deterioro de las células para liberación de moléculas (lisis) para la extracción final del polímero, una baja densidad celular y un excesivo tiempo en alcanzar la máxima densidad. Además, para abaratar los costos de producción surge la necesidad de utilizar como sustratos o materia prima simple y/o barata, reutilizar los residuos de otras industrias.\n",
    "\n",
    "Por otra parte, las bacterias Gram positivas como el género Bacillus no contienen LPS en su membrana, tienen un rápido crecimiento en sustratos económicos, toleran altas temperaturas, tienen altas presiones osmóticas y han sido utilizados para la producción industrial de PHB.\n",
    "\n",
    "**2.1.1.1 Bacillus subtilis iYO844**\n",
    "\n",
    "Las bacterias del género Bacillus son conocidas acumuladoras de PHA (G.-Q. Chen, König, & Lafferty, 1991, como se citó en Honorio, 2019). De hecho, el primer PHA en ser descubierto y el más estudiado, el poli-3-hidroxibutirato o PHB, fue reportado en la bacteria Bacillus megaterium, en 1926. En algunos casos, se ha reportado que las bacterias de este género pueden llegar a acumular más del 80% de su peso en PHA (Mohapatra, Maity, et al., 2017, como se citó en Honorio, 2019).\n",
    "\n",
    "Como productoras de PHA, las bacterias del género Bacillus ofrecen muchas otras ventajas. Al ser bacterias Gram-positivas, estas bacterias no poseen lipopolisacáridos, que son indeseables en aplicaciones biomédicas, lo que facilita el procesamiento del PHA después de la fermentación (Mohapatra, Maity, et al., 2017, como se citó en Honorio, 2019).\n",
    "\n",
    "Las cepas de B. subtilis utilizadas en prácticamente todas las investigaciones académicas y muchos procesos industriales se derivan de un solo auxótrofo que requiere triptófano, la cepa 168. A pesar de su importancia central para la investigación, nuestro conocimiento de la herencia genómica de esta cepa es incompleto. La cepa 168 se aisló después de que dos botánicos de la Universidad de Yale, Paul Burkholder y Norman Giles, mutagenizaran B. subtilis Marburg con rayos X (Burkholder y Giles, 1947, como se citó en Zeigler et al., 2008).\n",
    "\n",
    "La cepa 168 se difundió posteriormente por todo el mundo. Los investigadores pronto desarrollaron métodos genéticos clásicos, y más tarde, tecnologías genómicas y recombinantes, para dilucidar la fisiología y el desarrollo de esporas de esta cepa Fuente especificada no válida.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2415b87",
   "metadata": {},
   "source": [
    "**BiGG Models**\n",
    "\n",
    "Se descargo el modelo de BiGG Models “Bacillus subtilis iYO844” y se leyó en Jupyter Notebook\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00f501e7",
   "metadata": {},
   "source": [
    "**2.1.2\tMedio de cultivo de la Bacillus subtilis**\n",
    "\n",
    "La cepa se cultivó en medio de cultivo caldo nutritivo (Atlas, 1997) que contenía (g L-1) peptona, 2.5; NaCl, 2,5; extracto de levadura 1,0; extracto de carne de vacuno 0,5. Los cultivos (100 ml en copos de erlenmayer de 250 ml) se inocularon con un inóculo al 2% (v/v) y se incubaron a 37°-30°C con agitación orbital vigorosa a 225-250 rpm.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc4f571a",
   "metadata": {},
   "source": [
    "## **2.1.3\tCultivo batch**\n",
    "\n",
    "En la fermentación batch todos los sustratos - excepto el oxígeno en procesos aeróbicos - se encuentran disponibles en concentraciones suficientemente altas para que permitan el crecimiento celular no restringido en sustratos. La biomasa en un proceso batch crece exponencialmente al máximo permisible dependiendo del microorganismo, las condiciones físicas y ambientales. Dicho crecimiento exponencial ocurre a una tasa específica de crecimiento máxima (µmáx, función del medio de cultivo, temperatura y pH), hasta que algún factor, como concentración inhibitoria de un compuesto secundario, baja concentración del sustrato limitante, o el cambio metabólico a una fuente alternativa de sustrato (en medios complejos) reducen la tasa específica de crecimiento (Enfors, 2011, como se citó en Cárcamo, 2013). El sistema batch corresponde a un sistema cerrado, a excepción de sustratos gaseosos (e.g. oxígeno o gases inertes para desorber CO2), agentes controladores de pH y a veces antiespumante. La fermentación batch es iniciada por la adición de un inóculo celular al medio de cultivo. El inóculo es un cultivo celular con el que se inicia la proliferación celular. (Cárcamo, 2013)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "790d152b",
   "metadata": {},
   "source": [
    "**Ecuaciones:**\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19591142",
   "metadata": {},
   "source": [
    "### funcion especifica\n",
    "$$\n",
    "\\begin{align}\n",
    "    \\mu &= \\mu_{max} \\cdot \\frac{c_S}{k_S + c_S}\\\\\n",
    "    q_S &= -\\frac{1}{Y_{X/S}} \\cdot \\mu \\\\\n",
    "    q_P &= Y_{P/X} \\cdot \\mu\n",
    "\\end{align}\n",
    "$$\n",
    "\n",
    "### derivando \n",
    "$$\n",
    "\\begin{align}\n",
    "    \\frac{dc_X}{dt} &= \\mu \\cdot c_X & c_X \\left(t_0\\right) = c_{X,0}\\\\\n",
    "    \\frac{dc_S}{dt} &= q_S \\cdot c_X & c_S \\left(t_0\\right) = c_{S,0}\\\\\n",
    "    \\frac{dc_P}{dt} &= q_P \\cdot c_X & c_P \\left(t_0\\right) = c_{P,0}\\\\\n",
    "\\end{align}\n",
    "$$\n",
    "\n",
    "### modelo de parametros\n",
    "$$\n",
    "\\begin{align}\n",
    "    \\mathbf{\\theta_x} &= \\left[k_S \\; \\mu_{max} \\; Y_{P/X} \\; Y_{X/S}\\right]^T\\\\\n",
    "    \\mathbf{x}_0 &= \\left[c_{P,0} \\; c_{S,0} \\; c_{X,0}\\right]^T\n",
    "\\end{align}\n",
    "$$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80dee8c7",
   "metadata": {},
   "source": [
    "## **2.1.4        Cultivo feed batch** \n",
    "\n",
    "En los sistemas de cultivo fed batch una solución concentrada del sustrato limitante es alimentada al sistema con el fin de aumentar el crecimiento celular más allá del permisible en la fase batch (S-O Enfors 2011, como se citó en Cárcamo, 2013).\n",
    "El medio de alimentación es comúnmente una solución concentrada de la fuente de carbono/energía (en vez del medio completo utilizado para cultivos continuos), comúnmente la concentración máxima posible, de modo de minimizar el aumento del volumen del medio de cultivo. A su vez se suele complementar con elementos trazas y vitaminas (Cárcamo, 2013).\n",
    "La fermentación feed batch en general es la estrategia de elección con fines productivos. Sin embargo, dicha técnica posee una serie de desventajas, como la inhibición por sustrato, transferencia de oxígeno limitada, formación de compuestos secundarios inhibitorios y limitaciones de disipación de calor (Wang & Lee, 1998, como se citó en Cárcamo, 2013).\n",
    "\n",
    "La acción de control sigue una estrategia fija, que generalmente consiste en controlar la tasa específica de crecimiento a un valor lo más grande posible evitando el metabolismo de sobreflujo (ocasionado por una fermentación mixta). En la fermentación mixta, posiblemente por una limitación río abajo de la glicólisis, la fuente carbonada no es completamente oxidada produciendo compuestos secundarios, que a su vez son inhibidores del crecimiento celular. Al restringir su formación es posible prolongar el crecimiento celular y aumentar los parámetros productivos. En caso de existir restricciones físicas de transferencias de oxígeno, el control debe considerar dicha restricción adicional mediante una estrategia de control diferente (Cárcamo, 2013).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4b94019",
   "metadata": {},
   "source": [
    "**Ecuaciones:**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00a5047d",
   "metadata": {},
   "source": [
    "### Ecuaciones del Modelo\n",
    "\n",
    "### funcion especifica\n",
    "$$\n",
    "\\begin{align}\n",
    "    \\mu &= \\mu_{max} \\cdot \\frac{c_{S}}{k_{S} + c_{S}}\\\\\n",
    "    q_{S} &= -\\frac{1}{Y_{X/S}}\\cdot \\mu\\\\\n",
    "    q_P &= Y_{P/X} \\cdot \\mu \\\\\n",
    "\\end{align} \n",
    "$$\n",
    "\n",
    "\n",
    "#### perfil de alimentacion\n",
    "$$\n",
    "\\begin{align}\n",
    "    F &=\n",
    "    \\begin{cases}\n",
    "        0 & t < t_F \\wedge V_L > V_{L,max}\\\\\n",
    "        \\frac{c_{S,0} \\cdot V_{L,0} \\cdot \\mu_{Set}}{c_{S,F} - c_S} \\cdot e^{\\mu_{Set} \\left( t - t_F\\right)} & t \\geq t_F \\\\\n",
    "    \\end{cases}\\\\\n",
    "\\end{align}\n",
    "$$\n",
    "\n",
    "#### derivando\n",
    "$$\n",
    "\\begin{align}\n",
    "    \\frac{dc_X}{dt} &= \\mu \\cdot c_X - \\frac{F}{V_L} \\cdot c_x, & c_X \\left(t_0\\right) = c_{X,0}\\\\\n",
    "    \\frac{dc_{S}}{dt} &= q_{S} \\cdot c_X + \\frac{F}{V_L}\\left(c_{S,F} - c_S \\right), & c_{S} \\left(t_0\\right) = c_{S,0}\\\\\n",
    "    \\frac{dc_P}{dt} &= q_P \\cdot c_X - \\frac{F}{V_L} \\cdot c_x, & c_P \\left(t_0\\right) = c_{P,0}\\\\\n",
    "    \\frac{V_L}{dt} &= F, & V_L \\left(t_0\\right) = V_{L,0}\\\\\n",
    "\\end{align}\n",
    "$$\n",
    "\n",
    "\n",
    "### Parametrizacion\n",
    "$$\n",
    "\\begin{align}\n",
    "    \\mathbf{\\theta} &= \\left[c_{S,F} \\; k_{S} \\; m_S \\; \\mu_{max}\\; \\mu_{Set} \\; t_F \\; Y_{P/X} \\; Y_{X/S} \\; V_{L,max} \\; c_{P,0} \\; c_{S,0} \\; c_{X,0} \\; V_{L,0}\\right]^T\\\\\n",
    "\\end{align}\n",
    "$$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "96a06038",
   "metadata": {},
   "source": [
    "Los rendimientos en biomasa y producto son parámetros de gran importancia pues dan una medida de la eficacia de conversión del sustrato en biomasa y productos. Un proceso destinado a la producción de biomasa tendrá como objetivo maximizar Yx/s y minimizar Yp/s.\n",
    "\n",
    "Los parámetros cinéticos determinados mediante simulación y estimación de parámetros fueron: \n",
    "\n",
    "μmáx=velocidad máxima de crecimiento (s-1)\n",
    "\n",
    "S= Sustrato (g/L)\n",
    "\n",
    "Ks=Constante de Monod (g/L)\n",
    "\n",
    "μ=constante de crecimiento celular (s-1)\n",
    "\n",
    "X=concentración celular (Kg/m3)\n",
    "\n",
    "Yx/s= Rendimiento de biomasa durante el crecimiento (g/g)\n",
    "\n",
    "Yx/p= Rendimiento de producto durante el crecimiento (g/g)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57e1c540",
   "metadata": {},
   "source": [
    "**Procedimiento**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d9b978c0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "from cobra.io import read_sbml_model, write_sbml_model # Para leer archivo sbml\n",
    "model = read_sbml_model('datos/iYO844.xml')  \n",
    "write_sbml_model(model, 'datos/iYO844.xml')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fa42fd3b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Número de genes: 844\n",
      "Número de reacciones: 1250\n",
      "Número de metabolitos: 990\n"
     ]
    }
   ],
   "source": [
    "g = len(model.genes)\n",
    "r = len(model.reactions)\n",
    "m = len(model.metabolites)\n",
    "\n",
    "print('Número de genes:', g)\n",
    "print('Número de reacciones:', r)\n",
    "print('Número de metabolitos:', m)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21f2fc63",
   "metadata": {},
   "source": [
    "## FBA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "83dd12d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "solution = model.optimize()  # Para optimizar modelo "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d872b17a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX_etha_e 0.0 Ethanolamine exchange\n",
      "EX_drib_e 0.0 Deoxy D Ribose exchange\n",
      "EX_csn_e 0.0 Cytosine exchange\n",
      "EX_5mtr_e 0.0 5-Methylthio-D-ribose exchange\n",
      "EX_dtmp_e 0.0 DTMP exchange\n",
      "EX_6pgc_e 0.0 6 Phospho D gluconate exchange\n",
      "EX_ctbt_e 0.0 Crotonobetaine exchange\n",
      "EX_cu2_e 0.0 Cu2+ exchange\n",
      "EX_Larab_e 0.0 Alpha L Arabinan exchange\n",
      "EX_ectoine_e 0.0 Ectoine exchange\n",
      "EX_cys__D_e 0.0 D-Cysteine exchange\n",
      "EX_cys__L_e 0.0 L-Cysteine exchange\n",
      "EX_eths_e 0.0 Ethanesulfonate exchange\n",
      "EX_cyst__L_e 0.0 L Cystine exchange\n",
      "EX_Lcyst_e 0.0 L-Cysteate exchange\n",
      "EX_cytd_e 0.0 Cytidine exchange\n",
      "EX_abt__L_e 0.0 L Arabinitol exchange\n",
      "EX_etoh_e 0.0 Ethanol exchange\n",
      "EX_dad_2_e 0.0 Deoxyadenosine exchange\n",
      "EX_dcyt_e 0.0 Deoxycytidine exchange\n",
      "EX_fe2_e 0.0 Fe2+ exchange\n",
      "EX_dextrin_e 0.0 Dextrin exchange\n",
      "EX_dha_e 0.0 Dihydroxyacetone exchange\n",
      "EX_ac_e 0.0 Acetate exchange\n",
      "EX_acac_e 0.0 Acetoacetate exchange\n",
      "EX_fe3_e -0.00040698404316227453 Fe3+ exchange\n",
      "EX_ferrich_e 0.0 Ferrichrome exchange\n",
      "EX_diact_e 0.0 Diacetyl exchange\n",
      "EX_djenk_e 0.0 Djenkolate exchange\n",
      "EX_ferxa_e 0.0 Ferroxamine exchange\n",
      "EX_fol_e 0.0 EX fol LPAREN e RPAREN \n",
      "EX_acgam_e 0.0 N-Acetyl-D-glucosamine exchange\n",
      "EX_acmana_e 0.0 N-Acetyl-D-mannosamine exchange\n",
      "EX_acnam_e 0.0 N-Acetylneuraminate exchange\n",
      "EX_for_e 0.0 Formate exchange\n",
      "EX_fru_e 0.0 D-Fructose exchange\n",
      "EX_actn__R_e 0.0 EX actn R LPAREN e RPAREN \n",
      "EX_ade_e 0.0 Adenine exchange\n",
      "EX_adn_e 0.0 Adenosine exchange\n",
      "EX_akg_e 0.0 2-Oxoglutarate exchange\n",
      "EX_ala_B_e 0.0 Beta-Alanine exchange\n",
      "EX_ala__D_e 0.0 D-Alanine exchange\n",
      "EX_ala_L_Thr__L_e 0.0 Ala L Thr L exchange\n",
      "EX_ala_L_asp__L_e 0.0 Ala L asp L exchange\n",
      "EX_ala__L_e 0.0 L-Alanine exchange\n",
      "EX_ala_L_gln__L_e 0.0 Ala Gln exchange\n",
      "EX_ala_L_glu__L_e 0.0 Ala L glu L exchange\n",
      "EX_L_alagly_e 0.0 Ala L gly exchange\n",
      "EX_ala_L_his__L_e 0.0 Ala His exchange\n",
      "EX_ala_L_leu__L_e 0.0 Ala Leu exchange\n",
      "EX_g6p_e 0.0 D-Glucose 6-phosphate exchange\n",
      "EX_gal_e 0.0 D-Galactose exchange\n",
      "EX_galctr__D_e 0.0 D Galactarate exchange\n",
      "EX_galt_e 0.0 Galactitol exchange\n",
      "EX_galur_e 0.0 D-Galacturonate exchange\n",
      "EX_gam6p_e 0.0 D-Glucosamine 6-phosphate exchange\n",
      "EX_gam_e 0.0 D-Glucosamine exchange\n",
      "EX_gbbtn_e 0.0 Gamma-butyrobetaine exchange\n",
      "EX_glc__D_e -1.7 D-Glucose exchange\n",
      "EX_glcn__D_e 0.0 D Gluconate exchange\n",
      "EX_glcr_e 0.0 D-Glucarate exchange\n",
      "EX_glcur_e 0.0 D-Glucuronate exchange\n",
      "EX_gln__L_e 0.0 L-Glutamine exchange\n",
      "EX_glu__D_e 0.0 D Glutamate exchange\n",
      "EX_alaala_e 0.0 D-Alanyl-D-alanine exchange\n",
      "EX_alltn_e 0.0 Allantoin exchange\n",
      "EX_amp_e 0.0 AMP exchange\n",
      "EX_amylase_e 0.0 Mature Bacillus subtilis amylase exchange\n",
      "EX_fum_e 0.0 Fumarate exchange\n",
      "EX_antim_e 0.0 Antimonite exchange\n",
      "EX_g1p_e 0.0 D-Glucose 1-phosphate exchange\n",
      "EX_glu__L_e 0.0 L-Glutamate exchange\n",
      "EX_arab__D_e 0.0 D Arabinose exchange\n",
      "EX_arab__L_e 0.0 L-Arabinose exchange\n",
      "EX_arbt_e 0.0 Arbutin exchange\n",
      "EX_arg__L_e 0.0 L-Arginine exchange\n",
      "EX_argp_e 0.0 L Arginine phosphate exchange\n",
      "EX_arsenb_e 0.0 ARSENOBETAINE exchange\n",
      "EX_arsna_e 0.0 Arsenate exchange\n",
      "EX_arsni2_e 0.0 Arsenite exchange\n",
      "EX_asn__L_e 0.0 L-Asparagine exchange\n",
      "EX_asp__L_e 0.0 L-Aspartate exchange\n",
      "EX_bilea_e 0.0 Bile acid exchange\n",
      "EX_btd_RR_e 0.0  R R  2 3 Butanediol exchange\n",
      "EX_glx_e 0.0 Glyoxylate exchange\n",
      "EX_buts_e 0.0 Butanesulfonate exchange\n",
      "EX_ca2_e -0.0003780822777782869 Calcium exchange\n",
      "EX_cbl2_e 0.0 Cob II alamin exchange\n",
      "EX_pur_e 0.0 Puromycin exchange\n",
      "EX_pyr_e 0.0 Pyruvate exchange\n",
      "EX_cd2_e 0.0 Cadmium exchange\n",
      "EX_gly_asn__L_e 0.0 Gly asn L exchange\n",
      "EX_cellb_e 0.0 Cellobiose exchange\n",
      "EX_cgly_e 0.0 Cys-Gly exchange\n",
      "EX_raffin_e 0.0 Raffinose exchange\n",
      "EX_rib__D_e 0.0 D-Ribose exchange\n",
      "EX_gly_asp__L_e 0.0 Gly asp L exchange\n",
      "EX_ribflv_e 0.0 Riboflavin exchange\n",
      "EX_chitob_e 0.0 Chitobiose exchange\n",
      "EX_gly_e 0.0 Glycine exchange\n",
      "EX_chol_e 0.0 Choline exchange\n",
      "EX_rmn_e 0.0 L-Rhamnose exchange\n",
      "EX_gly_gln__L_e 0.0 Gly Gln exchange\n",
      "EX_chols_e 0.0 EX chols LPAREN e RPAREN \n",
      "EX_chor_e 0.0 Chorismate exchange\n",
      "EX_cit_e 0.0 Citrate exchange\n",
      "EX_salcn_e 0.0 Salicin exchange\n",
      "EX_gly_glu__L_e 0.0 Gly glu L exchange\n",
      "EX_sbt__D_e 0.0 D-Sorbitol exchange\n",
      "EX_ser__D_e 0.0 D-Serine exchange\n",
      "EX_citr__L_e 0.0 L-Citrulline exchange\n",
      "EX_cmp_e 0.0 CMP exchange\n",
      "EX_ser__L_e 0.0 L-Serine exchange\n",
      "EX_gly_met__L_e 0.0 Gly Met exchange\n",
      "EX_so4_e -0.02009156356383357 Sulfate exchange\n",
      "EX_spmd_e 0.0 Spermidine exchange\n",
      "EX_co2_e 5.980675626954003 CO2 exchange\n",
      "EX_cobalt2_e 0.0 Co2+ exchange\n",
      "EX_crn_e 0.0 L-Carnitine exchange\n",
      "EX_cro4_e 0.0 Chromate exchange\n",
      "EX_srb__L_e 0.0 L Sorbose exchange\n",
      "EX_starch_e 0.0 Starch exchange\n",
      "EX_subtilisin_e 0.0 Mature Bacillus subtilis subtilisin exchange\n",
      "EX_succ_e 0.0 Succinate exchange\n",
      "EX_sucr_e 0.0 Sucrose exchange\n",
      "EX_gly_pro__L_e 0.0 Gly pro L exchange\n",
      "EX_sula_e 0.0 Sulfoacetate exchange\n",
      "EX_taur_e 0.0 Taurine exchange\n",
      "EX_glyb_e 0.0 Glycine betaine exchange\n",
      "EX_thiog_e 0.0 Thioglycolate exchange\n",
      "EX_glyc3p_e 0.0 Glycerol 3-phosphate exchange\n",
      "EX_glyc_e 0.0 Glycerol exchange\n",
      "EX_glyclt_e 0.0 Glycolate exchange\n",
      "EX_glycogen_e 0.0 Glycogen exchange\n",
      "EX_gmp_e 0.0 GMP exchange\n",
      "EX_gsn_e 0.0 Guanosine exchange\n",
      "EX_thr__L_e 0.0 L-Threonine exchange\n",
      "EX_gthox_e 0.0 Oxidized glutathione exchange\n",
      "EX_thym_e 0.0 Thymine exchange\n",
      "EX_gthrd_e 0.0 Reduced glutathione exchange\n",
      "EX_thymd_e 0.0 Thymidine exchange\n",
      "EX_gua_e 0.0 Guanine exchange\n",
      "EX_tmp_e 0.0 Trimetaphosphate exchange\n",
      "EX_tre_e 0.0 Trehalose exchange\n",
      "EX_trp__L_e 0.0 L-Tryptophan exchange\n",
      "EX_h2o2_e 0.0 Hydrogen peroxide exchange\n",
      "EX_tyr__L_e 0.0 L-Tyrosine exchange\n",
      "EX_h2o_e 7.869819466992122 H2O exchange\n",
      "EX_ump_e 0.0 UMP exchange\n",
      "EX_ura_e 0.0 Uracil exchange\n",
      "3UMPt6 0.0 3  UMP transport in via proton symport\n",
      "EX_h_e 0.7676647449524197 H+ exchange\n",
      "5MTRt2 0.0 5 Methylthio D ribose transport inout via proton symport\n",
      "6PGCt6 0.0 6 Phospho D gluconate transport inout via proton symport\n",
      "EX_urate_e 0.0 Urate exchange\n",
      "2HXMPt6 0.0 2  hydroxymethyl phenol transport inout via proton symport\n",
      "EX_hexs_e 0.0 Hexanesulfonate exchange\n",
      "2PGLYCt6 0.0 2 Phosphoglycolate transport inout via proton symport\n",
      "2PGt6 0.0 D Glycerate 2 phosphate transport inout via proton symport\n",
      "3AMPt6 0.0 3  AMP transport inout via proton symport\n",
      "3CMPt6 0.0 3  CMP transport in via proton symport\n",
      "EX_urea_e 0.0 Urea exchange\n",
      "EX_uri_e 0.0 Uridine exchange\n",
      "EX_val__L_e 0.0 L-Valine exchange\n",
      "EX_xan_e 0.0 Xanthine exchange\n",
      "EX_xtsn_e 0.0 Xanthosine exchange\n",
      "EX_xyl__D_e 0.0 D-Xylose exchange\n",
      "EX_zn2_e 0.0 Zinc exchange\n",
      "3GMPt6 0.0 Guanosine 3  phsophate transport inout via proton symport\n",
      "3PGt6 0.0 3 Phospho D glycerate transport inout via proton symport\n",
      "ABTt_1 0.0 L arabinitol transport via passive diffusion\n",
      "F6Pt6_2 0.0 Fructose 6 phosphate transport via phosphate antiport\n",
      "EX_hg2_e 0.0 Hg2+ exchange\n",
      "EX_his__L_e 0.0 L-Histidine exchange\n",
      "EX_hqn_e 0.0 Hydroquinone exchange\n",
      "EX_hxan_e 0.0 Hypoxanthine exchange\n",
      "PHEt2r 0.0 L phenylalanine reversible transport via proton symport\n",
      "ACACt2 0.0 Acetoacetate transport via proton symport\n",
      "EX_icit_e 0.0 Isocitrate exchange\n",
      "2S6HCCi 3.137905955969026e-05 2 succinyl 6 hydroxy 2 4 cyclohexadiene 1 carboxylate synthase\n",
      "26DPAi 0.0 Dipicolinate synthase\n",
      "3AMBAt2 0.0 3 aminobutyrate transport in via proton symport\n",
      "FDH 0.0009328782067644727 Formate dehydrogenase\n",
      "FDMO_1 0.0 FmnRD dependent monooxygenase  methanesulfonate \n",
      "EX_ile__L_e 0.0 L-Isoleucine exchange\n",
      "AAMYL_1 0.0 Alpha amylase\n",
      "EX_inost_e 0.0 Myo-Inositol exchange\n",
      "EX_ins_e 0.0 Inosine exchange\n",
      "EX_istnt_e 0.0 Isethionate exchange\n",
      "ABTA 0.0 4-aminobutyrate transaminase\n",
      "EX_k_e -0.08332107637508172 K+ exchange\n",
      "EX_lac__L_e 0.0 L-Lactate exchange\n",
      "EX_lanth_e 0.0 Lanthionine exchange\n",
      "FBA 1.5836747984079802 Fructose-bisphosphate aldolase\n",
      "EX_lcts_e 0.0 Lactose exchange\n",
      "FBA2 0.0 D Fructose 1 phosphate D glyceraldehyde 3 phosphate lyase\n",
      "FBP 0.0 Fructose-bisphosphatase\n",
      "FCLT_2 0.0 Heme B synthesis reaction\n",
      "FDMO1 0.0 FMN dependent monooxygenase  ethanesulfonate \n",
      "FDMO2_1 0.0 FMN dependent monooxygenase  butanesulfonate \n",
      "FDMO3_1 0.0 FMN dependent monooxygenase  hexanesulfonate \n",
      "AB6PGH 0.0 Arbutin 6-phosphate glucohydrolase\n",
      "ABUTD 0.0 Aminobutyraldehyde dehydrogenase\n",
      "MACCOAT 1.734723475976807e-18 2-Methylacetyl-CoA thiolase\n",
      "ACACT2r 0.003300227707683838 Acetyl-CoA C-acyltransferase (butanoyl-CoA) (r)\n",
      "ACACT3r 0.003300227707683838 Acetyl-CoA C-acyltransferase (hexanoyl-CoA) (r)\n",
      "ACACT4r 0.003300227707683838 Acetyl-CoA C-acyltransferase (octanoyl-CoA) (r)\n",
      "ACACT5r_1 0.003300227707683838 Acetyl CoA C acyltransferase  decanoyl CoA \n",
      "ACACT6r_1 0.003300227707683838 Acetyl CoA C acyltransferase  dodecanoyl CoA \n",
      "ACACT7r 0.002538636698218338 Acetyl-CoA C-acyltransferase (tetradecanoyl-CoA) (r)\n",
      "FDMO4_1 0.0 FMN dependent monooxygenase  taurine \n",
      "FDMO5_1 0.0 FMN dependent monooxygenase  sulfoacetate \n",
      "FDMO6_1 0.0 FMN dependent monooxygenase  isethionate \n",
      "ACACT1r 0.003300227707683838 Acetyl-CoA C-acetyltransferase\n",
      "ACCOAC 0.1200775158257272 Acetyl-CoA carboxylase\n",
      "EX_leu__L_e 0.0 L-Leucine exchange\n",
      "ACGApts 0.0 N-Acetyl-D-glucosamine transport via PEP:Pyr PTS\n",
      "ACGK 0.022769990433398623 Acetylglutamate kinase\n",
      "FE2abc 0.0 Iron (II) transport via ABC system\n",
      "EX_lipt_e 0.0 Lipoate exchange\n",
      "ACGS 0.0 N-acetylglutamate synthase\n",
      "EX_lys__L_e 0.0 L-Lysine exchange\n",
      "FOLD3 4.329366488132846e-05 Dihydropteroate synthase\n",
      "ACMANAt2 0.0 N acetyl D mannosamine proton symport\n",
      "ACNAMt2 0.0 N-acetylneuraminate proton symport\n",
      "EX_madg_e 0.0 Alpha Methyl D glucoside exchange\n",
      "ACNPLYS 0.0 N-Acetylneuraminate pyruvate-lyase\n",
      "EX_mal__D_e 0.0 D-Malate exchange\n",
      "ALLTNt2r 0.0 Allantoin transport in via proton symport\n",
      "EX_mal__L_e 0.0 L-Malate exchange\n",
      "ACt2r 0.0 Acetate reversible transport via proton symport\n",
      "ADEt2 0.0 Adenine transport in via proton symport\n",
      "ACKr 0.050384624543489735 Acetate kinase\n",
      "ACLDC 0.002021000181871325 Acetolactate decarboxylase\n",
      "ALLTAHr 0.0 Allantoate amidinohydrolase  reversible\n",
      "EX_malt_e 0.0 Maltose exchange\n",
      "AMPt6 0.0 AMP transport inout via proton symport\n",
      "ACODA_1 0.0 Acetylornithine deacetylase\n",
      "ACOAD20 0.0 Acyl CoA dehydrogenase  2 methylbutnoyl CoA \n",
      "EX_malttr_e 0.0 Maltotriose exchange\n",
      "EX_man1p_e 0.0 D Mannose 1 phosphate exchange\n",
      "ACONT 1.9715697449757554 Aconitate hydratase\n",
      "ACOAD2 0.003300227707683838 Acyl-CoA dehydrogenase (hexanoyl-CoA)\n",
      "EX_man6p_e 0.0 D-Mannose 6-phosphate exchange\n",
      "ACOAD1 0.003300227707683838 Acyl-CoA dehydrogenase (butanoyl-CoA)\n",
      "ACOADH2 -5.421010862427522e-19 Acyl CoA dehydrogenase  isobutyryl CoA \n",
      "ARABDI 0.0 L-fucose isomerase\n",
      "ARAB_Dt 0.0 D arabinose reversible transport\n",
      "ACOTA -0.022769990433398623 Acetylornithine transaminase\n",
      "ARGKr 0.0 ATPL arginine N phosphotransferase\n",
      "EX_man_e 0.0 D-Mannose exchange\n",
      "ARGPt6 0.0 L arginine phosphate transport inout via proton symport\n",
      "ACPS1 0.0 Acyl-carrier protein synthase\n",
      "EX_3ump_e 0.0 3'-UMP exchange\n",
      "EX_mbdg_e 0.0 Beta Methylglucoside exchange\n",
      "APATi 0.05038462454348949 Tetrahydrodipicolinate acetyltransferase\n",
      "EX_melib_e 0.0 Melibiose exchange\n",
      "EX_meoh_e 0.0 Methanol exchange\n",
      "EX_met__D_e 0.0 D-Methionine exchange\n",
      "APRAUR 0.0 5-amino-6-(5-phosphoribosylamino)uracil reductase\n",
      "APSR 0.02009156356383357 Adenosine 5'-phosphosulfate reductase;\n",
      "ACPpds 0.0 [acyl-carrier-protein] phosphodiesterase\n",
      "ACS 0.0 Acetyl-CoA synthetase\n",
      "EX_met_L_ala__L_e 0.0 Met L ala L exchange\n",
      "ALKP 0.0 Alkaline phosphatase\n",
      "APTA1i 0.05038462454348949 Acetyl L L diaminopimelate aminotransferase\n",
      "EX_met__L_e 0.0 L-Methionine exchange\n",
      "ALCD19_L 0.0 Alcohol dehydrogenase  L glyceraldehyde to glycerol  NADP \n",
      "ACS2 0.0 Acetyl-CoA synthase (propionate)\n",
      "ACTD2 0.002021000181871325 Acetoin dehydrogenase\n",
      "ACTNabc 0.0 Acetoin transport via ABC system\n",
      "ACTNabc1 0.0 Acetoin efflux  via ABC system\n",
      "ALCD2y 0.0 Alcohol dehydrogenase (ethanol, NADP)\n",
      "ADADir 0.0 Deoxycytidine kinase\n",
      "EX_metox__R_e 0.0 L methionine R oxide exchange\n",
      "ADNK1 0.0 Adenosine kinase\n",
      "ADCL 4.329366488132846e-05 4-aminobenzoate synthase\n",
      "ADD 0.0 Adenine deaminase\n",
      "ALLTN 0.0 Allantoinase\n",
      "ALTRH 0.0 Altronate hydrolase\n",
      "EX_metox_e 0.0 L Methionine S oxide exchange\n",
      "ARABRr 0.0 Arabinose reductase\n",
      "ARAI 0.0 L-arabinose isomerase\n",
      "ARBTpts 0.0 Arbutin transport via PEPPyr PTS\n",
      "ADCS 4.329366488132846e-05 4-amino-4-deoxychorismate synthase\n",
      "ADK1 0.18483504727784855 Adenylate kinase\n",
      "EX_mg2_e -0.01199883332353841 Mg exchange\n",
      "AMAA 0.0 N-acetylmuramoyl-L-alanine amidase\n",
      "ADK2 0.0 Adenylate kinase (Inorganic triphosphate)\n",
      "AMAOTr 0.0 Adenosylmethionine-8-amino-7-oxononanoate transaminase\n",
      "EX_mn2_e 0.0 Mn2+ exchange\n",
      "AMMQT7 3.137905955975795e-05 S adenosylmethione2 demthylmenaquinone methyltransferase  menaquinone 7 \n",
      "ANPRT 0.00640982173022183 Anthranilate phosphoribosyltransferase\n",
      "ADMDC 0.0 Adenosylmethionine decarboxylase\n",
      "EX_mnl_e 0.0 D-Mannitol exchange\n",
      "ADNt2 0.0 Adenosine transport in via proton symport\n",
      "ADPDA 0.05038462454348949 N6 acetyl LL 2 6 diaminoheptanedioate amidohydrolase\n",
      "ADPRDP 0.0 ADPribose diphosphatase\n",
      "ANS 0.00640982173022183 Anthranilate synthase\n",
      "ANTIMt1 0.0 Antimonite  transporter via uniport\n",
      "EX_mobd_e 0.0 Molybdate exchange\n",
      "AOXSr 0.0 8-amino-7-oxononanoate synthase\n",
      "ARBabc 0.0 L-arabinose transport via ABC system\n",
      "ARBt2r 0.0 L-arabinose transport via proton symport\n",
      "ARGDC 0.0 Arginine decarboxylase\n",
      "ADPT 0.0 Adenine phosphoribosyltransferase\n",
      "ADSK 0.0 Adenylyl-sulfate kinase\n",
      "ADSL2r 0.021487577815071147 Adenylosuccinate lyase\n",
      "ADSL1r 0.021561188842008316 Adenylsuccinate lyase\n",
      "ADSS 0.021561188842008316 Adenylosuccinate synthase\n",
      "AGDC 0.0 N-acetylglucosamine-6-phosphate deacetylase\n",
      "ARGN_1 0.0 Arginase\n",
      "ARGSL 0.022769990433398623 Argininosuccinate lyase\n",
      "ARGSS 0.022769990433398623 Argininosuccinate synthase\n",
      "AGMT 0.0 Agmatinase\n",
      "AGPATr_BS 0.012693183491091662 1 acylglycerol 3 phosphate O acyltransferase\n",
      "EX_mops_e 0.0 MOPS exchange\n",
      "AGPR -0.022769990433398623 N-acetyl-g-glutamyl-phosphate reductase\n",
      "AHCYSNS 0.0 S-adenosylhomocysteine nucleosidase\n",
      "ASNt2r 0.0 L asparagine reversible transport via proton symport\n",
      "EX_mso3_e 0.0 Methanesulfonate exchange\n",
      "ASPt2r 0.0 L aspartate reversible transport via proton symport\n",
      "AHMMPS_1 0.0 4 amino 5 hydroxymethyl 2 methylpyrimidine synthetase\n",
      "AHSERL3 0.0 Cysteine synthase  Selenide \n",
      "AHSERL4 0.0 Cysteine synthase  Thiosulfate \n",
      "EX_na1_e 0.0 Sodium exchange\n",
      "AICART 0.031130032511894654 Phosphoribosylaminoimidazolecarboxamide formyltransferase\n",
      "AKGDH 1.8563371087595146 2-Oxogluterate dehydrogenase\n",
      "EX_nac_e 0.0 Nicotinate exchange\n",
      "AIRC1 0.021487577815071147 Phosphoribosylaminoimidazole carboxylase  ATP dependent \n",
      "ATPM 9.0 ATP maintenance requirement\n",
      "BIOMASS_BS_10 0.1179663893223982 B subtilis biomass  demand \n",
      "EX_nh4_e -0.9449636274148077 Ammonia exchange\n",
      "BTDt6_RR 0.0 Butanediol transport inout via proton symport\n",
      "CHORt2 0.0 Chorismate transport inout via proton symport\n",
      "ALAALAr 0.012010983861638616 D-alanine-D-alanine ligase (reversible)\n",
      "AKGt2r 0.0 2 oxoglutarate reversible transport via symport\n",
      "EX_ni2_e 0.0 Ni2+ exchange\n",
      "AKP1 4.329366488132846e-05 Alkaline phosphatase  Dihydroneopterin \n",
      "ALAD_L 0.0 L-alanine dehydrogenase\n",
      "ALAR 0.04157713595029 Alanine racemase\n",
      "ALATA_L -0.08507358505485534 L-alanine transaminase\n",
      "CITRt2r 0.0 Citrulline reversible transport via proton symport\n",
      "CMPt6 0.0 CMP transport inout via proton symport\n",
      "CO2t -5.980675626954003 CO2 transporter via diffusion\n",
      "DALAt2r 0.0 D-alanine transport via proton symport\n",
      "ARSNAt1 0.0 Arsenate transporter\n",
      "ALAt2r 0.0 L alanine reversible transport via proton symport\n",
      "EX_no2_e 0.0 Nitrite exchange\n",
      "ALCD19y 0.0 Alcohol dehydrogenase  glycerol  NADP \n",
      "ARSt1 0.0 Arsenite transporter via uniport\n",
      "ARGabc 0.0 L-arginine transport via ABC system\n",
      "EX_no3_e 0.0 Nitrate exchange\n",
      "ARGt2r 0.0 L arganine reversible transport via proton symport\n",
      "ARSENBabc 0.0 Arsenobetaine transport in via ABC system\n",
      "CYSGLYt2ir 0.0 Cys Gly transport in via symport\n",
      "CYSTabc 0.0 L cystine transport via ABC system\n",
      "ARSENBt2 0.0 Arsenobetaine transport in via proton symport\n",
      "EX_o2_e -5.705899229532262 O2 exchange\n",
      "ASAD -0.13051919281019625 Aspartate-semialdehyde dehydrogenase\n",
      "ASNN 0.0 L-asparaginase\n",
      "ASNS1 0.017457492056653742 Asparagine synthase (glutamine-hydrolysing)\n",
      "CYSabc 0.0 L-cysteine transport via ABC system\n",
      "DEXTRINt2 0.0 Dextrin transport in via proton symport\n",
      "ASP1DC 0.0 Aspartate 1-decarboxylase\n",
      "EX_orn__L_e 0.0 Ornithine exchange\n",
      "ASPCT 0.014388950337599519 Aspartate carbamoyltransferase\n",
      "ASPK 0.13051919281019625 Aspartate kinase\n",
      "DHAt 0.0 Dihydroxyacetone transport via facilitated diffusion\n",
      "EX_pala_e 0.0 Palatinose exchange\n",
      "EX_4abut_e 0.0 4-Aminobutanoate exchange\n",
      "ASPO6 0.0020424700647280296 L-aspartate oxidase\n",
      "ASPO1 0.0 L-aspartate oxidase\n",
      "ASPT 0.0 L-aspartase\n",
      "ASPTA -0.24768753950882105 Aspartate transaminase\n",
      "EX_pep_e 0.0 Phosphoenolpyruvate exchange\n",
      "ATPPRT 0.009642454696823507 ATP phosphoribosyltransferase\n",
      "DADK 0.0 Deoxyadenylate kinase\n",
      "EX_phe__L_e 0.0 L-Phenylalanine exchange\n",
      "DADNt2 0.0 Deoxyadenosine transport in via proton symport\n",
      "ATPS4r 18.771540152105295 ATP synthase (four protons for one ATP)\n",
      "BETALDHx 0.0 Betaine-aldehyde dehydrogenase\n",
      "BG_CELLB 0.0 Beta glucosidase  cellobiose \n",
      "CYSS_2 0.0 Cysteine synthase\n",
      "EX_pi_e -0.1810531628025574 Phosphate exchange\n",
      "DHPPDA 0.0 Diaminohydroxyphosphoribosylaminopyrimidine deaminase\n",
      "BG_MADG 0.0 Beta glucosidase  methyl alpha D glucoside \n",
      "BG_MBDG 0.0 Beta glucosidase\n",
      "BILEt4 0.0 Bile acid transport in via sodium symport\n",
      "DDPA 0.040312890155921724 3-deoxy-D-arabino-heptulosonate 7-phosphate synthetase\n",
      "EX_pnto__R_e 0.0 (R)-Pantothenate exchange\n",
      "CRNabc 0.0 L-carnitine transport via ABC system\n",
      "BTDD_RR 0.0  R R  butanediol dehydrogenase\n",
      "EX_ppa_e 0.0 Propionate exchange\n",
      "BLACT 0.0 Beta lactamase\n",
      "CTPS1 0.006760535805677318 CTP synthase  NH3 \n",
      "BTS_1 0.0 Biotin synthase\n",
      "CPPPGO 0.0 Coproporphyrinogen oxidase (O2 required)\n",
      "EX_ppi_e 0.0 Diphosphate exchange\n",
      "CRO4t3 0.0 Chromate transport out via proton antiport\n",
      "CS 1.9715697449757554 Citrate synthase\n",
      "EX_pro__L_e 0.0 L-Proline exchange\n",
      "CTPS2 0.0 CTP synthase (glutamine)\n",
      "CYOR3m 11.395323744997317 Cytochrome c reductase  menaquinol 7 3 protons \n",
      "BUTKr 0.0 Butyrate kinase\n",
      "BUTSabc 0.0 Butanesulfonate transport via ABC system\n",
      "EX_prolb_e 0.0 Proline betaine exchange\n",
      "CA2abc1 0.0 Calcium efflux  via ABC system\n",
      "CAT 0.00823735703360444 Catalase\n",
      "CAt4 -0.0003780822777782869 Calcium transport inout via proton antiporter\n",
      "CYOO3 11.395323744997317 Cytochrome c oxidase  He   2 \n",
      "CYSTGL 0.006722904527483473 Cystathionine g lyase\n",
      "CYSTL 0.0 Cystathionine b-lyase\n",
      "CYSTS 0.006722904527483473 Cystathionine beta synthase\n",
      "DCYTD 0.0 Deoxycytidine deaminase\n",
      "CYTB_B2 0.0 Menaquinol oxidase  71 protons \n",
      "CYTB_B3 0.0 Menauinol oxidase  72 protons \n",
      "CBL2abc 0.0 Cobalamin uptake in via ABC transport\n",
      "CD2t4 0.0 Cadminum transport out via antiport\n",
      "CBPS 0.03715894077099815 Carbamoyl-phosphate synthase (glutamine-hydrolysing)\n",
      "CYTD 0.0 Cytidine deaminase\n",
      "EX_pser__D_e 0.0 D O Phosphoserine exchange\n",
      "EX_pser__L_e 0.0 O-Phospho-L-serine exchange\n",
      "DJENKabc 0.0 Djenkolate transporter via ABC system\n",
      "CYTDK1 3.137905955975795e-05 ATP:cytidine 5'-phosphotransferase\n",
      "CD2abc1 0.0 Cadmium transport out via ABC system\n",
      "CDGPT_BS 0.002453700897907429 CDPdiacylglycerolsn glycerol 3 phosphate 3 phosphatidyltransferase\n",
      "CDPDSP_BS 0.0066061178020543135 CDPdiacylglycerol serine O phosphatidyltransferase\n",
      "CDPMEK 0.00021965341691830039 4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol kinase\n",
      "CHCOAL 0.0 6-carboxyhexanoate-CoA ligase\n",
      "CYTDK2 0.0 Cytidine kinase (GTP)\n",
      "DCYTt2 0.0 Deoxycytidine transport in via proton symport\n",
      "CHOLD2 0.0 Choline dehydrogenase  FAD dependent \n",
      "DTMPt6 0.0 DTMP transport inout via proton symport\n",
      "CHORM 0.033828395701258875 Chorismate mutase\n",
      "CHORS 0.040312890155921724 Chorismate synthase\n",
      "CYTK1 0.06356453735692542 Cytidylate kinase (CMP)\n",
      "CYTK2 0.0 Cytidylate kinase (dCMP)\n",
      "CYTOM -3.137905955975795e-05 Cytosine 5 methyltransferase\n",
      "CITt10 0.0 Citrate transport in via Mg complex\n",
      "CITt11 0.0 Citrate transport in via Mn complex\n",
      "CITt12 0.0 Citrate transport in via Ni complex\n",
      "D4DGCD 0.0 5 dehydro 4 deoxyglucarate dehydratase\n",
      "Cut1 0.0 Copper export via ATPase\n",
      "DAAOP 0.0 D alanyl aminopeptidase\n",
      "DAGK_BS 0.0 Diacylglycerol kinase\n",
      "DALAOX 0.0 D amino acid oxidase\n",
      "CITt13 0.0 Citrate transport in via Co complex\n",
      "CLPNS2_BS 5.8983194661199514e-05 Cardiolipin synthase\n",
      "CITt14 0.0 Citrate transport in via Ca complex\n",
      "CITt2r 0.0 Citrate reversible transport via symport\n",
      "DAPDC 0.03837364068185088 Diaminopimelate decarboxylase\n",
      "DAPE 0.05038462454348949 Diaminopimelate epimerase\n",
      "ETHAAL 0.0 Ethanolamine ammonia-lyase\n",
      "DB4PS 0.0 3,4-Dihydroxy-2-butanone-4-phosphate synthase\n",
      "DBTS 0.0 Dethiobiotin synthase\n",
      "FOLt 0.0 Folate transport via proton simport\n",
      "FORAMD 0.0 FORAMD\n",
      "DCMPDA 0.0 DCMP deaminase\n",
      "ETOHt3 0.0 Ethanol transport out via proton antiport\n",
      "CMPDAi 0.0 CMP deaminase\n",
      "FRTT 3.137905955975795e-05 Farnesyltranstransferase\n",
      "EX_2ddglcn_e 0.0 2-Dehydro-3-deoxy-D-gluconate exchange\n",
      "DDGLCNt2r 0.0 2-dehydro-3-deoxy-D-gluconate transport via proton symport, reversible\n",
      "DDGLK 0.0 2-dehydro-3-deoxygluconokinase\n",
      "EX_2hxmp_e 0.0 2  Hydroxymethyl phenol exchange\n",
      "DGK1 0.0020523792414310835 Deoxyguanylate kinase (dGMP:ATP)\n",
      "DGNSK 0.0020523792414310835 Deoxyguanosine kinase\n",
      "G1Pt6_2 0.0 Glucose 1 phosphate transport via phosphate antiport\n",
      "DHAD1 0.08543066931533524 Dihydroxy-acid dehydratase (2,3-dihydroxy-3-methylbutanoate)\n",
      "G3POA_BS 0.012693183491091662 Glycerol 3 phosphate O acyltransferase\n",
      "EX_2pg_e 0.0 D Glycerate 2 phosphate exchange\n",
      "DHAD2 0.012947047160913506 Dihydroxy-acid dehydratase (2,3-dihydroxy-3-methylpentanoate)\n",
      "DHBD 0.0 2,3-dihydro-2,3-dihydroxybenzoate dehydrogenase\n",
      "DHBS 0.0 2,3-dihydroxybenzoate adenylate synthase\n",
      "EX_2pglyc_e 0.0 2 Phosphoglycolate exchange\n",
      "DHDPRy 0.05038462454348949 Dihydrodipicolinate reductase (NADPH)\n",
      "DHDPS 0.05038462454348949 Dihydrodipicolinate synthase\n",
      "G5SADs 0.018950356713528693 L-glutamate 5-semialdehyde dehydratase (spontaneous)\n",
      "G6Pt6_2 0.0 Glucose-6-phosphate transport via phosphate antiport\n",
      "GALTpts 0.0 Galactitol transport via PEP:Pyr PTS\n",
      "FOLR2 0.0 Folate reductase\n",
      "DHFR 0.0027759850735346826 Dihydrofolate reductase\n",
      "DHFS 4.329366488132846e-05 Dihydrofolate synthase\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DHNAOT7 3.137905955975795e-05 1 4 dihydroxy 2 naphthoate octaprenyltransferase\n",
      "EX_3amba_e 0.0 3 aminobutanoic acid exchange\n",
      "EX_3amp_e 0.0 3'-AMP exchange\n",
      "GAM6Pt 0.0 D glucosamine 6 phosphate reversible uniport\n",
      "EX_3cmp_e 0.0 3'-cmp exchange\n",
      "GGTT 3.137905955975795e-05 Geranylgeranyltranstransferase\n",
      "GCALDD 4.329366488132846e-05 Glycolaldehyde dehydrogenase\n",
      "EX_3gmp_e 0.0 3'-GMP exchange\n",
      "GAPD 4.232253109449068 Glyceraldehyde-3-phosphate dehydrogenase\n",
      "G1Dx 0.0 Glucose 1 dehydrogenase  NAD \n",
      "GALt2 0.0 D galactose transport in via proton symport\n",
      "DHNPA_1 4.329366488132846e-05 Dihydroneopterin aldolase\n",
      "DHORTS -0.014388950337599519 Dihydroorotase\n",
      "DHORDi 0.014388950337599519 Dihydoorotic acid dehydrogenase  irreversible \n",
      "EX_3pg_e 0.0 3 Phospho D glycerate exchange\n",
      "GK1 0.007457127334626088 Guanylate kinase (GMP:ATP)\n",
      "ETHSabc 0.0 Ethanesulfonate transport via ABC system\n",
      "GBBTNabc 0.0 Gamma butyrobetaine transport in via ABC system\n",
      "GK2 0.0 Guanylate kinase  GMPdATP \n",
      "GLCRAL_1 0.0 5 dehydro 4 deoxyglucarate aldolase\n",
      "GLU_Dt2 0.0 D glutamate transport in via proton symport\n",
      "DINSK 0.0 Deoxyinosine kinase\n",
      "GLXt2 0.0 Glyoxylate transport in via proton symport\n",
      "GLYCOGENt 0.0 Glycogen transport in\n",
      "FE3t 0.00040698404316227453 Ferric iron uptake, plasma membrane\n",
      "DHQTi 0.040312890155921724 3-dehydroquinate dehydratase, irreversible\n",
      "FEDCabc 0.0 Iron  III  dicitrate transport via ABC system\n",
      "GLYt2r 0.0 Glycine reversible transport via proton symport\n",
      "FERXAabc 0.0 Ferroxamine transport via ABC system\n",
      "GMPt6 0.0 GMP transport inout via proton symport\n",
      "DHQS 0.040312890155921724 3-dehydroquinate synthase\n",
      "H2Ot -7.869819466992122 H2O transport via diffusion\n",
      "FFSD 0.0 Beta-fructofuranosidase\n",
      "FGLU_1 0.0 Formimidoylglutamase\n",
      "FLVR_1 0.0 Flavin reductase\n",
      "FMETDF 0.0 Formylmethionine deformylase\n",
      "FMNRy 0.0 FMN reductase  NADPH dependent \n",
      "HCO3E 0.4543378752313092 HCO3 equilibration reaction\n",
      "HEMECS 0.0 Heme C synthase\n",
      "HEMEDS 0.0 Heme D synthase\n",
      "FMETTRS_1 0.0 Methionyl tRNA formyltransferase\n",
      "FMNAT 0.0 FMN adenylyltransferase\n",
      "FORt2 0.0 Formate transport in via proton symport\n",
      "GSNt2 0.0 Guanosine transport in via proton symport\n",
      "DIPEPabc1 0.0 Dipeptide transport via ABC system  ala asp \n",
      "GLYBt2r 0.0 Glycine betaine transport via proton symport, reversible\n",
      "GLYO1 0.0 Glycine oxidase\n",
      "FTHFD 0.0008895845418838699 Formyltetrahydrofolate deformylase\n",
      "HEX7 0.0 Hexokinase (D-fructose:ATP)\n",
      "FUM 1.9221558658499922 Fumarase\n",
      "DKI 0.0 5 Dehydro 4 deoxy D glucuronate ketol isomerase\n",
      "FUMt2r 0.0 Fumarate reversible transport via symport\n",
      "GTPCII 0.0 GTP cyclohydrolase II\n",
      "G1PACT 0.03857194218230186 Glucosamine-1-phosphate N-acetyltransferase\n",
      "GRTT 3.137905955975795e-05 Geranyltranstransferase\n",
      "G1PPDM 0.0 Glucose 1 phosphate phosphodismutase\n",
      "DKII 0.0 2 dehydro 3 deoxy D gluconate 5 dehydrogenase\n",
      "G3PCT 0.05437660715815956 Glycerol-3-phosphate cytidylyltransferase\n",
      "GLCRt2r 0.0 D-glucarate transport via proton symport, reversible\n",
      "GLCURt2r 0.0 D-glucuronate transport via proton symport, reversible\n",
      "G1SAT 0.0 Glutamate-1-semialdehyde aminotransferase\n",
      "DKMD2K 0.0 1 2 dihydroxy 3 keto 5 methylthiopentene dioxygenase  2 keto 4 methylthiobutyrate \n",
      "G3PD1ir 0.06946450835249746 Glycerol 3 phosphate dehydrogenase  NAD \n",
      "G3PD4 0.0 Glycerol 3 phosphate dehydrogenase  menaquinone 7 \n",
      "GLCP 0.0 Glycogen phosphorylase\n",
      "GLCRD_1 0.0 Glucarate dehydratase\n",
      "GLCS1 0.0 Glycogen synthase (ADPGlc)\n",
      "G5SD 0.018950356713528693 Glutamate-5-semialdehyde dehydrogenase\n",
      "G6PDH2r 0.0 Glucose 6-phosphate dehydrogenase\n",
      "G6PDA 0.0 Glucosamine-6-phosphate deaminase\n",
      "G6PI 0.0 Glucose 6 phosphate isomerase\n",
      "GLCt2 0.0 D-glucose transport in via proton symport\n",
      "GALCTD_1 0.0 Galactarate dehydratase\n",
      "GLNS 0.19253872436615554 Glutamine synthetase\n",
      "GALCTRt2 0.0 Galactarate transport in via proton symport\n",
      "GLGC 0.0 Glucose-1-phosphate adenylyltransferase\n",
      "DKMD3M 0.0 1 2 dihydroxy 3 keto 5 methylthiopentene dioxygenase  3 methylthiopropionate \n",
      "GLNabc 0.0 L-glutamine transport via ABC system\n",
      "GLTPD 0.0 Galactitol-1-phosphate dehydrogenase\n",
      "GLUPRT 0.021487577815071147 Glutamine phosphoribosyldiphosphate amidotransferase\n",
      "GALKr 0.0 Galactokinase\n",
      "GALS1 0.0 A galactosidase  galactomannan \n",
      "GALS2 0.0 A galactosidase  galactose oligosaccharides \n",
      "GALS3 0.0 A-galactosidase (melibiose)\n",
      "GALS4 0.0 A galactosidase  stachyose \n",
      "GLU5K 0.018950356713528693 Glutamate 5-kinase\n",
      "GALT 0.0 Galactose 1 phosphate uridylyltransferase\n",
      "GLUDxi 0.0 Glutamate dehydrogenase  NAD \n",
      "GALUi 0.029682113049355294 UTP-glucose-1-phosphate uridylyltransferase (irreversible)\n",
      "DM1PE 0.0 2 3 diketo 5 methylthiopentyl 1 phosphate enolase\n",
      "GALURt2r 0.0 D-galacturonate transport via proton symport, reversible\n",
      "GLUN 0.0 Glutaminase\n",
      "GLUTRR 0.0 Glutamyl-tRNA reductase\n",
      "GAMpts 0.0 D-glucosamine transport via PEP:Pyr PTS\n",
      "DMATT 3.137905955975795e-05 Dimethylallyltranstransferase\n",
      "GAPDi_nadp 1.1764333411770378 Glyceraldehyde-3-phosphate dehydrogenase (NAD(P)+) (phosphorylating)\n",
      "GLUR -0.012010983861638616 Glutamate racemase\n",
      "GLUSy 0.0 Glutamate synthase (NADPH)\n",
      "GLUTRS 0.0 Glutamyl-tRNA synthetase\n",
      "GLUt4 0.0 Na+/glutamate symport\n",
      "GLUt2r 0.0 L glutamate transport via proton symport  reversible\n",
      "GLXCBL 2.164683244066422e-05 Glyoxalate carboligase\n",
      "GARFT 0.021487577815071147 Phosphoribosylglycinamide formyltransferase\n",
      "GART 0.0 GAR transformylase-T\n",
      "GBEZY 0.0 1 4 alpha glucan branching enzyme\n",
      "GF6PTA 0.03857194218230186 Glutamine-fructose-6-phosphate transaminase\n",
      "GGTAe2 0.0 Gamma glutamyltransferase\n",
      "GLY3Pt2 0.0 Glycerol 3 phosphate transport in via proton symport\n",
      "DMPPS 0.0 1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate reductase (dmpp)\n",
      "GLYAT 0.0 Glycine C-acetyltransferase\n",
      "GLYCK 2.164683244066422e-05 Glycerate kinase\n",
      "GHMT2r 0.06965183897873445 Glycine hydroxymethyltransferase, reversible\n",
      "GLCNt2ir 0.0 D gluconate transport via proton symport\n",
      "DPCOAK 0.0 Dephospho-CoA kinase\n",
      "GLYCTO1 4.329366488132845e-05 Glycolate oxidase\n",
      "HIBHr -9.76680237544791e-19 3 hydroxyisobutyryl CoA hydrolase\n",
      "GLYCLTt2r 0.0 Glycolate transport via proton symport, reversible\n",
      "HQNt6 0.0 Hydroquinone transport inout via proton symport\n",
      "DPR 0.0 2-dehydropantoate 2-reductase\n",
      "ISOMS 0.0 Isomaltulose synthase\n",
      "GLYCt 0.0 Glycerol transport via channel\n",
      "GLYK 5.898319466292975e-05 Glycerol kinase\n",
      "DRBK 0.0 Deoxyribokinase\n",
      "GMPR 0.0 GMP reductase\n",
      "GMPS2 0.009568843669886338 GMP synthase\n",
      "GNKr 0.0 Gluconokinase\n",
      "GPDDA4 0.0 Glycerophosphodiester phosphodiesterase (Glycerophosphoglycerol)\n",
      "GPDDA1 0.0 Glycerophosphodiester phosphodiesterase (Glycerophosphocholine)\n",
      "IMPC -0.031130032511894654 IMP cyclohydrolase\n",
      "HEXSabc 0.0 Hexanesulfonate transport via ABC system\n",
      "ICITt10 0.0 Isocitrate transport in via Mg complex\n",
      "ICITt2 0.0 Isocitrate transport inout via proton symport\n",
      "DRIBabc 0.0 Deoxyribose transport via ABC system\n",
      "DRPA -0.002021000181871325 Deoxyribose-phosphate aldolase\n",
      "IPDPS 0.00021965341691830039 1-hydroxy-2-methyl-2-(E)-butenyl 4-diphosphate reductase (ipdp)\n",
      "IPDDI 3.137905955975795e-05 Isopentenyl-diphosphate D-isomerase\n",
      "GTHPe 0.0 Glutathione peroxidase\n",
      "RAFH 0.0 Raffinose hydrolyzing enzyme\n",
      "DTMPK 0.002701312349093596 DTMP kinase\n",
      "HXPRT 0.0 Hypoxanthine phosphoribosyltransferase (Hypoxanthine)\n",
      "HETZK 0.0 Hydroxyethylthiazole kinase\n",
      "GTPCI 4.329366488132846e-05 GTP cyclohydrolase I\n",
      "GTPDPK 0.0 GTP diphosphokinase\n",
      "DURIK1 0.0 Deoxyuridine kinase (ATP:Deoxyuridine)\n",
      "GUAD 0.0 Guanine deaminase\n",
      "GUAPRT -0.0020523792414310835 Guanine phosphoribosyltransferase\n",
      "HXANt2r 0.0 Hypoxanthine reversible transport via proton symport\n",
      "HEMEOS_1 0.0 Heme O synthase\n",
      "GUI1 0.0 Glucuronate isomerase (D-glucuronate)\n",
      "DURIPP 0.0 Deoxyuridine phosphorylase\n",
      "GUAt2r 0.0 Guanine reversible transport via proton symport\n",
      "HEX1 0.0 Hexokinase (D-glucose:ATP)\n",
      "HGt 0.0 Mercury transport\n",
      "GUI2 0.0 Glucuronate isomerase (D-galacturonate)\n",
      "DUTPDP 0.0 DUTP diphosphatase\n",
      "HACD2 0.003300227707683838 3-hydroxyacyl-CoA dehydrogenase (3-oxohexanoyl-CoA)\n",
      "HACD1_1 0.003300227707683838 3 hydroxyacyl CoA dehydrogenase  acetoacetyl CoA \n",
      "HEXTT 3.137905955975795e-05 Trans-hexaprenyltranstransferase\n",
      "HIBD 0.0 3 hydroxyisobutyrate dehydrogenase\n",
      "HEMEAS 0.0 Heme A synthase\n",
      "DXPRIi 0.00021965341691830039 1-deoxy-D-xylulose reductoisomerase\n",
      "HISDr 0.0 Histidase r\n",
      "HISTD 0.009642454696823507 Histidinol dehydrogenase\n",
      "LAAL17 0.0 L amino acid ligase  Ala Ile \n",
      "DXPS 0.00021965341691830039 1-deoxy-D-xylulose 5-phosphate synthase\n",
      "LAAL18 0.0 L amino acid ligase  Ser Ile \n",
      "LAAL19 0.0 L amino acid ligase  Gly Met \n",
      "HISTP 0.009642454696823507 Histidinol-phosphatase\n",
      "HKM1PP 0.0 2 hydroxy 3 keto 5 methylthiopentenyl 1 phosphate phophatase\n",
      "LAAL2 0.0 L amino acid ligase  L alanylglycine \n",
      "Kt2r 0.0 Potassium reversible transport  via proton symport\n",
      "LAAL20 0.0 L amino acid ligase  Ala Met \n",
      "LAAL21 0.0 L amino acid ligase  Ser Met \n",
      "LAAL22 0.0 L amino acid ligase  Thr Met \n",
      "HISt2r 0.0 L histidine reversible transport via proton symport\n",
      "HMBS 0.0 Hydroxymethylbilane synthase\n",
      "HMGCOAS 0.0 Hydroxymethylglutaryl CoA synthase\n",
      "LAAL23 0.0 L amino acid ligase  met met \n",
      "ECOAH1_1 0.003300227707683838 3 hydroxyacyl CoA dehydratase  3 hydroxybutanoyl CoA \n",
      "LAAL24 0.0 L amino acid ligase  gly phe \n",
      "Kt3r 0.0 Potassium transport out via proton antiport\n",
      "HMGL 0.0 Hydroxymethylglutaryl-CoA lyase\n",
      "HOXPRx -2.164683244066422e-05 2 hydroxy 3 oxopropionate reductase  NAD \n",
      "HMPK1 0.0 Hydroxymethylpyrimidine kinase (ATP)\n",
      "ECOAH2 0.003300227707683838 3-hydroxyacyl-CoA dehydratase (3-hydroxyhexanoyl-CoA)\n",
      "HP5CD_1 0.0 3 hydroxy 1 pyrroline 5 carboxylate dehydrogenase\n",
      "Kt1 0.08332107637508172 Potassium transport via uniport (facilitated diffusion)\n",
      "HPPK 4.329366488132846e-05 2-amino-4-hydroxy-6-hydroxymethyldihydropteridine diphosphokinase\n",
      "LAAL25 0.0 L amino acid ligase  Ala phe \n",
      "ECOAH2c -1.0842021724855044e-18 Enoyl CoA hydratase  bacterial\n",
      "LAAF 0.0 Alpha N arabinofuranosidase\n",
      "LAAL1 0.0 L amino acid ligase  L glycine L glycine \n",
      "HSDy -0.08013456826670674 Homoserine dehydrogenase (NADPH)\n",
      "HSK 0.021979143759381266 Homoserine kinase\n",
      "HSST 0.05815542450732549 Homoserine O-succinyltransferase\n",
      "LAAL10 0.0 L amino acid ligase  Thr Thr \n",
      "LAAL11 0.0 L amino acid ligase  Ala Val \n",
      "HXAD 0.0 Hexosaminidase\n",
      "ECOAH3 0.003300227707683838 3-hydroxyacyl-CoA dehydratase (3-hydroxyoctanoyl-CoA)\n",
      "HSTPTr 0.009642454696823507 Histidinol phosphate transaminase\n",
      "LAAL12 0.0 L amino acid ligase  Ser Val \n",
      "LAAL13 0.0 L amino acid ligase  Gly Leu \n",
      "ECOAH4 0.003300227707683838 3-hydroxyacyl-CoA dehydratase (3-hydroxydecanoyl-CoA)\n",
      "LAAL14 0.0 L amino acid ligase  ser leu \n",
      "LAAL15 0.0 L amino acid ligase  Ala Leu \n",
      "LAAL16 0.0 L amino acid ligase  Thr Leu \n",
      "LAAL26 0.0 L amino acid  ser phe \n",
      "LAAL27 0.0 L amino acid ligase  Thr Phe \n",
      "ICDHyr 1.9715697449757554 Isocitrate dehydrogenase (NADP)\n",
      "ICHORT 0.0 Isochorismatase\n",
      "IG3PS_1 0.009642454696823507 Imidazole glycerol 3 phosphate synthase\n",
      "IGPDH_1 0.009642454696823507 Imidazoleglycerol phosphate dehydratase\n",
      "IGPS 0.00640982173022183 Indole-3-glycerol-phosphate synthase\n",
      "ILEDHr -0.03183971831006189 Isoleucine dehydrogenase\n",
      "IMPD 0.009568843669886338 IMP dehydrogenase\n",
      "ILETA 0.0 Isoleucine transaminase\n",
      "ILEt2r 0.0 L isoleucine reversible transport via proton symport\n",
      "LAAL28 0.0 L amino acid ligase  met phe \n",
      "LAAL29 0.0 L amino acid ligase  gly tyr \n",
      "ECOAH5_1 0.003300227707683838 3 hydroxyacyl CoA dehydratase  3 hydroxydodecanoyl CoA \n",
      "LAAL3 0.0 L amino acid ligase  L serylglycine \n",
      "LAAL30 0.0 L amino acid  Ala Tyr \n",
      "INS2D 0.0 Inositol 2-dehydrogenase\n",
      "ECOAH6 0.003300227707683838 3-hydroxyacyl-CoA dehydratase (3-hydroxytetradecanoyl-CoA)\n",
      "INSTt2 0.0 Inositol transport in via proton symport\n",
      "IPDPUPT -4.956352788505163e-19 Di trans poly cis Decaprenyl diphosphateisopentenyl diphosphate undecaprenylcistransferase\n",
      "LANTHabc 0.0 Lanthionine transport via ABC system\n",
      "LIPO1S24_BS 8.25764725256792e-07 Lipoteichoic acid synthesis  n24   linked  glucose substituted\n",
      "IPMD 0.04772318483398774 3-isopropylmalate dehydrogenase\n",
      "ECOAH7 0.002538636698218338 3-hydroxyacyl-CoA dehydratase (3-hydroxyhexadecanoyl-CoA)\n",
      "IPPS 0.04772318483398774 2-isopropylmalate synthase\n",
      "IZPN_1 0.0 Imidazolonepropionase\n",
      "LIPO2S24_BS 7.077983359343932e-07 Lipoteichoic acid synthesis  n24   linked  N acetylglucosamine substituted\n",
      "LIPO4S24_BS 1.7694958398359574e-06 Lipoteichoic acid synthesis  n24   linked  unsubstituted\n",
      "LIPTabc 0.0 Lipoic acid transport via ABC system\n",
      "IPPMIa -0.04772318483398774 3-isopropylmalate dehydratase\n",
      "IPPMIb -0.04772318483398774 2-isopropylmalate hydratase\n",
      "KARA1 -0.08543066931533524 Ketol-acid reductoisomerase (2,3-dihydroxy-3-methylbutanoate)\n",
      "KARA2 0.0447867654709754 Ketol-acid reductoisomerase (2-Acetolactate)\n",
      "LYSLG_BS 0.00025952605650927745 Lysylphosphatidylglycerol synthesis\n",
      "MALTTRabc 0.0 Maltotriose transport via ABC system\n",
      "MAL_Dt2 0.0 D malate transport in via proton symport\n",
      "MAN1Pt6 0.0 MAN1P transport inout via proton symport\n",
      "MAN6Pt6 0.0 MAN6P transport inout via proton symport\n",
      "KAS1 0.0002538636698218334 B-ketoacyl synthetase (Iso-C14:0)\n",
      "LAAL31 0.0 L amino acid ligase  Ser Tyr \n",
      "ECOAH9ir 1.734723475976807e-18 2 Methylprop 2 enoyl CoA  2 Methylbut 2 enoyl CoA \n",
      "EDA 0.0 2-dehydro-3-deoxy-phosphogluconate aldolase\n",
      "MANAO 0.0 Mannonate oxidoreductase\n",
      "METt3 0.0 L methionine transport out via proton antiport\n",
      "ENO 2.9409448651314127 Enolase\n",
      "MELIBt2 0.0 Melibiose transport in via symport\n",
      "NACt 0.0 Nicotinic acid transport\n",
      "MALt2r 0.0 L malate reversible transport via proton symport\n",
      "ETHAt6 0.0 Ethanolamine transport inout via proton symport\n",
      "MECDPS 0.00021965341691830039 2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase\n",
      "NMNHYD 0.0 Nicotinamide ribonucleotide phosphohydrolase\n",
      "LCYSTabc 0.0 Cysteate transport via ABC system\n",
      "MANt2 0.0 D mannose transport in via proton symport\n",
      "LLEUDr -0.040868865748798236 L leucine dehydrogenase\n",
      "L_LACt2r 0.0 L lactate reversible transport via proton symport\n",
      "MTAN 0.0 Methylthioadenosine nucleosidase\n",
      "MOPSabc 0.0 MOPS transport via ABC system\n",
      "MLTG1 0.0 Maltodextrin glucosidase (maltotriose)\n",
      "NAPRT 0.0 NAPRTase (rev)\n",
      "MLTG2 0.0 Maltodextrin glucosidase (maltotetraose)\n",
      "LAAL32 0.0 L amino acid ligase  Met Tyr \n",
      "LEUTA 0.0 Leucine transaminase\n",
      "METS 0.013368659036350098 Methionine synthase\n",
      "LEUt2r 0.0 L leucine reversible transport via proton symport\n",
      "O2t 5.705899229532262 O2 transport  diffusion \n",
      "LAAL33 0.0 L amino acid ligase  ala trp \n",
      "LACZ 0.0 B-galactosidase\n",
      "LIPO3S24_BS 2.1233950078031893e-06 Lipoteichoic acid synthesis  n24   unliked  D alanine substituted\n",
      "METSR_R1 0.0 L methionine R oxide reductase  trdrd \n",
      "METSR_R2 0.0 L methionine R oxide  H2O2 \n",
      "METSabc 0.0 L methionine S oxide transport via ABC system\n",
      "METDabc 0.0 D-methionine transport via ABC system\n",
      "METabc 0.0 L-methionine transport via ABC system\n",
      "METSR_S1 0.0 L methionine S oxide reductase  trdrd \n",
      "METSR_S2 0.0 L methionine S oxide reductase  H2O2 \n",
      "METTRS 0.0 Methionyl-tRNA synthetase\n",
      "MGSA 0.0 Methylglyoxal synthase\n",
      "LYSAM 0.0 Lysine 2,3-aminomutase\n",
      "OMCDC 0.04772318483398774 2-Oxo-4-methyl-3-carboxypentanoate decarboxylation\n",
      "LCTSabc 0.0 Lactose transport via ABC system (import)\n",
      "MGt5 -0.01199883332353841 Magnesium transport in/out via permease (no H+)\n",
      "PALAt2 0.0 Palatinose transport in via proton symport\n",
      "MMSAD2 0.0 Methylmalonate-semialdehyde dehydrogenase (propanol)\n",
      "MI1PP 0.0 Myo-inositol 1-phosphatase\n",
      "LDH2 0.0 L lactate dehydrogenase  2 Hydroxybutanoic acid \n",
      "LDH_L 0.0 L-lactate dehydrogenase\n",
      "PAP_BS 0.00363336479112992 Phosphatidic acid phosphatase\n",
      "LYSDC 0.0 Lysine decarboxylase\n",
      "MCITL2 0.0 Methylisocitrate lyase\n",
      "PEPt6 0.0 Phosphoenolpyruvate transport in via proton symport\n",
      "MMSAD3 0.0 Methylmalonate-semialdehyde dehydrogenase (malonic semialdehyde)\n",
      "PGL 0.0 6-phosphogluconolactonase\n",
      "MMSDHir -1.734723475976807e-18 Methylmalonate semialdehyde dehydrogenase  acylating \n",
      "MMSAD1 0.0 Methylmalonate semialdehyde dehydrogenase\n",
      "MNNH 0.0 D-mannonate hydrolyase\n",
      "LYSt3r 0.0 L lysine transport out via proton antiport reversible\n",
      "MAN6PI 0.0 Mannose-6-phosphate isomerase\n",
      "M1PD 0.0 Mannitol-1-phosphate 5-dehydrogenase\n",
      "MAL6PG 0.0 Maltose 6  phosphate glucosidase\n",
      "MALT 0.0 Alpha glucosidase\n",
      "MNabc 0.0 Manganese transport via ABC system\n",
      "MNt2 0.0 Manganese transport in via proton symport\n",
      "MOHMT 0.0 3-methyl-2-oxobutanoate hydroxymethyltransferase\n",
      "MPL 0.0 Maltose phosphorylase\n",
      "MOBDabc 0.0 Molybdate transport via ABC system\n",
      "MTHFC 0.05355048853373099 Methenyltetrahydrofolate cyclohydrolase\n",
      "MALTATr 0.0 Maltose O-acetyltransferase\n",
      "MCA 0.0 Methyl cis aconitase\n",
      "MTHFD 0.05355048853373099 Methylenetetrahydrofolate dehydrogenase (NADP)\n",
      "PGPPH_BS 0.0024537008979058847 Phosphatidylglycerophosphate phosphohydrolase\n",
      "MTHFR3 0.013368659036350095 5 10 methylenetetrahydrofolatereductase  NADPH \n",
      "MALTabc 0.0 Maltose transport via ABC system\n",
      "MALTpts 0.0 Maltose transport via PEP:Pyr PTS\n",
      "MALt10 0.0 Malate HNa lactate antiporter\n",
      "MCITS 0.0 2-methylcitrate synthase\n",
      "OXGDC 0.0 2 oxoglutarate decarboxylase\n",
      "MALt4 0.0 Namalate symporter\n",
      "MCSNAH -3.137905955975795e-05 5-Methylcytosine aminohydrolase\n",
      "MTRI 0.0 5-methylthioribose-1-phosphate isomerase\n",
      "MTRK 0.0 5-methylthioribose kinase\n",
      "NADK 0.00013566134772075703 NAD kinase\n",
      "NADK1 0.0 NAD kinase  GTP \n",
      "NADH4 9.538986636237809 NADH dehydrogenase  Menaquinone 7  no proton \n",
      "MDH 1.9221558658499922 Malate dehydrogenase\n",
      "NADK2 0.0 NAD kinase  dATP \n",
      "ORNTAC_1 0.022769990433398623 Ornithine transacetylase\n",
      "NADK3 0.0 NAD kinase  dGTP \n",
      "MDRPD 0.0 5-Methylthio-5-deoxy-D-ribulose 1-phosphate dehydratase\n",
      "MALDDH 0.0 Malate decarboxylating oxidoreductase (decarboxylating)\n",
      "ME2 0.0 Malic enzyme (NADP)\n",
      "NADK4 0.0 NAD kinase  dCTP \n",
      "BGLA1 0.0 6-phospho-beta-glucosidase\n",
      "NADK5 0.0 NAD kinase  dTTP \n",
      "NADS1 0.0020424700647280296 NAD synthase (nh3)\n",
      "NAt3_1 0.0 Sodium proton antiporter  HNA is 11 \n",
      "NAabcO 0.0 Sodium transport out via ABC system\n",
      "MECDPDH2 0.00021965341691830039 2C-methyl-D-erythritol 2,4 cyclodiphosphate dehydratase\n",
      "MEPCT 0.00021965341691830039 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase\n",
      "METAT 0.0 Methionine adenosyltransferase\n",
      "NDPK7 0.0020502558464232804 Nucleoside-diphosphate kinase (ATP:dCDP)\n",
      "PENCAC 0.0 Penicillin acylase\n",
      "NDPK3 0.06148467194678227 Nucleoside-diphosphate kinase (ATP:CDP)\n",
      "NDPK4 0.002701312349093596 Nucleoside-diphosphate kinase (ATP:dTDP)\n",
      "PAPSR 0.0 Phosphoadenylyl-sulfate reductase (thioredoxin)\n",
      "NDPK5 0.0020523792414310835 Nucleoside-diphosphate kinase (ATP:dGDP)\n",
      "PMDPHT 0.0 Pyrimidine phosphatase\n",
      "ORNt2 0.0 Ornithine transport in via proton symport\n",
      "PPAt6 0.0 Propionate transport inout via proton symport\n",
      "PPIabc 0.0 Diphosphate transport in via ABC system\n",
      "NDPK6 0.0 Nucleoside-diphosphate kinase (ATP:dUDP)\n",
      "NDPK1 0.02899708222655637 Nucleoside-diphosphate kinase (ATP:GDP)\n",
      "NDPK2 0.07991031416060332 Nucleoside-diphosphate kinase (ATP:UDP)\n",
      "PPTGS_BS 0.012010983861638616 Peptidoglycan subunit synthesis\n",
      "PGMT_B 0.0 B phosphoglucomutase\n",
      "PPTT 3.137905955975795e-05 Trans pentaprenyltranstransferase\n",
      "PSER_Dt6 0.0 D O Phosphoserine transport inout via proton symport\n",
      "PSER_Lt6 0.0 O Phospho L serine transport in via proton symport\n",
      "RAFFINt2 0.0 Raffinose transport in via proton symport\n",
      "PGI 1.6703178869506379 Glucose-6-phosphate isomerase\n",
      "RHCYS 0.0 5 deoxyribos 5 ylhomocysteinase\n",
      "P5CD 0.0 1-pyrroline-5-carboxylate dehydrogenase\n",
      "RIBFLVt2 0.0 Riboflavin transport in via proton symport\n",
      "NDPK9 0.0 Nucleoside diphosphate kinase  ATPIDP \n",
      "RMNt 0.0 L-rhamnose transport via proton symport\n",
      "PYRt2 0.0 Pyruvate transport in via proton symport\n",
      "PYNP1 3.137905955975795e-05 Pyrimidine-nucleoside phosphorylase (cytosine)\n",
      "PHETA1 -0.02075488857099342 Phenylalanine transaminase\n",
      "NDPK8 0.0027111035594073545 Nucleoside-diphosphate kinase (ATP:dADP)\n",
      "PYNP2r 0.0 Pyrimidine-nucleoside phosphorylase (uracil)\n",
      "RMPA 0.0 Rhamnulose-1-phosphate aldolase\n",
      "SALADC2 0.0 Sulfinoalanine decarboxylase\n",
      "NH4t 0.9449636274148077 Ammonia reversible transport\n",
      "DSERt2 0.0 D-serine transport via proton symport\n",
      "SERt2r 0.0 L serine reversible transport via proton symport\n",
      "PTAr -0.050384624543489735 Phosphotransacetylase\n",
      "PMPK 0.0 Phosphomethylpyrimidine kinase\n",
      "SORD_D 0.0 D sorbitol dehydrogenase\n",
      "NMNAT 0.0 Nicotinamide-nucleotide adenylyltransferase\n",
      "SRB_Lt 0.0 L sorbose reversible transport\n",
      "PTA2 0.0 Phosphate acetyltransferase\n",
      "PHYT3 0.0 3 phytase\n",
      "NNATr 0.0020424700647280296 Nicotinate-nucleotide adenylyltransferase\n",
      "PIabc 0.0 Phosphate transport via ABC system\n",
      "SO4t2 0.02009156356383357 Sulfate transport in via proton symport\n",
      "NNAM 0.0 Nicotinamidase\n",
      "SARCOX 0.0 Sarcosine oxidase\n",
      "PIt2r 0.1810531628025574 Phosphate reversible transport via symport\n",
      "PIt7 0.0 Phosphate transport in/out via three Na+ symporter\n",
      "SUCR 0.0 Sucrose hydrolyzing enzyme\n",
      "NNDPR 0.0020424700647280296 Nicotinate-nucleotide diphosphorylase (carboxylating)\n",
      "PMANM 0.0 Phosphomannomutase\n",
      "SALCpts 0.0 Salicin transport via PEP:Pyr PTS\n",
      "NO2t2r 0.0 Nitrite transport in via proton symport, reversible\n",
      "NPHS 3.137905955975795e-05 Naphthoate synthase\n",
      "RBK 0.0 Ribokinase\n",
      "NO2t3 0.0 Nitrite transport out via proton antiport\n",
      "PNTEH 0.0 Hydrolase Class   RXN R02973 \n",
      "NO3t2 0.0 Nitrate transport in via proton symport\n",
      "PNTK 0.0 Pantothenate kinase\n",
      "RBK_Dr 0.0 D-ribulokinase (D-ribulose)\n",
      "RIBabc 0.0 D-ribose transport via ABC system\n",
      "NTR3B 0.0 Nitrate reductase\n",
      "SUCCt2r 0.0 Succinate transport via proton symport\n",
      "Q23DO 0.0 Quercetin 2 3 dioxygenase\n",
      "PNP 0.0 Purine nucleoside phosphorylase\n",
      "QULNS 0.0020424700647280296 Quinolinate synthase\n",
      "RBFK 0.0 Riboflavin kinase\n",
      "APNPT 0.0 ATP:pantetheine 4'-phosphotransferase\n",
      "PNTOt2 0.0 Pantothenate reversible transport via proton symport\n",
      "PPA 0.3075311810137445 Inorganic diphosphatase\n",
      "PPBNGS 0.0 Porphobilinogen synthase\n",
      "PPNCL 0.0 Phosphopantothenate-cysteine ligase\n",
      "RBK_L1 0.0 L-ribulokinase (L-ribulose)\n",
      "RBFSa 0.0 Riboflavin synthase\n",
      "PPCDC 0.0 Phosphopantothenoylcysteine decarboxylase\n",
      "PPCK 0.0 Phosphoenolpyruvate carboxykinase\n",
      "OCBT_1 0.022769990433398623 Ornithine carbamoyltransferase\n",
      "PPM -3.137905955975795e-05 Phosphopentomutase\n",
      "PPCOAC 0.0 Propionyl-CoA carboxylase\n",
      "RBFSb 0.0 Riboflavin synthase\n",
      "RBP4E 0.0 L-ribulose-phosphate 4-epimerase\n",
      "RMI 0.0 L-rhamnose isomerase\n",
      "RMK 0.0 Rhamnulokinase\n",
      "PPM2 -0.002021000181871325 Phosphopentomutase 2 (deoxyribose)\n",
      "PPND 0.013073507130265458 Prephenate dehydrogenase\n",
      "RNDR4 0.002732691408653353 Ribonucleoside-diphosphate reductase (UDP)\n",
      "NTRIR2x 0.0 Nitrite Reductase (NADH)\n",
      "RPE -0.048071146360355727 Ribulose 5-phosphate 3-epimerase\n",
      "RPI -0.048071146360355727 Ribose-5-phosphate isomerase\n",
      "SADT 0.02009156356383357 Sulfate adenylyltransferase\n",
      "PPNDH 0.02075488857099342 Prephenate dehydratase\n",
      "PPPGO 0.0 Protoporphyrinogen oxidase (aerobic)\n",
      "PPS 0.0 Phosphoenolpyruvate synthase\n",
      "PRAGSr 0.021487577815071147 Phosphoribosylglycinamide synthase\n",
      "PRAIi 0.00640982173022183 Phosphoribosylanthranilate isomerase (irreversible)\n",
      "SBTD_D2 0.0 D sorbitol dehydrogenase  D fructose producing \n",
      "SBTt2 0.0 D sorbitol transport in via proton symport\n",
      "SDPDS 0.0 Succinyl-diaminopimelate desuccinylase\n",
      "SERAT 0.0 Serine O-acetyltransferase\n",
      "SERD_D 0.0 D-serine deaminase\n",
      "SHCHD2_1 0.0 Sirohydrochlorin dehydrogenase  NAD \n",
      "PRAIS_1 0.021487577815071147 Phosphoribosylaminoimidazole synthetase\n",
      "PRATPP 0.009642454696823507 Phosphoribosyl-ATP pyrophosphatase\n",
      "PRASCSi 0.021487577815071147 Phosphoribosylaminoimidazolesuccinocarboxamide synthase\n",
      "PRFGS_1 0.021487577815071147 Phosphoribosylformylglycinamidine synthase\n",
      "PRMICI 0.009642454696823507 1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino)imidazole-4-carboxamide isomerase\n",
      "SERD_L 0.0 L-serine deaminase\n",
      "SHCHF_2 0.0 Sirohydrochlorin ferrochetalase\n",
      "OMPDC 0.014388950337599519 Orotidine-5'-phosphate decarboxylase\n",
      "SHK3Dr 0.040312890155921724 Shikimate dehydrogenase\n",
      "SHKK 0.040312890155921724 Shikimate kinase\n",
      "SHSL1 0.0 O-succinylhomoserine lyase (L-cysteine)\n",
      "URATEt_1 0.0 Urate permease\n",
      "PRO1x 0.0 Proline oxidase  L proline  NAD \n",
      "PRPPS 0.05191889540301295 Phosphoribosylpyrophosphate synthetase\n",
      "SHSL2 0.02009156356383357 O succinylhomoserine lyase  H2S \n",
      "OCOAT1 0.0 3-oxoacid CoA-transferase (Succinyl-CoA: acetoacetate)\n",
      "SHSL3 0.0 O succinylhomoserine lyase  methanethiol \n",
      "SHSL4r 0.03806386094349192 O succinylhomoserine lyase  elimination   reversible\n",
      "PROt2r 0.0 L proline reversible transport via proton symport\n",
      "PSCVT 0.040312890155921724 3-phosphoshikimate 1-carboxyvinyltransferase\n",
      "PSDC_BS 0.0066061178020543135 Phosphatidylserine decarboxylase\n",
      "SLCYSS 0.0 O-acetyl-L-serine sulfhydrylase\n",
      "ORNTA_1 0.0 Ornithine transaminase\n",
      "SPODM 0.0 Superoxide dismutase\n",
      "PSERT 0.11489654997305775 Phosphoserine transaminase\n",
      "PSP_D 0.0 Phosphoserine phosphatase  D serine \n",
      "PTHPS 0.0 6-pyruvoyltetrahydropterin synthase\n",
      "SPMDAT1 0.0 Spermidine acetyltransferase\n",
      "ORPT -0.014388950337599519 Orotate phosphoribosyltransferase\n",
      "SPMDAT3 0.0 Spermidine acetyltransferase  spermine \n",
      "SPMDt3i 0.0 Spermidine transport in via proton antiport\n",
      "SPRMS 0.0 Spermine synthase\n",
      "SPMS 0.0 Spermidine synthase\n",
      "PTPATi 0.0 Pantetheine-phosphate adenylyltransferase\n",
      "PYK 1.148308100957936 Pyruvate kinase\n",
      "PUNP5 0.0 Purine-nucleoside phosphorylase (Inosine)\n",
      "PUNP6 0.0 Purine-nucleoside phosphorylase (Deoxyinosine)\n",
      "PURabc1 0.0 Puromycin efflux  via ABC system\n",
      "PYRIS 0.0 Pyridoxal 5  phsophate synthase\n",
      "OIVD1r 0.006854319085189502 2-oxoisovalerate dehydrogenase (acylating; 4-methyl-2-oxopentaoate)\n",
      "EX_f6p_e 0.0 D-fructose 6-phosphate exchange\n",
      "SSALy 0.0 Succinate-semialdehyde dehydrogenase (NADP)\n",
      "SUCBZL 3.137905955969026e-05 O-succinylbenzoate-CoA ligase\n",
      "SUCBZS 3.137905955969026e-05 O-succinylbenzoate-CoA synthase\n",
      "TGBPA 0.0 Tagatose-bisphosphate aldolase\n",
      "SUCD1 1.8563371087595062 Succinate dehydrogenase\n",
      "OIVD2 0.001523182018930999 2-oxoisovalerate dehydrogenase (acylating; 3-methyl-2-oxobutanoate)\n",
      "SUCD5_BS 1.8563371087595062 Succinate dehydrogenase\n",
      "SUCOAS -1.7981816842521807 Succinyl-CoA synthetase (ADP-forming)\n",
      "THIOGabc 0.0 Thioglycolate transport via ABC system\n",
      "THMt6 0.0 Thymine transport inout via proton symport\n",
      "OIVD3 0.012947047160913506 2-oxoisovalerate dehydrogenase (acylating; 3-methyl-2-oxopentanoate)\n",
      "THRt2r 0.0 L threonine reversible transport via proton symport\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P5CR 0.018950356713528693 Pyrroline-5-carboxylate reductase\n",
      "TMPHi 0.0 Trimetaphosphate hydrolase\n",
      "UREAt 0.0 Urea transport via facilitate diffusion\n",
      "TMPt6 0.0 TMP transport inout via proton symport\n",
      "OXADC 6.776263578034403e-21 Oxalate decarboxylase\n",
      "TYRt2r 0.0 L tyrosine reversible transport via proton symport\n",
      "UGLYCH 0.0 Ureidoglycolate hydrolase\n",
      "UMPt6 0.0 UMP transport in via proton symport\n",
      "SULAabc 0.0 Sulfoacetate transport via ABC system\n",
      "TAURabc 0.0 Taurine transport via ABC system\n",
      "TRPAS1 0.0 Tryptophanase  L cysteine \n",
      "THMDt2 0.0 Thymidine transport in via proton symport\n",
      "TMDK1 0.0 Thymidine kinase (ATP:thymidine)\n",
      "TMDPP 3.137905955975795e-05 Thymidine phosphorylase\n",
      "UAGDP 0.03857194218230186 UDP-N-acetylglucosamine diphosphorylase\n",
      "TARTDC 0.0 Tartrate decarboxylase\n",
      "TYRTA -0.013073507130265458 Tyrosine transaminase\n",
      "TECA1S45 0.0004275101949043718 Glycerol teichoic acid  n45   unlinked  unsubstituted\n",
      "TARTRDH 0.0 Tartrate dehydrogenase\n",
      "TAGURr 0.0 Tagaturonate reductase\n",
      "SULR_1 -0.02009156356383357 Sulfite reductase  NADPH2 \n",
      "TALA -0.003879128102217004 Transaldolase\n",
      "THRD 0.0 L-threonine dehydrogenase\n",
      "TECA4S_BS 0.0003671114035713038 Minor teichoic acid synthesis  n30 \n",
      "TEICH45 3.854941057726238e-20 Teichuronic acid  n45   unlinked  GalNAc GlcA repeated\n",
      "THRD_L 0.0 L-threonine deaminase\n",
      "THRS 0.021979143759381266 Threonine synthase\n",
      "TKT1 -0.003879128102217004 Transketolase\n",
      "TKT2 -0.044192018258138725 Transketolase\n",
      "TMN 0.0 Thiaminase\n",
      "TMDS 0.002732691408653354 Thymidylate synthase\n",
      "TMPK 0.0 Thiamine-phosphate kinase\n",
      "TMPPP 0.0 Thiamine-phosphate diphosphorylase\n",
      "TPI 1.5121678199907547 Triose-phosphate isomerase\n",
      "TRDR 0.027585614378317556 Thioredoxin reductase (NADPH)\n",
      "TRE6PH 0.0 Trehalose-6-phosphate hydrolase\n",
      "TRPt2r 0.0 L tryptophan reversible transport via proton symport\n",
      "TRPS1 0.00640982173022183 Tryptophan synthase (indoleglycerol phosphate)\n",
      "UAAGDS 0.012010983861638616 UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminopimelate synthetase\n",
      "UAG2E 0.0009189581728214837 UDP-N-acetylglucosamine 2-epimerase\n",
      "UAG2EMA 0.0 UDP-N-acetyl-D-glucosamine 2-epimerase (Hydrolysis)\n",
      "UDPG4E 0.0 UDPglucose 4-epimerase\n",
      "UAGPT3 0.012010983861638616 UDP-N-acetylglucosamine-N-acetylmuramyl-(pentapeptide)pyrophosphoryl-undecaprenol N-acetylglucosamine transferase\n",
      "UAG4E 0.011013342107139116 UDP N acetylglucosamine 4 epimerase\n",
      "UAGCVT 0.012010983861638616 UDP-N-acetylglucosamine 1-carboxyvinyltransferase\n",
      "UAMAGS 0.012010983861638616 UDP-N-acetylmuramoyl-L-alanyl-D-glutamate synthetase\n",
      "UAMAS 0.012010983861638616 UDP-N-acetylmuramoyl-L-alanine synthetase\n",
      "UAPGR 0.012010983861638616 UDP-N-acetylenolpyruvoylglucosamine reductase\n",
      "UDCPDP 0.012010983861638616 Undecaprenyl-diphosphatase\n",
      "UDPGD 1.734723475976807e-18 UDPglucose 6-dehydrogenase\n",
      "UGMDDS 0.012010983861638616 UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminopimeloyl-D-alanyl-D-alanine synthetase\n",
      "UGT1_BS 0.0010145109481726282 UDP glucosyltransferase  monoglucosyl \n",
      "UGT2_BS 0.0007785781695278321 UDP glucosyltransferase  triglucosyl \n",
      "UGT_BS 0.0018402756734294155 UDP glucosyltransferase  diglucosyl \n",
      "UNK5 0.0 2 keto 4 methylthiobutyrate transamination\n",
      "UPP3S 0.0 Uroporphyrinogen-III synthase\n",
      "UPP3MT_2 0.0 Uroporphyrinogen methyltransferase\n",
      "ALDD9 0.0 Aldehyde dehydrogenase  nonanal  NAD \n",
      "FRUpts 0.0 D-fructose transport via PEP:Pyr PTS\n",
      "CYTDt2 0.0 Cytidine transport in via proton symport\n",
      "GLYBabc 0.0 Glycine betaine transport via ABC system\n",
      "ACLS 0.08745166949720656 Acetolactate synthase\n",
      "ACHBS 0.0447867654709754 2-aceto-2-hydroxybutanoate synthase\n",
      "X00002 0.0 Lumped reaction for synthesis of pime\n",
      "NTD8 0.0 5'-nucleotidase (dGMP)\n",
      "23CN2P1 0.0 2  3  cyclic nucleotide 2  phosphodiesterase  2  3  cyclc AMP \n",
      "23CN2P2 0.0 2  3  cyclic nucleotide 2  phosphodiesterase  2  3  cyclic UMP \n",
      "23CN2P3 0.0 2  3  cyclic nucleotide 2  phosphodiesterase  2  3  cyclic CMP \n",
      "23CN2P4 0.0 2  3  cyclic nucleotide 2  phosphodiesterase  2  3   cyclic GMP \n",
      "NTD6 0.0 5'-nucleotidase (dAMP)\n",
      "NTD9 0.0 5'-nucleotidase (GMP)\n",
      "NTD1 0.0 5'-nucleotidase (dUMP)\n",
      "NTD10 0.0 5'-nucleotidase (XMP)\n",
      "NTD2 0.0 5'-nucleotidase (UMP)\n",
      "3NUCLE1 0.0 3  nucleotidase  3  AMP \n",
      "3NUCLE2 0.0 3  nucleotidase  3  UMP \n",
      "3NUCLE3 0.0 3  nucleotidase  3  cMP \n",
      "3NUCLE4 0.0 3  nucleotidase  guanosine 3  phosphate \n",
      "NTD3 0.0 5'-nucleotidase (dCMP)\n",
      "NTD4 0.0 5'-nucleotidase (CMP)\n",
      "NTD5 3.137905955975795e-05 5'-nucleotidase (dTMP)\n",
      "NTD7 0.0 5'-nucleotidase (AMP)\n",
      "ABUTt2r 0.0 4 aminobutyrate reversible transport in via proton symport\n",
      "BALAt2r 0.0 Beta Alanine transport in via proton symport\n",
      "S6PG 0.0 Salicin 6 phosphate glucohydrolase\n",
      "URIK3 0.0 Uridine kinase (ITP:Uridine)\n",
      "LCADi 0.0 Lactaldehyde dehydrogenase\n",
      "XTSNt2 0.0 Xanthosine transport in via proton symport\n",
      "XYLt2 0.0 D-xylose transport in via proton symport\n",
      "ZN2t4 0.0 Zinc transport out via antiport\n",
      "URIK1 0.0 Uridine kinase  ATPUridine \n",
      "URIK2 0.0 Uridine kinase (GTP:Uridine)\n",
      "URIt2 0.0 Uridine transport in via proton symport\n",
      "UPPRT 0.0 Uracil phosphoribosyltransferase\n",
      "XPPT 0.0 Xanthine phosphoribosyltransferase\n",
      "HXAND 0.0 Hypoxanthine dehydrogenase\n",
      "URIDK2r_copy2 -0.002732691408653353 Uridylate kinase (dUMP)\n",
      "VALDHr -0.7156686995308021 Valine dehydrogenase\n",
      "VALTA 0.6794843970683857 Valine transaminase\n",
      "VALt2r 0.0 L valine reversible transport via proton symport\n",
      "UMPK 0.038332234479198736 UMP kinase\n",
      "XYL14BASE 0.0 Xylan 1 4 beta xylosidase\n",
      "UPPDC1 0.0 Uroporphyrinogen decarboxylase (uroporphyrinogen III)\n",
      "URAt2 0.0 Uracil transport in via proton symport\n",
      "URCN 0.0 Urocanase\n",
      "URIDK2r_copy1 0.0 Uridylate kinase (dUMP)\n",
      "UREA 0.0 Urease\n",
      "XAND 0.0 Xanthine dehydrogenase\n",
      "URIC 0.0 Uricase\n",
      "XYLI1 0.0 Xylose isomerase\n",
      "XANt2 0.0 Xanthine transport in via proton symport\n",
      "XYLI2 0.0 Xylose isomerase\n",
      "XYLK 0.0 Xylulokinase\n",
      "ZNabc 0.0 Zinc transport via ABC system\n",
      "ALDD10 0.0 Aldehyde dehydrogenase  decanal  NAD \n",
      "ALDD11 0.0 Aldehyde dehydrogenase  undecanal  NAD \n",
      "ALDD12 0.0 Aldehyde dehydrogenase  dodecanal  NAD \n",
      "ALDD13 0.0 Aldehyde dehydrogenase  tridecanal  NAD \n",
      "ALDD14 0.0 Aldehyde dehydrogenase  tetradecanal  NAD \n",
      "ALDD15 0.0 Aldehyde dehydrogenase  pentadecanal  NAD \n",
      "ALDD16 0.0 Aldehyde dehydrogenase  hexadecanal  NAD \n",
      "ALDD17 0.0 Aldehyde dehydrogenase  heptadecanal  NAD \n",
      "ALDD18 0.0 Aldehyde dehydrogenase  octadecanal  NAD \n",
      "ALDD1 0.0 Aldehyde dehydrogenase  formaldehyde  NAD \n",
      "ALDD20x_1 0.0 Aldehyde dehydrogenase  NAD \n",
      "ALDD2x 0.0 Aldehyde dehydrogenase (acetaldehyde, NAD)\n",
      "ALDD3 0.0 Aldehyde dehydrogenase  propanal  NAD \n",
      "GLXO1 0.0 Glyoxylate oxidase\n",
      "ALDD31_1 0.0 Aldehyde dehydrogenase  aminoacetaldehyde  NAD \n",
      "ALDD32 0.0 Aldehyde dehydrogenase  D Glucuronolactone  NADH \n",
      "ALDD4 0.0 Aldehyde dehydrogenase (butanal, NAD)\n",
      "ALDD4x 0.0 Aldehyde dehydrogenase  methylmalonate semialdehyde  NAD \n",
      "ALDD5 0.0 Aldehyde dehydrogenase  pentanal  NAD \n",
      "ALDD6 0.0 Aldehyde dehydrogenase  hexanal  NAD \n",
      "ALDD7 0.0 Aldehyde dehydrogenase  heptanal  NAD \n",
      "ALDD8 0.0 Aldehyde dehydrogenase  octanal  NAD \n",
      "GLYALDDr 0.0 D-Glyceraldehyde dehydrogenase\n",
      "DIPEPabc12 0.0 Dipeptide transport via ABC system  met ala \n",
      "DIPEPabc13 0.0 Dipeptide transport via ABC system  gly asp \n",
      "DIPEPabc14 0.0 Dipeptide transport via ABC system  gly pro L \n",
      "DIPEPabc15 0.0 Dipeptide transport via ABC system  cgly \n",
      "DIPEPabc2 0.0 Dipeptide transport via ABC system  ala gln \n",
      "DIPEPabc3 0.0 Dipeptide transport via ABC system  ala glu \n",
      "DIPEPabc4 0.0 Dipeptide transport via ABC system  ala gly \n",
      "DIPEPabc5 0.0 Dipeptide transport via ABC system  ala his \n",
      "DIPEPabc6 0.0 Dipeptide transport via ABC system  ala leu \n",
      "DIPEPabc7 0.0 Dipeptide transport via ABC system  ala thr \n",
      "DIPEPabc8 0.0 Dipeptide transport via ABC system  gly asn \n",
      "DIPEPabc9 0.0 Dipeptide transport via ABC system  gly gln \n",
      "RIBabc1 0.0 D ribose transport out via ABC system\n",
      "FRUK 0.0 Fructose-1-phosphate kinase\n",
      "FACOAL140 0.0 Fatty acid  CoA ligase  tetradecanoate \n",
      "FACOAL140_ISO 0.0002538636698218334 Fatty acid  CoA ligase  Iso C140 \n",
      "FACOAL150_ISO 0.005077273396436668 Fatty acid  CoA ligase  Iso C150 \n",
      "PPAKr 0.0 Propionate kinase\n",
      "ACOAD3 -0.003300227707683838 Acyl-CoA dehydrogenase (octanoyl-CoA)\n",
      "ACOAD4_1 -0.003300227707683838 Acyl CoA dehydrogenase  decanoyl CoA \n",
      "ACOAD5_1 -0.003300227707683838 Acyl CoA dehydrogenase  dodecanoyl CoA \n",
      "ACOAD6 -0.003300227707683838 Acyl-CoA dehydrogenase (tetradecanoyl-CoA)\n",
      "ACOAD7 -0.002538636698218338 Acyl-CoA dehydrogenase (hexadecanoyl-CoA)\n",
      "DADNK 0.0 Deoxyadenosine kinase\n",
      "CSNt2 0.0 Cytosine transport in via proton symport\n",
      "INSt2 0.0 Inosine transport in via proton symport\n",
      "DHAD3 0.03183971831006189 Dihydroxy acid dehydratase  2 3 dihydroxy 3 methylpentanoate \n",
      "CELBpts 0.0 Cellobiose transport via PEPPyr PTS\n",
      "MSO3abc 0.0 Methanesulfonate transport via ABC system\n",
      "ISTNTabc 0.0 Isethionate transport via ABC system\n",
      "CTBTabc 0.0 Crotonobetaine transport in via ABC system\n",
      "CHLabc 0.0 Choline transport via ABC system\n",
      "CITt15 0.0 Citrate transport in via Zn complex\n",
      "CHOLSabc_1 0.0 Choline sulfate transport in via ABC system\n",
      "ECTOINEabc 0.0 Ecotine transport in via ABC system\n",
      "ITCY 0.0 ITP:cytidine 5'-phosphotransferase\n",
      "PROLBabc 0.0 Proline betaine transport in via ABC system\n",
      "GLCpts 1.699999999999993 D-glucose transport via PEP:Pyr PTS\n",
      "MANpts 0.0 D-mannose transport via PEP:Pyr PTS\n",
      "MNLpts 0.0 Mannitol transport via PEP:Pyr PTS\n",
      "DIPEPabc10 0.0 Dipeptide transport via ABC system  gly glu \n",
      "DIPEPabc11 0.0 Dipeptide transport via ABC system  gly met \n",
      "SUCpts 0.0 Sucrose transport via PEP:Pyr PTS\n",
      "TREpts 0.0 Trehalose transport via PEP:Pyr PTS\n",
      "FERRICHabc 0.0 Ferrichrome transport via ABC system\n",
      "GPDDA2 0.0 Glycerophosphodiester phosphodiesterase (Glycerophosphoethanolamine)\n",
      "HACD3 0.003300227707683838 3-hydroxyacyl-CoA dehydrogenase (3-oxooctanoyl-CoA)\n",
      "HACD4 0.003300227707683838 3-hydroxyacyl-CoA dehydrogenase (3-oxodecanoyl-CoA)\n",
      "HACD5 0.003300227707683838 3-hydroxyacyl-CoA dehydrogenase (3-oxododecanoyl-CoA)\n",
      "HACD6 0.003300227707683838 3-hydroxyacyl-CoA dehydrogenase (3-oxotetradecanoyl-CoA)\n",
      "HACD7 0.002538636698218338 3-hydroxyacyl-CoA dehydrogenase (3-oxohexadecanoyl-CoA)\n",
      "HACD9 1.734723475976807e-18 3 hydroxyacyl CoA dehydrogenase  2 Methylacetoacetyl CoA \n",
      "HACOADr -9.76680237544791e-19 3 hydroxyacyl CoA dehydrogenase\n",
      "ICHORS 3.137905955969026e-05 Isochorismate synthase\n",
      "INSCR 0.0 Inositol catabolic reactions  lumped \n",
      "FACOAL161_ISO 0.0 Fatty acid  CoA ligase  Iso C161 \n",
      "FACOAL170_ISO 0.0017770456887528337 Fatty acid  CoA ligase  Iso C170 \n",
      "FACOAL170_anteiso 0.004315682386971168 Fatty acid  CoA ligase  anteiso C170 \n",
      "FACOAL171_ISO 0.0 Fatty acid  CoA ligase  Iso C171 \n",
      "FACOAL171_anteiso 0.0 Fatty acid  CoA ligase  anteiso C171 \n",
      "FACOAL180_2 0.0007615910094654997 Fatty acid  CoA ligase  octadecanoate \n",
      "FACOAL181 0.0 Fatty acid  CoA ligase  octadecenoate \n",
      "KAS13 0.0007615910094655002 B-ketoacyl synthetase (octadecanoate)\n",
      "KAS17 0.0 B-ketoacyl synthetase (n-C18:1)\n",
      "FACOAL150_anteiso 0.008631364773942336 Fatty acid  CoA ligase  anteiso C150 \n",
      "FACOAL160 -1.0634514948460414e-18 Fatty acid  CoA ligase  hexadecanoate \n",
      "FACOAL160_ISO 0.0012693183491091667 Fatty acid  CoA ligase  Iso C160 \n",
      "FACOAL161 0.0 Fatty acid  CoA ligase  hexadecenoate \n",
      "KAS2 0.0 B-ketoacyl synthetase (n-C14:0)\n",
      "KAS3 0.005077273396436668 B-ketoacyl synthetase (Iso-C15:0)\n",
      "KAS4 0.008631364773942336 B-ketoacyl synthetase (Anteiso-C15:0)\n",
      "KAS5 0.0 B-ketoacyl synthetase (Iso-C16:1)\n",
      "KAS6 0.0012693183491091667 B-ketoacyl synthetase (Iso-C16:0)\n",
      "KAS7 0.0 B-ketoacyl synthetase (n-C16:1)\n",
      "KAS8 0.0 B-ketoacyl synthetase (palmitate, n-C16:0)\n",
      "KAS9 0.0 B-ketoacyl synthetase (Iso-C17:1)\n",
      "KAS10 0.0 B-ketoacyl synthetase (Anteiso-C17:1)\n",
      "KAS11 0.0017770456887528337 B-ketoacyl synthetase (Iso-C17:0)\n",
      "PUNP7 0.0 Purine-nucleoside phosphorylase (Xanthosine)\n",
      "PUNP8 0.0 Purine nucleoside phosphorylase  nebularine \n",
      "PUNP1 0.0 Purine-nucleoside phosphorylase (Adenosine)\n",
      "PUNP2 0.0 Purine-nucleoside phosphorylase (Deoxyadenosine)\n",
      "KAS12 0.004315682386971168 B-ketoacyl synthetase (Anteiso-C17:0)\n",
      "LAAL34 0.0 L amino acid ligase  ser trp \n",
      "LAAL35 0.0 L amino acid ligase  gly gln \n",
      "LAAL36 0.0 L amino acid ligase  ala gln \n",
      "LAAL37 0.0 L amino acid ligase  ser gln \n",
      "LAAL38 0.0 L amino acid ligase  thr gln \n",
      "LAAL39 0.0 L amino acid ligase  ala asn \n",
      "PANTS 0.0 Pantothenate synthase\n",
      "LAAL4 0.0 L amino acid ligase  thr gly \n",
      "LAAL40 0.0 L amino acid ligase  ala his \n",
      "LAAL41 0.0 L amino acid ligase  ser his \n",
      "LAAL42 0.0 L amino acid ligase  met his \n",
      "PAPPT3 0.012010983861638616 Phospho-N-acetylmuramoyl-pentapeptide-transferase (meso-2,6-diaminopimelate)\n",
      "LAAL5 0.0 L amino acid ligase  Ala Ala \n",
      "LAAL6 0.0 L amino acid ligase  Ser Ala \n",
      "PC 0.29710141863458384 Pyruvate carboxylase\n",
      "LAAL7 0.0 L amino acid ligase  Ala Ser \n",
      "LAAL8 0.0 L amino acid ligase  Ser Ser \n",
      "LAAL9 0.0 L amino acid ligase  Gly Cys \n",
      "PBUTT 0.0 Phosphate butyryltransferase\n",
      "MLTG3 0.0 Maltodextrin glucosidase (maltopentaose)\n",
      "MLTG4 0.0 Maltodextrin glucosidase (maltohexaose)\n",
      "PDH 2.202323209297355 Pyruvate dehydrogenase\n",
      "MLTG5 0.0 Maltodextrin glucosidase (maltoheptaose)\n",
      "MLTG6 0.0 Maltodextrin glucosidase  dextrin \n",
      "MLTG7 0.0 Maltodextrin glucosidase  isomaltose \n",
      "PFK_2 0.0 Phosphofructokinase\n",
      "TARTRDH2 0.0 Tartrate dehydrogenase\n",
      "PUNP3 0.0 Purine-nucleoside phosphorylase (Guanosine)\n",
      "PUNP4 -0.0020523792414310835 Purine-nucleoside phosphorylase (Deoxyguanosine)\n",
      "PFK 1.5836747984079802 Phosphofructokinase\n",
      "METSRabc 0.0 L methionine R oxide transport via ABC system\n",
      "PRAMPC_1 0.009642454696823507 Phosphoribosyl AMP cyclohydrolase\n",
      "PSP_L 0.11489654997305775 Phosphoserine phosphatase (L-serine)\n",
      "PGAMT -0.03857194218230186 Phosphoglucosamine mutase\n",
      "TECA2S45 0.0002768671157396692 Glycerol teichoic acid  n45   unlinked  D ala substituted\n",
      "RNDR1 0.0027111035594073554 Ribonucleoside-diphosphate reductase (ADP)\n",
      "PGMT -0.029682113049355294 Phosphoglucomutase\n",
      "RNDR2 0.0 Ribonucleoside-diphosphate reductase (GDP)\n",
      "PGCD 0.11489654997305775 Phosphoglycerate dehydrogenase\n",
      "RNDR3 0.0020502558464232804 Ribonucleoside-diphosphate reductase (CDP)\n",
      "TECA3S45 0.0002145808621774427 Glycerol teichoic acid  n45   unlinked  glucose substituted\n",
      "GND 0.0 Phosphogluconate dehydrogenase\n",
      "PGK -3.0558197682720296 Phosphoglycerate kinase\n",
      "PGLYCP 0.0 Phosphoglycolate phosphatase\n",
      "PGM -2.9409448651314127 Phosphoglycerate mutase\n",
      "PHCYT_BS 0.009059818699961742 Phosphatidate cytidylyltransferase\n"
     ]
    }
   ],
   "source": [
    "for reaction in model.reactions:\n",
    "    print(reaction.id, reaction.flux, reaction.name)  # Para observar ID, valor de flujo y nombre de cada reacción del modelo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d47f42d1",
   "metadata": {},
   "source": [
    "# umax"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5bb4344d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.1179663893223982"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.reactions.BIOMASS_BS_10.flux # Para mostrar velocidad de crecimiento de modelo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b63aa909",
   "metadata": {},
   "source": [
    "The growth rate is 0.11796 $h^{-1}$."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "13166575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.7"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "-1 * model.reactions.EX_glc__D_e.flux # Para mostrar el flujo de glucosa de intercambio de modelo"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0cabf323",
   "metadata": {},
   "source": [
    "# YX/S"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "272c6284",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de biomasa: 0.06939199371905777 mol biomasa/mol glucosa\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de biomasa en glucosa\n",
    "print('Rendimiento de biomasa:', solution.fluxes['BIOMASS_BS_10'] / (-1 * solution.fluxes['EX_glc__D_e']), 'mol biomasa/mol glucosa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8e5adfb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fluxes</th>\n",
       "      <th>reduced_costs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EX_fe3_e</th>\n",
       "      <td>-0.000407</td>\n",
       "      <td>1.075735e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_glc__D_e</th>\n",
       "      <td>-1.700000</td>\n",
       "      <td>-1.780625e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_ca2_e</th>\n",
       "      <td>-0.000378</td>\n",
       "      <td>-3.108624e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_so4_e</th>\n",
       "      <td>-0.020092</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_co2_e</th>\n",
       "      <td>5.980676</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNDR3</th>\n",
       "      <td>0.002050</td>\n",
       "      <td>-2.678413e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TECA3S45</th>\n",
       "      <td>0.000215</td>\n",
       "      <td>2.996214e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGK</th>\n",
       "      <td>-3.055820</td>\n",
       "      <td>-6.938894e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGM</th>\n",
       "      <td>-2.940945</td>\n",
       "      <td>-0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHCYT_BS</th>\n",
       "      <td>0.009060</td>\n",
       "      <td>-3.608225e-15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>347 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               fluxes  reduced_costs\n",
       "EX_fe3_e    -0.000407   1.075735e-14\n",
       "EX_glc__D_e -1.700000  -1.780625e-01\n",
       "EX_ca2_e    -0.000378  -3.108624e-14\n",
       "EX_so4_e    -0.020092   0.000000e+00\n",
       "EX_co2_e     5.980676   0.000000e+00\n",
       "...               ...            ...\n",
       "RNDR3        0.002050  -2.678413e-14\n",
       "TECA3S45     0.000215   2.996214e-14\n",
       "PGK         -3.055820  -6.938894e-18\n",
       "PGM         -2.940945  -0.000000e+00\n",
       "PHCYT_BS     0.009060  -3.608225e-15\n",
       "\n",
       "[347 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "solution_frame = solution.to_frame()\n",
    "solution_frame[solution_frame.fluxes != 0.] # Muestra flujos de reacciones diferentes a 0\n",
    "# Columna 'reduced_costs' muestra la mínima concentración de cada metabolito para optimizar el flujo de Biomasa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "437bc126",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fluxes</th>\n",
       "      <th>reduced_costs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EX_etha_e</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-6.677343e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_drib_e</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.502402e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_csn_e</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.298372e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_5mtr_e</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-2.244329e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EX_dtmp_e</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-3.375768e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GND</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGK</th>\n",
       "      <td>-3.055820</td>\n",
       "      <td>-6.938894e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGLYCP</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-5.065393e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGM</th>\n",
       "      <td>-2.940945</td>\n",
       "      <td>-0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PHCYT_BS</th>\n",
       "      <td>0.009060</td>\n",
       "      <td>-3.608225e-15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1250 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             fluxes  reduced_costs\n",
       "EX_etha_e  0.000000  -6.677343e-02\n",
       "EX_drib_e  0.000000  -1.502402e-01\n",
       "EX_csn_e   0.000000  -1.298372e-01\n",
       "EX_5mtr_e  0.000000  -2.244329e-01\n",
       "EX_dtmp_e  0.000000  -3.375768e-01\n",
       "...             ...            ...\n",
       "GND        0.000000   0.000000e+00\n",
       "PGK       -3.055820  -6.938894e-18\n",
       "PGLYCP     0.000000  -5.065393e-16\n",
       "PGM       -2.940945  -0.000000e+00\n",
       "PHCYT_BS   0.009060  -3.608225e-15\n",
       "\n",
       "[1250 rows x 2 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "solution_frame # Muestra soluciones para el flujo incluyendo flujos iguales a 0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f402c669",
   "metadata": {},
   "source": [
    "## Optimización de producción y rendimiento de PHB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "01ab2aa8",
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.io import read_sbml_model, write_sbml_model # Para leer archivo sbml\n",
    "model = read_sbml_model('datos/iYO844.xml')  \n",
    "write_sbml_model(model, 'datos/iYO844.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfbdc2b9",
   "metadata": {},
   "source": [
    "- **Función objetivo: Biomasa**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "54d49365",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Flujo de PHB: 0.0033002277076833566 mol gDCW^(-1) h^(-1)\n",
      "Rendimiento de biomasa: 0.0019413104162843274 mol biomasa/mol glucosa\n"
     ]
    }
   ],
   "source": [
    "model = model.copy()\n",
    "solution1 = model.optimize() # Optimizando flujo de biomasa\n",
    "print('Flujo de PHB:', solution1.fluxes['ACACT1r'], 'mol gDCW^(-1) h^(-1)') # Flujo de PHB\n",
    "print('Rendimiento de biomasa:', solution1.fluxes['ACACT1r'] / (-1 * solution1.fluxes['EX_glc__D_e']), 'mol biomasa/mol glucosa')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "98c0931d",
   "metadata": {},
   "source": [
    "- **Función objetivo para calcular PRODUCCION Y RENDIMIENTO DE PHB**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c26b2dff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "velocidad de produccion: 1.928571428571398 mol gDCW^(-1) h^(-1)\n",
      "Rendimiento de phb: 1.134453781512587 mol phb/mol glucosa\n"
     ]
    }
   ],
   "source": [
    "model = model.copy()\n",
    "model.objective = \"ACACT1r\" # Función objetivo: flujo de PHB\n",
    "solution2 = model.optimize() # Optimizando flujo de PHB\n",
    "print('velocidad de produccion:', solution2.fluxes['ACACT1r'], 'mol gDCW^(-1) h^(-1)') # Flujo de PHB\n",
    "print('Rendimiento de phb:', solution2.fluxes['ACACT1r'] / (-1 * solution2.fluxes['EX_glc__D_e']), 'mol phb/mol glucosa')"
   ]
  },
  {
   "attachments": {
    "image.png": {
     "image/png": "iVBORw0KGgoAAAANSUhEUgAABA8AAADNCAYAAADXL1l3AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAACZ6SURBVHhe7d1Njh1HcsDx162FSB6BALWVMN7IBzDgC/gKhG/gpXezmJ2XvoHBK/gCA/gA1sYDrnolEboARXFFM6gOTjAYGflRWV/v/X9Aol/lR2RUvXrVVclu9t2Hjy4AAAAAAODQHh4eLvf391+Vu7u7z1+lCPt6hvvHrwAAAAAAACEWDwAAAAAAQIrFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACkWDwAAAAAAACpuw8fPb7Gyt6+fXv55ZdfLr/99ttjDQAAAADgGj158uTy3XffXZ49e/ZYs9zDw8Pl/v7+q3J3d/f5qxRhX8/A4sGGXr9+fXnx4sXnN3Urb968uTx//vxxCwAAAACwtvfv319+/fXXy/fff/9Ys9yeiwf82sKGfv/998s333wz9Q0EAAAAABzPt99+e3n37t3j1vmxeAAAAAAAAFIsHgAAAAAAgBSLBwAAAAAAIMXiAQAAAAAASLF4AAAAAAAAUiweAAAAAACA1NUtHvzjP//X5zLLGjERe/qXvzy+mkditsYdnX/rcVtbO0+JX5ujpc+otfcPAAAAOLvdFg/sA/nsB/P//eu/fipWNl+pXkXx9vTq1atPZRZ9cIoeoLK2SKlf63ggoufPuz//+dPXo5H8tHi+zm7bcb4fAAAAcCS7LR7YB/KtHs51Hj+fr48WEI5CFg1evnz5uHVM/iFItvd86JO5W+YffXjbetzWtsiz9h61vocjZu2f5tgbb819AwAAAGa5+/DR4+tdyIO6fZAX/uHdt2eieKJUH+mNofnW4v/000+XH3744XFrmZ5FhDdv3lyeP3/+uPV3pYec7AGo9pCj42w/qfPblrZFYy07zveZEVPYfqNtmZkxpb/0seN8vFq78nPZNqHtGlNF21ZLm9bbvpYdZ8fI66itRSkXkbVFtL/Pzb9WWpeNAwAAwHV4/fr15ccff3zcWu7h4eFyf3//Vbm7u/v8VYqwr2c49P95IA/jUvThfCmJo8WzbbVFgGuhDyr2a1SnX/V1xvfxD0S63RpP+XH64CV8mxRbX5LlMtqW2SKmbFu23crG+bZWfpwdW2rT+hI/TraVbevhY1pZW0b72vyU1NniaX3PfAAAAMDWDrt4MPsBXhcitPgFhKytRschpw9O+rAUPTBFD1DZg1XWlj2MjcbM2jK1mPpVS+sctX62XV/Xcmltk201GrNUL7JxImsrqeWpX7W0zqH9hR8j21o826bjAQAAgCO6ur+2gDb2YSf6WmrrIWP8A5N9WPJta4hyiGT7N9qW8eNmH5fRvLxanFm5zogzYtb70DOmJy4AAABwFDe5eND7kwU1+usOZ6IPMPZrVKdf9fVWsofW2Q+btblENF82LjM6rsYfl9Zj5MepLM+19iGjefq5S/nXzNiHLEaWl2+bkQsAAACwpt3+w8ToYVt/9F/a7K8B+O1Mqa+fz8e3SnPVYtdynPEfJkZ/orH2Hydm/2GiPoz5h5ysrUYfhKJx2qZsH9sW5aOyNjESU0mfUr2qtZdk43zbSDyfYxbDjvX9Snn6MX4OP26kzdaLqC2a14/L+FyEjW3V4mZzR21al+UAAACA63FN/2Hi7n9tYbbSA/4MS2PP/GsLPUqLBziv7KEVAAAAwDHw1xYAAAAAAMDNuMrFA/kJAf1VghlmxwMAAAAA4Eyu7tcWjoxfWwAAAACA28GvLQAAAAAAgJvB4gEAAAAAAEixeIBhT//j6acy29Onccz/+dOfPpUZNNaseFZL3NF5tx63ta3ylHmyuWrto7bavzWsdUx6yLVBy2xrxc34/fHbe6vlc4RzYqnSPtT2fdRWx0znWXOuLfZDzZxrJNaM+bfcB2mP+sw+nwHcHhYPTujVq1efy9nUFhvkG9u7d+8et3Klb54t36D/6W9/+1QAT8+fo54felOoxcq27Rjf7yzk2tB6ffCOdtOs1zq7T0v2by29OfnzbMa5lsXLtu0Y36/miO9Fj7N9j8veH2nr3ZfsPZdYvedDizPsg5zTLCAAWILFg5ORBYOXL19+LnsuILz79483Vx8LviTf1HtvErCt7D06w/unOUrpuYEcHXcLzv6weCT2PJMyg43HOX9d5H2Z+f5ovC1dwz4AQAv+2sKG1vhrC7qYkCn9tQX/UwB2IUDaZNv20faoTpVi+nr1xZwdP3Ug9Ju0/Qbrv+H6b+T+m3HpG3Q2rtRm65fE9GOzcZnW+URLzFo82bZ9WucrtWlMFW1brTGV7a9axolobMlInhkZ42PYPEfalNQJX78G/69f9rOftYnoWuHrdNvHUtrXtvuYwo+PxolobCTKx4+VPrU6u60xa9vKx7FKc7SKzivlz6/W883HtNujba1GjkEPyWmE3UfL79/IPkc5tcxXavM52O2sTfjtHrWxPbG1r3wVWY617R61saOx1z6vAXyJv7aAqyIP8FL8A75sa5sV1XnaR2PaMfpat0fVvmHqN1VbpK5G++gYoXW+TduF37Z8Lp5tsznWxpVk43ybFDtnTWmMjWtpP9umdXaMH5fxMe3YrM1vW1meStt8fYn00zEaU/k2Ka1xMxJDi2fbfD5b0htXW5Q+zNp6rRth4+hr3RZ+22rJRdtac7Sx7OslbAybs5BtnUeL9onaZrDnmRRlzzmtn3EeRnMp27bnOV9jc/N5ltp0f6VO66Nj0EOPk41pY/s25XPx7WJpbnuz+3/2fQGAUSwenFjLTx3UZA/wtq3nQX/pokAv/SYefTOXOlta2RsffxMU3RQtNZLjEna+njmzfW89ZrOOX2suPbI8Z+Vtjb4PdkyUpxbPtkXzlcatQR5c9SHWsw+xsx5oR2W57J2bJbno8fR56bGOjnlUt5Q9z/z5JNt67vm2zFrn/BHYvP0+ZG2i1r6lkfn9mJ4Y9pzY4r0t5XamfQCApVg8OKkZCwfXoPZNW7Z9ORq5Ydg6PzvflvPiS6PvQ2//o5GHWy1rPLzeIjmGftFA2eOtxddv9T7o9a7X2c/5M7EPs7OPt8ScxZ4TZz0vrmEfANwWFg9OiIWDdqM3Knacj7H05icbP9qWicYt3Ydedr7S3FlOvfuQtWXsuNEYJb37sAfJZySnJQ+e+iBr2VgjcWc+BI/mIn1n5mH5uHZbFwKsljyi92E2e27Jg9LRzv8eckxbjuva7DGcdTxbHmajubL5pY2HYwA4P/7DxA3N+A8To7+usOZ/mBjx44T29eOibeuLto83YiM3sHrDEt2YRDcz0q9Ur3x7S1trTJ2/d75oXKZnPlGL68f4eNn42lghdT5OyzjV0ubrRes4389uZzSm9I/ieLW42dxRfN32c0UxtE8tB08folo/v/6hy48rtUcPa3ZsFnd2TKn3/aJxts7y/YWfS0QxlbRF8/i6UtwoniXtvq7Gn2dCzqfo3Go936RfqY9vs9saX9XmiZSOgT/Go3y+PvdSm++nSvVC2zK1cdoudTYPUcpF2L5+XKanr/J5iCjGkjzOuA/WyGcbwLhr+g8TWTzY0Bp/baFFtnhQWiDYC9/Qjmv0JgXXb9aDFI6F63H5GFzrOR9d5/e+9q8x/9b7dKR94HMNbI+/tgAAwEcsHODWcM4DAG4VP3mwIX7yoA2r4se0978+AdiWPiSLNa/Jcm0p2euas9W+H5F/P45w3Z/5/Wev72VH2Afur4B98GsLGHK0xQMAAAAAwHr4tQUAAAAAAHAzWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8WBDT548eXwFAAAAALhm79+/v6pnQP7awobevn17+fnnn/kzOQAAAABw5eRPpL548eLy7Nmzx5rl+FONAAAAAAAgxZ9qBAAAAAAAh8XiAQAAAAAASLF4AAAAAAAAUiweAAAAAACAFIsHAAAAAAAgxeIBAAAAAABIsXgAAAAAAABSLB4AAAAAAIAUiwcndnd397nM1hJ3dN6tx21tqzxlnmyuWvuorfYPuCb6eVzz87PlZ3PmXCOxZsy/9z4AAHA2LB7srHTD0XIj8uHDh08F8PT8Oer5IfnZYmXbdozvt8RIXD/GjyvVC9tWa/eiets/ay8ZGVcaI2ybby+1lerVSJutj9q3dLbrdXaspK13X7LjL7HWeG+uYR8AADgSFg8Qkhuh3hsrbCt7j87w/mmOUnpuukfH1WhM0RrX5qJjhYy39VE8O8a2+7FWlpcdY8dpvJLSfNk4P8bm5dt8u9B6oW22v5QsZmubsG1SUKfH1B/LURpvS9ewDwAAHA2LBwfgb278TYps29IqG1dqi+qsrH20LZONs21ReyQbo9tZe0ub1bJti1Vqi+os2277lOpbZONsW9S+td48pO8aDwWlHLL5WnLpzTXrn82XjevNQY0e6zVyKZEcR4oq1S/hY9q4UZ2ybbbd9/Vtekzla9S395i39I/maiFjshzPsA8AAJwFiwc7q92Q6E2OLS03J9pHxwit823aLvy25XPxbJvNsTauJBvn26TYOWtKY2xcS/vZNq2zY/y4jI9px2ZtftvK8lTa5utLpJ+O0ZjKt0lpjZuRGFo82+bzWao3ps3F0hjaVopZatNxUlQWR0RjWi0ZW1OKKfV+f2we2b72sDF72Pl9LqU2nUPqtL53Xk/GazyNaWP7NuVz8e1iaW57s/t/9n0BAOAMWDw4CL3xiW6ApM6WVvZm0d84RjeSS43kuISdr2fObN9bj9ms49eaS48sz1l5W6Pvgx0T5anFs23RfKVxNVEeGZ1Hi81FX2u8KM/SfFpv45b6Kttfx7SK5uvhc9MYWmybKtVLnRbpEymNFb7NxpNSiun5GFbWJmrtWxqZ34/piSHH15a1lXI70z4AAHAmLB4cQO1GR7Z9ORq5ydo6PzvflvPiS6PvQ2//Nen5O5PGi+Jm85Xq9WFGit1easl+y/zReKnT4pXGtMjGLol7dLJvWmbvo8ScRXKz5YyuYR8AAFgLiwcnMHpzZ8f5GEtvGLPxo22ZaNzSfehl5yvNneXUuw9ZW8aOG41R0rsPe5B8luQkY+1DQ0u8mfNF7MOM9rWvrSW59GjJW2T9WmOIpXGkzxbsPLPmlH3TUhLNlc0vbbVjBgAAYLF4cCDRzZxuS5stts6/Fn6c0LooprLbvk3GaZ3GULbNy8ZlavMJbdfSwvZvzcfPJ7ROvpbi2TbPx7R9sja77dv8OOFz6mX3wceK8pSyJjvP0n0Tfh9a2TyEzUVe23bfJuxY5cdJsWNLbH9hx9h6+1pk8+m2fy18vW0Ttt7mEsVWdowUvw/6VYuqtdk6G3MN0f5FdfZ1C+3vx0lsrfP75uf145QftwY7v8/lLK5hHwAAWOru443D+ncOwEHIDR+nPICziK5Ze1/H1ph/6326hn0AANymh4eHy/39/VdFvg/pVynCvp6BnzwAAAAAAAApFg8AADgo+Zds/VcDLXv/67bmNMse+3QN+wAAwNZYPMBN4eYOwNnIdcuWI5iZx177dA37AADAllg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACkWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8QAAAAAAAKRYPAAAAAAAACkWDwAAAAAAQIrFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACk7j589PgaK3v79u3ll19+ufz222+PNQAAAACAa/TkyZPLd999d3n27NljzXIPDw+X+/v7r8rd3d3nr1KEfT0Diwcbev369eXFixef39StvHnz5vL8+fPHLQAAAADA2t6/f3/59ddfL99///1jzXJ7Lh7wawsb+v333y/ffPPN1DcQAAAAAHA833777eXdu3ePW+fH4gEAAAAAAEixeAAAAAAAAFIsHgAAAAAAgBSLBwAAAAAAIMXiAQAAAAAASLF4AAAAAAAAUle3ePCP//xfn8ssa8REm6d/+cvjqz9e2+1Wo+P2dsacAQAAAFyn3RYP7AP57Afz//3rv34qXm2uUnsp3l5evXr1qcyiD6k9D6v6QK7l1pxhn7P3x9fZbTvO9wMAAABwm3ZbPLAP5Fs8nMuCgM4jxS8Q+PajkkWDly9fPm7t692f//yprMU/uI7Ot3aea5j10K773hvvjMcMAAAAwHoO+WsL+q//WmbIFgR04WDUzDxrZi4c2H9Z9l8zrX1s0Tr9aust36YPsFF/3bZfoz6+ztJ2289+tfXCbkftolRf48fZh3fbpu2ZrI+0lRYGWmIDAAAAuD2H/j8P9KcAtnow10WArebbmz5A2q+lh8qIPsj6MVoXxbNtXmlcNIeSNv3q+2TjRGk+YdssW6evbR8fU7ZbtObi2zLaN8pB6mzxtL5nPgAAAADX67CLB2v+6oAsDvj4WjeyYKFjbknpQVYfRPXh0z+Alh5GswfVUr3QtqhPNm40T6FjPR+jVTZO55KvWlrn0P7Cj5FtLZ5t0/EAAAAAbtuhf/JgDdHCgbi1h3/7YOm/LmUfPqWM8LnI9kisbJzNsTd2b/8eft9H8+wZ0xMXAAAAwO25qcWD0sLBUhK35ycVjkAfFu3XlgfI0QWGnsWAljmy8Uv05FkzMrZl32uyGFlOvm1GLgAAAACuw26LB/aBe4uHbz+XnU8WFGy9lKP+JIL9M42z/2Rjq+yBWNrkodMW1TqupNZeEo0bzVP4scrX1+IoO87zMaM+kda5PT/PaBwAAAAA1+Xuw0ePr6/Cmg/+S2P/9NNPlx9++OFxaztv3ry5PH/+/HELAAAAALCF169fX3788cfHreUeHh4u9/f3X5W7u7vPX6UI+3qGm/s/DwAAAAAAQJ+rXDzQXz2YZXY8AAAAAADO5Op+beHI+LUFAAAAALgd/NoCAAAAAAC4GSweAAAAAACAFIsHGPb0P55+KrM9fRrH/J8//elT2Vo052gue+3DqDPl6h3hWMu5rGW2teJm/P747b3V8jnb5y9S2ofavtf89z88eXzVZ3S+EZLjaJ49lh7LzFpxM35//Pbeavkc4XOrOeydxwyl/cjeh9J7s9TMY6qxZsWbaUZOa+zfGjHXVMp1rfMTMRYPTujVq1efy9nUFhvkAvDu3bvHrTp74YsuKNfuDPucvUfZth3j+52FnMs957N1tG+G+tm0+7Rk/9bSm5M/z2aca1m8bNuO8f1qlrwXSxYOeud8+p8fH04+lh5+0WDtRYQlx5LP7ZjenPxnRT8v9rWK6nr909/+9qlsRXOekXur7D2Q+qOd297oe5Qd62zbjvH9LGmbce6scQ6uEXMPZzg/rwmLBycjCwYvX778XPZcQHj37x+/0Xwse9OLn5TsAj7T6AV3dNzZ6X5L6XmPRsfdguxGD33seSZlBhuPc/4Psmjw7t/6z9l/+b/fPxXlt8+Ez+089rMiResipfojs/vG9791jR7r0XHAmfHXFja0xl9b0MWETOmvLfifArALAdIm27aPtkd1qhTT16sv5hz4Vyy5WMtFW0XblraV6lXLuGweS/qVxqmW+USpXmU5RfNGsnEjMWWMj6Hbo21K6oSvX4Nf1bbnatYmonPb1+m2j6W0r233MYUfH40T0dhIlI8fK31qdXZbY9a2lY9jleZoFZ1Xyp9freebj2m3R9tajRwD+Vf83ofxkXlU7yJC6acMWnIePZdEtI++Trd9LKV9bbuPKfz4aJyIxkaifPxY6VOrs9sas7atfByrNEer2mdD23s/Q9Lfa/lM+nE9c0aieVr3ZWku2fsx8l718Llbuh+1/Ws9Tsr3t9ujbZYfU1Lq42OW5slE89oYUUxfF21bs9tsfau1z88l+GsLuCryAC/FP+DLtrZZUZ2nfTSmHaOvdXsteqGzJbpo+Xo/zorqLNvmX5fGjeSp2/a1bouon48Z8blYvk1KS8waiaHFs20+ny3pNyRblLQJW691I2wcfa3bwm9bLbloW2uONpZ9vYSNYXMWsq3zaNE+UdsM9jyTouw5p/UzzsNoLmXbtjjn1/zx/1n8Txr47ZLsfLHnndZr3QgbR1/rtvDbVksu2taao41lXy9hY9ichWzrPFq0T9Q2g/2sSPGiuoz0l8+cFtHyGdR5/Ng9+H2Q0nscjsAeQ/vaH2ux9v5JfC2ebbN5qlK9rbOvdQ6p03qtG6U52Jj6dZTP08bL9sHnou3K1usYHA+LByfW8lMHNdkDvG3redBfe1HAkwuMFn8hsm1SLN93KRuvN/YaeS7JpyTLM2PH+FxkW4tn26L5SuPWIDfAejPs2ZvhWTfGo7Jc9s7Nklz0ePq89FhHxzyqW8qeZ/58km0993xbZq1zfg0tD+I99P8k0DJL66KBlZ0v2Wdla1kue+dmSS56PH1eeqyjYx7VLWU/K1L2tPf8lr32bHUNmckey+i41tpH2OPlY8q2Fs+2tR5rG6cUU0XtR5Hllu2DHueItrUeS+yDxYOTmrFwcC3kwqQXJ3/B0TZbjuja8+ztfzRyk6xljZvgWyTH0D98KHu8tfj6rd4HuaaMnLdnOOfX+qkDfcgfedifaY/z5dqd5XMrRj579uHlyJ/djL32aEFujWN1pHPoKOe1Pc6aj9LctOC4WDw4IRYOYtHFxl6YRiwd32qNefxFuVc0Zqvj0UryGclpyQ2s3hBbNtZI3Jk306O5SN+ZeVg+rt3WBwqrJY/ofZjNnlt6s4Nz4nPbz8e120f/3Or9QO/nVvpr6bH2tcHuU80e1yl571ve/xns/u2xr2tbY/9Gz2sV5ZHl1rIPWS6z9hvr4D9M3NCM/zAx+usKa/6HiRE/TmhfPy7atr5o+/iNp/fGQi4w9gKkFxytiy5A0haNK8WxbS3xotelcaolroi2rda2jI4rze/V4voYVhRft/1cUQztU8vB05uc1vPN3xT5caX26GbKjs3izo4p9b5fNM7WWb6/8HOJKKaStmgeX1eKG8WzpN3X1fjzTMj5FJ1breeb9Cv18W12W+Or2jyRnmMgP3kw+pMBI8c6+hONI399oVXL+WJpu68XdmwWd3ZMqff9onG2zvL9hZ9LRDGVtEXz+LpS3CieJe2+rsZ/VoR8XrTef8ZE7fNUiimyNuHba3PVLIlXy7Umez9Kbfoe976PnuRuj3nptVWqFy37bWN7vq0lDxXF9eNLr61SvfDxI9m4WsxSLmKkLRsjau01I9eSrVzTf5jI4sGG1vhrCy2yxYPSAsFejvzBx/WYdaODY+H60X4M9FcWlvxaAccbMxzlPJIHl+hhpvcB5uxK70f2Pkmb4HrwhyOdN7dyXh/9+xF/bQEAToqbHGDsPx8EAI/vqV+7tQUn3BZ+8mBD/ORBm6OvHgI4Jr2JFWteQ+RfbUr2umncat89rtdY6kifW99nyec5my+TzbnmtSd7H/icn9vM8/qIznB+8msLGHK0xQMAAAAAwHr4tQUAAAAAAHAzWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8WBDT5788Xe1AQAAAADX7f3791f1DMhfW9jQ27dvLz///DN/7gYAAAAArpz8KckXL15cnj179lizHH+qEQAAAAAApPhTjQAAAAAA4LBYPAAAAAAAACkWDwAAAAAAQIrFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvHgRtzd3T2++jupi+prRsft5Uy5Akehn/Mjfn5m5LTG/q0REwAA4ChYPDgAe8N5izedZ9jn7D3Ktu0Y32+JkbgtY0pttXpb1GibKLX5ei21NhHVqda2Wrtl60ttJVm7tmXjZ/jw4cOn0svm53PMtu0Y38+StpG8vNH9y6wREwAA4ChYPDgIvemUkt04zzR6ozs67ux0v6X0vEej42o0pmiJa/OQ4seUYki9zhPxca2RNp3PFs3N10tRWZuPafe1p63W7tl6HadjSmx8rzbfUdgcs/3xRscBAABgfSwenIDcRNuiSvWq1NZS74tvj9h226dWb1/bdmHrfVsmG2Pbovat9eYhfeXhag1R3LXmmiE7FrYt24fR/Wud22uZL+oz8r7LmFJRpfoz8Lnbokr1S/iYLXF9n2jbFmtGGwAAwFIsHhyc3PTJA4Mt/kYwqvfjrKjOsm3+dWncSJ66bV/rtoj6+ZgRn4vl26S0xKyRGFo82+bzWao3ZimXJbnZmN5om1Vqz8bVYo6K4kqdlojUtx7blr61+SI2pn/fhdRpfU/cEVn+ti06DqV6W2df6xxSp/VaN0pzsDH16yifp42X7YPPRduVrdcxAAAAo1g8OAi5sdPibwBtmxTL913KxuuNvUaeS/IpyfLM2DE+F9nW4tm2aL7SuJoojxqdS4rmMhJH2Xg2pljSJttaZLtkpC2L6dvkdZaL1mmRbSsaU9LStzZfxMaM4tfaR0heWnxM2dbi2baWfRM2TimmitqPIsst2wc9zhFtaz2WAAAAGRYPDkJuCPWm0N/oaZstR3Ttefb2X5OcIzPz8A8Z9vUeasc6y63WlsWM2rJcSrFENleJjNFit1VvvL3o8ZqZrxyHo+y/vi9752SPs+ajNDctAAAAS7F4cDDRTZ69IRyxdHyrNebxN8O9ojFbHY9Wks+SnGSsPW9649kHDI1jX7fK5hxtE9Ie5ZLlF7WV4oiszWrt57WMk3ZbbN2aJDdlX1+LNfbPv0+9ojyy3Fr2Ictl1n4DAIDbxuLBQenNnt4QyrYtNTJO+/qbShvDvq7Jxo3mqaIxPqbw+xLJ9n1pniPsPC351/h9aGXzkNKai/b3r4Vua52NOdombHvUVlJq03qNaftlbcLW21zktW2z7fpV6y1bF7WXZPONivKM6uzrNek8Upbum1hr/7S/H2e3fZvM6+uUz8mPE7Y+6i9F60U2HwAAwIi7jzcYf7/bAADgYPyD8Z6iXI6UHwAAuG4PDw+X+/v7r4rcj+hXKcK+noGfPAAAHBoP5gAAAPtj8QAAgEaykKGr+FpY3AAAALeAxQMAADrIYoEtAAAAt4DFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACkWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8QAAAAAAAKRYPAAAAAAAACkWDwAAAAAAQIrFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACkWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8QAAAAAAAKRYPAAAAAAAACkWDwAAAAAAQIrFAwAAAAAAkGLxAAAAAAAApFg8AAAAAAAAKRYPAAAAAABAisUDAAAAAACQYvEAAAAAAACkWDwAAAAAAAApFg8AAAAAAECKxQMAAAAAAJBi8QAAAAAAAKRYPAAAAAAAACkWDwAAAAAAQIrFgwO6u7sLy2xRzLXmWyvuEi05tfRZ25o52NhrzTHLkXMTZziGPa5tf2bLjs+MY5bFx1wc6zmOdhx5P780cjz2ek9574DjYvHgQGoX6JkXcC7MfT58+PD46nqc6RzgfMVZzD5Xr/Hac1Qc6+vB94wv8f0ewCwsHhxEz8Vy6YU1Gy83T1pmWivuErWc9DgdKee11I7FnriRwBFFn5lZ5+otXXv2xrG+PnzP+NKS4xFd59bEewccH4sHB2Uv2FtdtPEljj2APXDt2Q7HGgCAdncfv2nyXXNnfqW19JaU+tl6qcvilVZ1S7FEFr8U29aLWtxIFkP5Ppbv35KTVRsvfIyWMZlofJan7y9a5ozGCR3bOudIPn68H2P7R/HEyLyqp29JlLOtW5JfJpo30jKfz7dnW5TmFi19147fqzefWn+l47L+WWzh262ob09skfXx8URvPjO1xPf5q6i+1FdF82VjfH+R9fHxRJSH6ulrZTmLWrsX5ZHF8P1Fa+5WNK8VzSO0n8+xJ+cs32jerL+V5SSyvGptUV5Cx2X9o20rip318fGE7R/FEyPzqp6+wJk8PDxc7u/vvypyzutXPf/t6xn4yYODyS5qLRe86OSYecL4WHpCRvUzleJF9VE+ojWnbHwWozSmVe/4UlvPnDON5BO19ebfM29P35KeuGLWnK3zjsyXtYmeuXvqVWmMV4pTqh9VmmOWUqzSfmT1XlSnSm29Y0p1PfW91o4fKc1XUmrrHVOq66lfUymPklJbT97StzRvTxyrNK53rixOr574s/XMM7LPM/atZ96evgDasXhwhWSRobTQ4OuzviWlMb6+doHW/rZENE6pbzZPT9+IH18T9R/5RlWb18bUvrZ/bU4f24/vtSSfaG7t7+uzuNpWal+SY4mP4W0xZ7SPdq5SXy/qb2VtkRn9W/Jdy0g+Vi0/bS/1qY33av1t7trX9u891kd7byzbpnn6/e9Vm/PoxzfLZcmxqeWy5LiUZHP6+qyv0PZSn9r4rfZvJI7wcaLYlrZnfZbscxRb+/v6LK62ldqX5Aggx+LBlbEXxzUulDamVarv4S/2Sl63xs9i2K8ltfGidCxL/Vu0zHuNjrCvMq+Wmj3ep55zWl6P5FIb4+dQNreWY2P7WKX+1pL4vVryWSKK2bt/Lf1naIm99nszEr/UZuszLXPO0BJ7ZP/XstVxsdba/yjnWXNJfy09SnNuYe359tw3JfNqATCGxQMcQukbtif9tPS61m8WS47JGvbKp2fenr6zrDHnWp8VXDfOjXVxfGPXflyuff8ie+1zz7w9fQHUsXiA3fVc/IFRpQftazu3+KwAf/Cf+WtdQG5l91+vEfZacevHZ8StfF85G94XYD0sHhxMdlG7hQtedMH3+y19uMn5Oz0evuwlykXK2qI5pVilerH258vObctMfFbQwp5/tlwb/3nY6ntodGyl3LromEg5u2xftjrn9mL33Za1RXNKsUr14trfF2BNLB4cgL+wRRc1XxddDM/I7lfLPi3Zb75ZwJJz6eyfo+ycvpZrBIB5StcFrhdzXMP3lWvE+wLMw+LBQclDgS2tbF/7+ogXzSX5lY6JjRMdi9I4VRsv1jiWLfOW+P5aZvHxa3x/LWurzRvV9drjfVp6TrfMURPNK2xuLcfG9unVG1/qtZxB7/619C/x/bWMWvu96Y0vSm0z5yzx/bWMGtn/FiM57XFc1tr/SO9cUq9lD6Uct+Ln17K22rxRHYAZLpf/B74CP7cApOjpAAAAAElFTkSuQmCC"
    }
   },
   "cell_type": "markdown",
   "id": "be14555f",
   "metadata": {},
   "source": [
    "# Nota\n",
    "aca agrege una funcion objetivo de PHB pero usando Acetyl-CoA C-acetyltransferase osea estoy dando a entender como seria la funcion que se deberia de dar o que formula se aplica para que nos el parametro verdadero de phb que la vuelvo aplicar abajo pero ya con otros datos \n",
    "![image.png](attachment:image.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d088c31d",
   "metadata": {},
   "source": [
    "### AGREGANDO METABOLITOS/REACCIONES PARA LA PRODUCCIÓN DE PHB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1cab8466",
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra.test\n",
    "from cobra.io import read_sbml_model # Importando lector de archivos 'sbml'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "65cc3d0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cargar el modelo metabólico para la escala del genoma \n",
    "model = read_sbml_model('datos/iYO844.xml')\n",
    "model = model.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "eb8aa97b",
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra import Metabolite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "625fb8bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Metabolitos \n",
    "accoa_c = model.metabolites.get_by_id('accoa_c')\n",
    "coa_c = model.metabolites.get_by_id('coa_c')\n",
    "h_c = model.metabolites.get_by_id('h_c')\n",
    "nad_c = model.metabolites.get_by_id('nad_c')\n",
    "nadp_c = model.metabolites.get_by_id('nadp_c')\n",
    "nadph_c = model.metabolites.get_by_id('nadph_c')\n",
    "co2_c = model.metabolites.get_by_id('co2_c')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d93dfba3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creando nuevos metabolitos\n",
    "aaccoa_c = Metabolite( 'aaccoa_c', formula='C25H36N7O18P3S', name='Acetoacetyl-CoA', compartment ='c')\n",
    "rhybc_c = Metabolite( 'rhybc_c', formula='C25H28N7O18P3S', name='R-3-hydroxybutanoyl-CoA', compartment ='c')\n",
    "polhybu_c = Metabolite( 'polhybu_c', formula='C12H18O6', name='Poly-beta-hidroxybutyrate', compartment ='c')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f155358a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Nuevas reacciones\n",
    "from cobra import Reaction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6ad081e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reacción de phbA\n",
    "reaction1 = Reaction('phbA')\n",
    "reaction1.name = 'Acetyl-CoA C-acetyltransferase'\n",
    "reaction1.subsystem = 'PHB biosynthesis'\n",
    "reaction1.lower_bound = 0\n",
    "reaction1.upper_bound = 1000\n",
    "reaction1.add_metabolites({accoa_c: -2.0, aaccoa_c: 1.0, coa_c: 1.0})\n",
    "model.add_reaction(reaction1)\n",
    "\n",
    "# Reacción de phbB\n",
    "reaction2 = Reaction('phbB')\n",
    "reaction2.name = 'Acetoacetyl-CoA reductase'\n",
    "reaction2.subsystem = 'PHB biosynthesis'\n",
    "reaction2.lower_bound = 0\n",
    "reaction2.upper_bound = 1000\n",
    "reaction2.add_metabolites({aaccoa_c: -1.0, h_c: -1.0, nadph_c: -1.0, nadp_c: 1.0, rhybc_c: 1.0})\n",
    "model.add_reaction(reaction2)\n",
    "\n",
    "# Reacción de phbC\n",
    "reaction3 = Reaction('phbC')\n",
    "reaction3.name = 'Polyhydroxyalkanoate synthase'\n",
    "reaction3.subsystem = 'PHB biosynthesis'\n",
    "reaction3.lower_bound = 0\n",
    "reaction3.upper_bound = 1000\n",
    "reaction3.add_metabolites({rhybc_c: -3.0, coa_c: 3.0, polhybu_c: 1.0})\n",
    "model.add_reaction(reaction3)\n",
    "\n",
    "# Reacción de Ex_PHB:\n",
    "reaction4 = Reaction('Ex_PHB')\n",
    "reaction4.name = 'Exchange PHB'\n",
    "reaction4.subsystem = 'PHB biosynthesis'\n",
    "reaction4.lower_bound = 0\n",
    "reaction4.upper_bound = 1000\n",
    "reaction4.add_metabolites({polhybu_c: -1.0})\n",
    "model.add_reaction(reaction4)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b14ac6a",
   "metadata": {},
   "source": [
    "### CUANTIFICACIÓN DE LA PRODUCCIÓN DE PHB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d44dd7ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Establecer el objetivo de PHB\n",
    "model.reactions.get_by_id('BIOMASS_BS_10').objective_coefficient = 0\n",
    "model.reactions.get_by_id('BIOMASS_BS_10').objective_coefficient = 0\n",
    "model.reactions.get_by_id('Ex_PHB').objective_coefficient = 1.0\n",
    "model.objective = model.reactions.Ex_PHB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1119085f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Solución\n",
    "solution = model.optimize() # 'solution' se almacena en 'model.solution'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "50355845",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de produccion de Biomasa: 0.0 h^(-1)\n"
     ]
    }
   ],
   "source": [
    "## VELOCIDAD PRODUCCION DE BIOMASA\n",
    "print('Velocidad de produccion de Biomasa:', solution.fluxes['BIOMASS_BS_10'], 'h^(-1)')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a66abcca",
   "metadata": {},
   "source": [
    "# NOTA \n",
    "me olvide de editar aca seria velocidad de produccion de biomasa va ser 0 por que solamente esta calculando la cuantificacion del phb y cuando le damos solution fluxes a la biomasa que ya esta dado en la lista de reacciones al inicio entonces va ser 0 asi lo lee el jupyteR PERO SI SE ESTUBIESE CUANTIFICANDO  BIOMASA NOS DARIA OTRO RESULTADO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "f2ab494b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de producción de PHB: 0.5806451612903225 mmol gdcw^(-1) h^(-1)\n"
     ]
    }
   ],
   "source": [
    "## Velocidad de producción de PHB\n",
    "print('Velocidad de producción de PHB:', solution.fluxes['Ex_PHB'], 'mmol gdcw^(-1) h^(-1)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bfde558c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de PHB: 0.34155597722960146 mol PHB/mol glucosa\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de PHB\n",
    "print('Rendimiento de PHB:', solution.fluxes['Ex_PHB'] / (-1 * solution.fluxes['EX_glc__D_e']), 'mol PHB/mol glucosa')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d36e0bf3",
   "metadata": {},
   "source": [
    "## Optimización de fuentes de carbono, nitrógeno, oxígeno  y de otros nutrientes "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e455cf6",
   "metadata": {},
   "source": [
    "### Sacarosa (carbono)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "0766140e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de producción de PHB: 1.2387096774193567 mmol gdcw^(-1) h^(-1)\n"
     ]
    }
   ],
   "source": [
    "medium = model.medium\n",
    "with model:\n",
    "    medium['EX_glc__D_e'] = 0\n",
    "    medium['EX_sucr_e'] = 1.7\n",
    "    model.medium = medium\n",
    "    solution = model.optimize()\n",
    "    print('Velocidad de producción de PHB:', solution.fluxes['Ex_PHB'], 'mmol gdcw^(-1) h^(-1)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "610f91d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de PHB: 0.728652751423151 mol PHB/mol sacarosa\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de PHB\n",
    "print('Rendimiento de PHB:', solution.fluxes['Ex_PHB'] / (-1 * solution.fluxes['EX_sucr_e']), 'mol PHB/mol sacarosa')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1bc5731e",
   "metadata": {},
   "source": [
    "### Glicina(nitrogeno)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3c0c2b35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de producción de PHB: 0.7122580645161287 mmol gdcw^(-1) h^(-1)\n"
     ]
    }
   ],
   "source": [
    "medium = model.medium\n",
    "with model:\n",
    "    medium['EX_sucr_e'] = 0\n",
    "    medium['EX_gly_e'] = 1.7\n",
    "    model.medium = medium\n",
    "    solution = model.optimize()\n",
    "    print('Velocidad de producción de PHB:', solution.fluxes['Ex_PHB'], 'mmol gdcw^(-1) h^(-1)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "a68f7874",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de PHB: 0.418975332068311 mol PHB/mol glicina\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de PHB\n",
    "print('Rendimiento de PHB:', solution.fluxes['Ex_PHB'] / (-1 * solution.fluxes['EX_gly_e']), 'mol PHB/mol glicina')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60af99b4",
   "metadata": {},
   "source": [
    "### Fructosa (carbono)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "523efca4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de producción de PHB: 1.238709677419353 mmol gdcw^(-1) h^(-1)\n"
     ]
    }
   ],
   "source": [
    "medium = model.medium\n",
    "with model:\n",
    "    medium['EX_gly_e'] = 0\n",
    "    medium['EX_fru_e'] = 1.7\n",
    "    model.medium = medium\n",
    "    solution = model.optimize()\n",
    "    print('Velocidad de producción de PHB:', solution.fluxes['Ex_PHB'], 'mmol gdcw^(-1) h^(-1)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b994586d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de PHB: 0.7286527514231488 mol PHB/mol fructosa\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de PHB\n",
    "print('Rendimiento de PHB:', solution.fluxes['Ex_PHB'] / (-1 * solution.fluxes['EX_fru_e']), 'mol PHB/mol fructosa')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "caf5933d",
   "metadata": {},
   "source": [
    "### Volviendo al medio predeterminado (glucosa mejor fuente)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "8d3a01eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Velocidad de producción de PHB: 0.5806451612903216 mmol gdcw^(-1) h^(-1)\n"
     ]
    }
   ],
   "source": [
    "medium = model.medium\n",
    "with model:\n",
    "    medium['EX_fru_e'] = 0\n",
    "    medium['EX_glc__D_e'] = 1.7\n",
    "    model.medium = medium\n",
    "    solution = model.optimize()\n",
    "    print('Velocidad de producción de PHB:', solution.fluxes['Ex_PHB'], 'mmol gdcw^(-1) h^(-1)')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52bcf4a3",
   "metadata": {},
   "source": [
    "# Yp/X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e2a173bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rendimiento de PHB: 0.34155597722960096 mol PHB/mol glucosa\n"
     ]
    }
   ],
   "source": [
    "# Rendimiento de PHB\n",
    "print('Rendimiento de PHB:', solution.fluxes['Ex_PHB'] / (-1 * solution.fluxes['EX_glc__D_e']), 'mol PHB/mol glucosa')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af15e18c",
   "metadata": {},
   "source": [
    "## Knock-out de reacciones"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8c3aed8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "from cobra.io import read_sbml_model, write_sbml_model # Para leer archivo sbml\n",
    "model = read_sbml_model('datos/iYO844.xml')  # Archivo \"iMZ1055.xml\"\n",
    "write_sbml_model(model, 'datos/iYO844.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1051e92b",
   "metadata": {},
   "source": [
    "1.928571428571398"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08c35c01",
   "metadata": {},
   "source": [
    "**a) En reacción de biomasa**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "4a806601",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.9285714285714315"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutant1 = model.copy()\n",
    "mutant1.reactions.BIOMASS_BS_10.knock_out() # KO a biomasa\n",
    "mutant1.objective = \"ACACT1r\" # Función objetivo: flujo de PHB\n",
    "solution1 = mutant1.optimize() # Optimizando flujo de PHB\n",
    "solution1.fluxes['ACACT1r']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "869d0858",
   "metadata": {},
   "outputs": [],
   "source": [
    "import cobra.test\n",
    "from cobra.io import read_sbml_model # Importando lector de archivos 'sbml'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74d4e493",
   "metadata": {},
   "source": [
    "## **III. Capitulo 3. Resultados y aplicación de Jupyter Notebook**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "832a60b6",
   "metadata": {},
   "source": [
    "# BATCH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3cf8f3ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import ipywidgets as wg\n",
    "from IPython.display import display\n",
    "from scipy.integrate import odeint\n",
    "import matplotlib.animation as animation\n",
    "from matplotlib.widgets import Slider\n",
    "\n",
    "# tenemos los parametros a evaluar \n",
    "#mumax= Velocidad específica maxima de crecimiento\n",
    "# Ks=  constante de saturacion .\n",
    "#yxs= rendimiento del crecimiento de biomasa en consumo de sustrato\n",
    "#ypx= rendimiento del producto generado por biomasa\n",
    "\n",
    "mumax = 0.1179663893223982     # 1/hora\n",
    "Ks = 1.00         # g/litro\n",
    "Yxs = 0.06939199371905777         # g/g\n",
    "Ypx = 0.34155597722960096        # g/g\n",
    "\n",
    "\n",
    "#ecuaciones a aplicar para  batch\n",
    "def mu(S):\n",
    "    return mumax*S/(Ks + S)\n",
    "\n",
    "def Rg(X,S):\n",
    "    return Ypx*mu(S)\n",
    "    \n",
    "def Rp(X,S):\n",
    "    return -1/Yxs*mu(S)\n",
    "\n",
    "# calculando \n",
    "\n",
    "def xdot(x,t):\n",
    "    X,P,S = x\n",
    "    dX = mu(S)*X\n",
    "    dP = Rg(X,S)*X\n",
    "    dS = Rp(X,S)*X\n",
    "    print(dP)\n",
    "    print(dS)\n",
    "    print(dX)\n",
    "    \n",
    "    return [dX,dP,dS]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "29f94a47",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(a) :  \n",
    "    IC = [0.05, 0.0, 10.0]\n",
    "\n",
    "    t = np.linspace(0,30)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S = sol.transpose()\n",
    "    \n",
    "    plt.plot(t,S)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de Sustrato.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d7a70f39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ff06de5027364fd0ad98d0411c0c9da0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='a', max=10.0, step=5.0), Output()), _dom_classes=('w…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(a)>"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de sustrato.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=10,step=5)\n",
    "\n",
    "wg.interact(myPlot,  a=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b1f3f2e",
   "metadata": {},
   "source": [
    "En la Figura 1. Se muestra la disminución de la concentración (g/litros) en función al tiempo (hr), en azul. La concentración de sustrato se puede observar como una curva con una pendiente que disminuye constantemente."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "1fa93148",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(b) :  \n",
    "    IC = [0.05, 0.0, 10.0]\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S = sol.transpose()\n",
    "    \n",
    "    plt.plot(t,X)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de Biomasa.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "772a3317",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "bc44229bda494a95a5497d37cb86c2f7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='b'), Output()), _dom_classes=('widget-interact',))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(b)>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de sustrato.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot,  b=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e0f61b6",
   "metadata": {},
   "source": [
    "En la Figura 2. Se muestra el aumento de la concentración (g/litros) en función al tiempo (hr), en azul. La concentración de biomasa se puede observar como una curva sigmoidal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "166eeeca",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(c) :  \n",
    "    IC = [0.05, 0.0, 10.0]\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S = sol.transpose()\n",
    "    \n",
    "    plt.plot(t,P)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de Producto.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "0f6ce36b",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2a470a31e7e04dba9e82ddba2278aef5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='c'), Output()), _dom_classes=('widget-interact',))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(c)>"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de sustrato.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=100,step=0.1)\n",
    "wg.interact(myPlot,  c=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56825157",
   "metadata": {},
   "source": [
    "En la Figura 3. Se muestra el aumento de la concentración (g/litros) en función al tiempo (hr), en azul. La concentración de producto se puede observar como una curva sigmoidal."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "ca581e34",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(d) :  \n",
    "    IC = [0.05, 0.0, 10.0]\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S = sol.transpose()\n",
    "    \n",
    "    plt.plot(t,S)\n",
    "    plt.plot(t,X)\n",
    "    plt.plot(t,P)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de Sustrato.',\n",
    "            'Concentracion de Biomasa.',\n",
    "            'Concentracion de Producto',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "51eb2485",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8dfa5b7e6fa44e1bac1c2b0e37d3da7a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='d', max=20.0, step=5.0), Output()), _dom_classes=('w…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(d)>"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de sustrato.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=20,step=5)\n",
    "\n",
    "wg.interact(myPlot,  d=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34b7ef40",
   "metadata": {},
   "source": [
    "En la Figura 4. Se muestra el recorrido de los parámetros las concentraciones en función al tiempo (en azul, naranja, verde), se puede observar cuando la concentración de sustrato disminuye, la concentración de biomasa y producto  están en una misma línea."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6c62b27",
   "metadata": {},
   "source": [
    "# FED BATCH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "a675dc55",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import ipywidgets as wg\n",
    "from IPython.display import display\n",
    "from scipy.integrate import odeint\n",
    "import numpy as np\n",
    "import matplotlib.animation as animation\n",
    "from matplotlib.widgets import Slider\n",
    "\n",
    "# tenemos los parametros a evaluar \n",
    "#mu= Velocidad específica maxima de crecimiento\n",
    "# Ks= constante de semisaturación\n",
    "#yxs= rendimiento del crecimiento de biomasa en consumo de sustrato\n",
    "#Sf=sustrato fresco a una determinada concentración\n",
    "#f(t): caudad  en litros por hora\n",
    "\n",
    "mumax = 0.1179663893223982     # 1/hora\n",
    "Ks = 1.00         # g/litro\n",
    "Yxs = 0.06939199371905777         # g/g\n",
    "Ypx = 0.34155597722960096        # g/g\n",
    "Sf = 1.7         # g/litro\n",
    "\n",
    "# caudal de entrada\n",
    "\n",
    "def F(t):\n",
    "    return 0.05\n",
    "\n",
    "\n",
    "#ecuaciones a aplicar para feed batch\n",
    "def mu(S):\n",
    "    return mumax*S/(Ks + S)\n",
    "\n",
    "def Rg(X,S):\n",
    "    return mu(S)*X\n",
    "    \n",
    "def Rp(X,S):\n",
    "    return Ypx*Rg(X,S)\n",
    "\n",
    "# calculando \n",
    "\n",
    "def xdot(x,t):\n",
    "    X,P,S,V = x\n",
    "    dX = -F(t)*X/V + Rg(X,S)\n",
    "    dP = -F(t)*P/V + Rp(X,S)\n",
    "    dS = F(t)*(Sf-S)/V - Rg(X,S)/Yxs\n",
    "    dV = F(t)\n",
    "    print(dP)\n",
    "    print(dS)\n",
    "    print(dX)\n",
    "    print(dV)\n",
    "    return [dX,dP,dS,dV]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "fb374ae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(f) :\n",
    "    \n",
    "\n",
    "    IC = [0.05, 0.0, 10.0, 1.0]\n",
    "\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S,V = sol.transpose()\n",
    "\n",
    "\n",
    "\n",
    "    plt.plot(t,S)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de Sustrato.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f199a70c",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1c8950985ec1435b9df5a5711cfeb80f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='f'), Output()), _dom_classes=('widget-interact',))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(f)>"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de sustrato.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot,  f=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30ec0bb0",
   "metadata": {},
   "source": [
    "Se muestra la disminución de la concentración en función al tiempo (en azul), el sustrato límite que podría haber en el reactor considerando que el volumen varía y la concentración son 1,7 g/L."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "45bb98cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(e) :\n",
    "\n",
    "    IC = [0.05, 0.0, 10.0, 1.0]\n",
    "\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S,V = sol.transpose()\n",
    "\n",
    "\n",
    "\n",
    "    plt.plot(t,P)\n",
    "\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Concentracion de biomasa.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "99e22bd9",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "071f552c1cb04c49a1f121b43ab145ec",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='e'), Output()), _dom_classes=('widget-interact',))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(e)>"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de biomasa.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot,  e=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8f6ab88",
   "metadata": {},
   "source": [
    "En la Figura 6. Se muestra la concentración de biomasa en función al tiempo (en azul),se puede observar cuando la concentración inicial 0 se obtuvo un menor crecimiento y hubo un aumento de biomasa a las 30 horas de cultivo llegando a un máximo de 0,10 g/L ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "57d2748d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#ejecutamos para poder calcular la simulacion .\n",
    "def myPlot(d) :\n",
    "    \n",
    "    IC = [0.05, 0.0, 10.0, 1.0]\n",
    "    \n",
    "    t = np.linspace(0,50)\n",
    "    sol = odeint(xdot,IC,t)\n",
    "    X,P,S,V = sol.transpose()\n",
    "    plt.plot(t,X)\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "    plt.legend(['Concentracion de Producto.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "81b7b0c7",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5fe00f6d3eb1402c80cfd0644a84e021",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=50.0, description='d', min=50.0), Output()), _dom_classes=('widget-int…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(d)>"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de producto.\n",
    "c_slide= wg.FloatSlider(value=1,min=50,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot, d=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e345f25",
   "metadata": {},
   "source": [
    "En la Figura 7. Se muestra la concentración de producto en función al tiempo (en azul),se puede observar cuando la concentración inicial 0,05 se obtuvo un menor crecimiento y hubo un aumento de crecimiento al as 30 horas de cultivo llegando a un máximo de 0,35 g/L ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a890d1ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "def myPlot(g) :\n",
    "\n",
    "    IC = [0.05, 0.0, 10.0, 1.0]\n",
    "\n",
    "\n",
    "    t = np.linspace(0,80)\n",
    "\n",
    "    sol = odeint(xdot,IC,t)\n",
    "\n",
    "    X,P,S,V = sol.transpose()\n",
    "\n",
    "\n",
    "\n",
    "    plt.plot(t,V)\n",
    "\n",
    "\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "\n",
    "    plt.legend(['Volumen [litros]',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "025adb95",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0e1ffbbf4eef4e46aded601fef63d45c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=1.0, description='g'), Output()), _dom_classes=('widget-interact',))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(g)>"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria del parametro de volumen.\n",
    "c_slide= wg.FloatSlider(value=1,min=0,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot,  g=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58598aa5",
   "metadata": {},
   "source": [
    "En la Figura 8. Se muestra la concentración del volumen en función al tiempo (en azul),se puede observar cuando la concentración se aumenta hay un mayor volumen en el tiempo en este caso al iniciar con 0 h hasta 80 h obtenemos un volumen de 5 g/L."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "b21c2fa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib notebook\n",
    "%matplotlib notebook\n",
    "def myPlot(c) :\n",
    "    \n",
    "    IC = [0.05, 0.0, 10.0, 1.0]\n",
    "    \n",
    "    t = np.linspace(0,50)\n",
    "    \n",
    "    sol = odeint(xdot,IC,t)\n",
    "    \n",
    "    X,P,S,V = sol.transpose()\n",
    "    \n",
    "    plt.plot(t,X)\n",
    "    plt.plot(t,P,)\n",
    "    plt.plot(t,S,)\n",
    "    plt.plot(t,V,)\n",
    "    plt.xlabel('Tiempo [hr]')\n",
    "    plt.ylabel('Concentracion [g/litros]')\n",
    "    plt.legend(['Concentracion de biomasa.',\n",
    "            'Concentracion de Producto.',\n",
    "            'Concentracion de Sustrato',\n",
    "            'Volumen [litros]'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "50f43e39",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "03685c3cccc64a06b9f25305bd53b6b3",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "interactive(children=(FloatSlider(value=50.0, description='c', min=50.0), Output()), _dom_classes=('widget-int…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<function __main__.myPlot(c)>"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#ejecutar para mostrar la trayectoria de los parametros .\n",
    "#ejecutamos interacccion para poder ajustar el acercamiento y alejamiento y ver los valores del eje x como y\n",
    "c_slide= wg.FloatSlider(value=1,min=50,max=100,step=0.1)\n",
    "\n",
    "wg.interact(myPlot, c=c_slide)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3a3528c",
   "metadata": {},
   "source": [
    "En la Figura 9. Se muestra el recorrido de los parámetros las concentraciones en función al tiempo (en azul, naranja, verde y rojo), se puede observar cuando la concentración se aumenta hay un mayor rendimiento del cultivo Feed Batch"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a37bdbb",
   "metadata": {},
   "source": [
    "# **IV.\tConclusiones** "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8549f9a2",
   "metadata": {},
   "source": [
    "Los PHB son plásticos microbianos biodegradables, biocompatibles que se obtienen a partir de fuentes renovables. Sus propiedades físicas dependen en gran medida de su estructura química, en especial el PHB, las propiedades físicas de los PHB que se conocen hasta el momento, estos podrían utilizarse como sustitutos de algunos plásticos derivados del petróleo.\n",
    "Se considera que Bacillus subtilis es productor de PHB, utilizando como mejor fuente de carbono a la glucosa y nitrógeno a Glicina.\n",
    "\n",
    "En el cultivo batch y Feed Batch se obtuvieron los parámetros cineticos: velocidad específica máxima de crecimiento 0.1179663893223982 1/hora, constante de afinidad del sustrato 1.00 g/litro, rendimiento del crecimiento de biomasa en consumo de sustrato 0.06939199371905777 g/g, rendimiento del producto generado por biomasa 0.34155597722960096 g/g y sustrato 1.7 g/litro.\n",
    "\n",
    "El cultivo Feed Batch se utiliza para describir a un cultivo por lote al que se le añade, continua o secuencialmente el medio. Por lo tanto, el volumen de un cultivo por lote alimentado aumenta con el tiempo. La adición de sustrato produce cambios en las características de la célula que se ven reflejados en las variaciones de los parámetros cinéticos donde se muestra la disminución de la concentración en función al tiempo , el sustrato límite que podría haber en el reactor considerando que el volumen varía y la concentración son 1,7 g/L,la concentración de biomasa se puede observar cuando la concentración inicial 0 se obtuvo un menor crecimiento y hubo un aumento de biomasa a las 30 horas de cultivo llegando a un máximo de 0,10 g/L dando como resultado concentración se obtuvo un menor crecimiento y hubo un aumento de crecimiento al as 30 horas de cultivo llegando a un máximo de 0,35 g/L .\n",
    "\n",
    "Se define como un cultivo en batch donde se alimenta continuamente medio nutritivo, si el nutriente que se alimenta es el limitante del crecimiento, esta técnica permite controlar la velocidad de crecimiento, por lo tanto se muestra la disminución de la concentración (g/litros) en función al tiempo (hr), la concentración de sustrato se puede observar como una curva con una pendiente que disminuye constantemente cuanto mayor tiempo pase 25 a 30 h la concentarcion del sustrato disminuye 1 a 0, la concentración de biomasa y producto se puede observar como una curva sigmoidal.\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "589bdad6",
   "metadata": {},
   "source": [
    "# **V. Referencia Bibliográfica**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c230b98",
   "metadata": {},
   "source": [
    "Almeida, A. (2010). Análisis de cepas recombinantes de Escherichia coli productoras de poli (3- hidroxibutirato): efecto de la acumulación del polímero y de la proteína PhaP sobre el metabolismo y la expresión génica. Universidad de Buenos Aires. Obtenido de https://bibliotecadigital.exactas.uba.ar/download/tesis/tesis_n4656_DeAlmeida.pdf\n",
    "        \n",
    "Arraño, N. (2015). Desarrollo de una Estrategia de Operación en Cultivo por Lote Alimentado. UNIVERSIDAD CATÓLICA DE VALPARAÍSO.\n",
    "\n",
    "Cabrera, E., & Díaz, E. (2018). MANUAL DE USO DE JUPYTER NOTEBOOK PARA APLICACIONES DOCENTES. Obtenido de https://eprints.ucm.es/id/eprint/48304/1/ManualJupyter.pdf\n",
    "    \n",
    "Cárcamo, M. (2013). PRODUCCIÓN DE PROTEÍNAS RECOMBINANTES EN CULTIVOS FED-BATCH DE SACCHAROMYCES CEREVISIAE Y ESCHERICHIA COLI. \n",
    "\n",
    "Honorio, C. (2019). Efecto de la concentración de azúcares reductores y extracto de levadura en la producción de polihidroxibutirato por Bacillus sp. empleando residuos de cervecería. Universidad Nacional del Santa.\n",
    "\n",
    "Loera, P. (2003). Simulación estocástica y control óptimo de procesos para el tratamiento de aguas residuales de excretas porcinas. \n",
    "\n",
    "Toro, L. (2017). Anaconda Distribution: La Suite más completa para la Ciencia de datos con Python. \n",
    "    \n",
    "López, A. (2011). biopolímeros de interés industrial síntesis y caracterización de polihidroxibutirato (phb). universidad nacional del sur.Argentina\n",
    "\n",
    "Hassan, MA, Bakhiet, EK, Hussein, HR et al. (2019).  Estudios de optimización estadística para la producción de polihidroxibutirato (PHB) por el nuevo Bacillus subtilis utilizando desechos agrícolas e industriales. En t. J. Medio Ambiente. ciencia Tecnología 16, 3497–3512 (2019). https://doi.org/10.1007/s13762-018-1900-y\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "310c86d1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
